\documentclass{article}
\usepackage{graphicx}
\usepackage{hyperref}
\usepackage{fullpage}
\usepackage{parskip} 

\title{SplitsTree Community Edition - Algorithms}
\author{Daniel H. Huson and Dave Bryant}
\date{March 2024}

\begin{document}

\maketitle


\section{List of algorithms sorted by input}


\subsection{Input Characters Block}

\subsubsection{Base Freq Distance}

The ``Base Freq Distance'' algorithm takes a Characters block as input and produces a Distances block as output. It calculates distances from differences in the base composition.

\subsubsection{Binary To Splits}

The ``Binary To Splits'' algorithm takes a Characters block as input and produces a Splits block as output. It converts binary characters directly into splits.

{\footnotesize\obeylines
\verb^MinSplitWeight =  <Double>^ - minimum split weight threshold
\verb^HighDimensionFilter =  <Boolean>^ - activate high-dimensional filter to avoid exponential graph size
\verb^AddAllTrivial =  <Boolean>^ - ensure all trival splits are present
}

See: D.H. Huson, R. Rupp and C. Scornavacca, Phylogenetic Networks, Cambridge, 2012.

\subsubsection{Characters Filter}

The ``Characters Filter'' algorithm takes a Characters block as input and produces a Characters block as output. It provides several ways of filtering characters.

{\footnotesize\obeylines
\verb^ExcludeGapSites =  <Boolean>^ - exclude all sites that contain a gap
\verb^ExcludeParsimonyUninformativeSites =  <Boolean>^ - exclude all sites that are parsimony uninformative
\verb^ExcludeConstantSites =  <Boolean>^ - exclude all sites that are constant
\verb^ExcludeFirstCodonPosition =  <Boolean>^ - exclude first and then every third site
\verb^ExcludeSecondCodonPosition =  <Boolean>^ - exclude second and then every third site
\verb^ExcludeThirdCodonPosition =  <Boolean>^ - exclude third and then every third site
}

\subsubsection{Codominant}

The ``Codominant'' algorithm takes a Characters block as input and produces a Distances block as output. It computes the co-dominant genetic distance.

See:  Smouse PE, Peakall R. Spatial autocorrelation analysis of individual multiallele and multilocus genetic structure. Heredity, 82, 561-573, 1999.

\subsubsection{Dice}

The ``Dice'' algorithm takes a Characters block as input and produces a Distances block as output. It computes distances using the Dice coefficient distance.

See:  Dice, Lee R. (1945). Measures of the Amount of Ecologic Association Between Species. Ecology. 26 (3): 297–302.

\subsubsection{DNA To Splits}

The ``DNA To Splits'' algorithm takes a Characters block as input and produces a Splits block as output. It converts DNA characters directly into splits.

{\footnotesize\obeylines
\verb^Method =  {MajorityState | RYAlphabet}^ - use either majority-state-vs-others or RY alphabet
\verb^MinSplitWeight =  <Double>^ - minimum split weight threshold
\verb^HighDimensionFilter =  <Boolean>^ - activate high-dimensional filter to avoid exponential graph size
}

See: D.H. Huson, R. Rupp and C. Scornavacca, Phylogenetic Networks, Cambridge, 2012.

\subsubsection{Estimate Invariable Sites}

The ``Estimate Invariable Sites'' algorithm takes a Characters block as input and produces a Report block as output. It estimates the proportion of invariant sites using capture-recapture.

See:  MA Steel, DH Huson, and PJ Lockhart. Invariable site models and their use in phylogeny reconstruction. Sys. Biol. 49(2):225-232, 2000

\subsubsection{External Program}

The ``External Program'' algorithm takes a Characters block as input and produces a Trees block as output. It runs an external program.

{\footnotesize\obeylines
\verb^Name =  <String>^ - specify a name for this calculation
\verb^ProgramCall =  <String>^ - specification of external program: replace 'path-to-program' by path to program and
\verb^use '%i' and '%o' as place^ - holders for the program's input and output files
\verb^CharactersFormat =  {Phylip | Nexus | FastA}^ - specify the format to write out the current data in
}

\subsubsection{F81}

The ``F81'' algorithm takes a Characters block as input and produces a Distances block as output. It calculates distances under the Felsenstein81 model.

{\footnotesize\obeylines
\verb^BaseFrequencies =  <doubleArray>^ - base frequencies (in order ACGT/U)
\verb^SetBaseFrequencies =  {fromChars | defaultValues}^ - set base frequencies to default values, or to estimations from characters (using Capture-recapture for invariable sites)
\verb^PropInvariableSites =  <Double>^ - proportion of invariable sites
\verb^SetSiteVarParams =  {fromChars | defaultValues}^ - set site variation parameters to default values, or to estimations from characters
\verb^UseML_Distances =  <Boolean>^ - use maximum likelihood estimation of distances (rather than exact distances)
}

See:  Felsenstein J (1981). Evolutionary trees from DNA sequences: a maximum likelihood approach. Journal of Molecular Evolution. 17 (6): 368–376.

\subsubsection{F84}

The ``F84'' algorithm takes a Characters block as input and produces a Distances block as output. It calculates distances under the Felsenstein84 model.

{\footnotesize\obeylines
\verb^BaseFrequencies =  <doubleArray>^ - base frequencies (in order ACGT/U)
\verb^SetBaseFrequencies =  {fromChars | defaultValues}^ - set base frequencies to default values, or to estimations from characters (using Capture-recapture for invariable sites)
\verb^PropInvariableSites =  <Double>^ - proportion of invariable sites
\verb^SetSiteVarParams =  {fromChars | defaultValues}^ - set site variation parameters to default values, or to estimations from characters
\verb^UseML_Distances =  <Boolean>^ - use maximum likelihood estimation of distances (rather than exact distances)
}

See:  Felsenstein J, Churchill GA (1996). A Hidden Markov Model approach to variation among sites in rate of evolution, and the branching order in hominoidea. Molecular Biology and Evolution. 13 (1): 93–104.

\subsubsection{GTR}

The ``GTR'' algorithm takes a Characters block as input and produces a Distances block as output. It calculates distances under the general time-reversible model.

{\footnotesize\obeylines
\verb^PropInvariableSites =  <Double>^ - proportion of invariable sites
\verb^SetSiteVarParams =  {fromChars | defaultValues}^ - set site variation parameters to default values, or to estimations from characters
\verb^RateMatrix =  <doubleSquareMatrix>^ - rate matrix for GTR (in order ACGT/U)
\verb^UseML_Distances =  <Boolean>^ - use maximum likelihood estimation of distances (rather than exact distances)
}

See:  Tavaré S (1986). Some Probabilistic and Statistical Problems in the Analysis of DNA Sequences. Lectures on Mathematics in the Life Sciences. 17: 57–86

\subsubsection{Gene Content Distance}

The ``Gene Content Distance'' algorithm takes a Characters block as input and produces a Distances block as output. It computes distances based on the presence/absence of genes.

{\footnotesize\obeylines
\verb^Method =  {MLE | SharedGenes}^ - choose Maximum likelihood distance estimation (Huson and Steel 2004, eq. 4), or shared genes distance (Snel et al, 1997)
}

See:  D.H. Huson  and  M. Steel. Phylogenetic  trees  based  on  gene  content. Bioinformatics, 20(13):2044–9, 2004.

\subsubsection{HKY85}

The ``HKY85'' algorithm takes a Characters block as input and produces a Distances block as output. It calculates distances under the Hasegawa-Kishino-Yano model.

{\footnotesize\obeylines
\verb^TsTvRatio =  <Double>^ - ratio of transitions vs transversions
\verb^BaseFrequencies =  <doubleArray>^ - base frequencies (in order ACGT/U)
\verb^SetBaseFrequencies =  {fromChars | defaultValues}^ - set base frequencies to default values, or to estimations from characters (using Capture-recapture for invariable sites)
\verb^PropInvariableSites =  <Double>^ - proportion of invariable sites
\verb^SetSiteVarParams =  {fromChars | defaultValues}^ - set site variation parameters to default values, or to estimations from characters
}

See:  Hasegawa M, Kishino H, Yano T. Dating of human-ape splitting by a molecular clock of mitochondrial DNA. Journal of Molecular Evolution. 22 (2): 160–174. PMID 3934395. doi:10.1007/BF02101694, 1985.

\subsubsection{Hamming Distances}

The ``Hamming Distances'' algorithm takes a Characters block as input and produces a Distances block as output. It computes distances based on the number of character-state differences.

{\footnotesize\obeylines
\verb^Normalize =  <Boolean>^ - normalize distances
}

See:  Hamming, Richard W. Error detecting and error correcting codes. Bell System Technical Journal. 29 (2): 147–160. MR 0035935, 1950.

\subsubsection{Hamming Distances Ambig States}

The ``Hamming Distances Ambig States'' algorithm takes a Characters block as input and produces a Distances block as output. It computes distances based on the number of character-state differences.

{\footnotesize\obeylines
\verb^Normalize =  <Boolean>^ - normalize distances
\verb^HandleAmbiguousStates =  {Ignore | AverageStates | MatchStates}^ - choose how to handle ambiguous states
}

See:  Hamming, Richard W. Error detecting and error correcting codes. Bell System Technical Journal. 29 (2): 147–160. MR 0035935, 1950.

\subsubsection{Jaccard}

The ``Jaccard'' algorithm takes a Characters block as input and produces a Distances block as output. It computes distances based on the Jaccard index.

See:  Jaccard, Paul (1901). Étude comparative de la distribution florale dans une portion des Alpes et des Jura, Bulletin de la Société Vaudoise des Sciences Naturelles, 37: 547–579.

\subsubsection{Jukes Cantor}

The ``Jukes Cantor'' algorithm takes a Characters block as input and produces a Distances block as output. It calculates distances under the Jukes-Cantor model.

{\footnotesize\obeylines
\verb^PropInvariableSites =  <Double>^ - proportion of invariable sites
\verb^SetSiteVarParams =  {fromChars | defaultValues}^ - set site variation parameters to default values, or to estimations from characters
\verb^UseML_Distances =  <Boolean>^ - use maximum likelihood estimation of distances (rather than exact distances)
}

See:  Jukes TH \& Cantor CR (1969). Evolution of Protein Molecules. New York: Academic Press. pp. 21–132

\subsubsection{K2P}

The ``K2P'' algorithm takes a Characters block as input and produces a Distances block as output. It calculates distances under the Kimura-2P model.

{\footnotesize\obeylines
\verb^TsTvRatio =  <Double>^ - ratio of transitions vs transversions
\verb^Gamma =  <Double>^ - alpha value for the Gamma distribution
\verb^PropInvariableSites =  <Double>^ - proportion of invariable sites
\verb^SetSiteVarParams =  {fromChars | defaultValues}^ - set site variation parameters to default values, or to estimations from characters
\verb^UseML_Distances =  <Boolean>^ - use maximum likelihood estimation of distances (rather than exact distances)
}

See:  Kimura M (1980). A simple method for estimating evolutionary rates of base substitutions through comparative studies of nucleotide sequences. Journal of Molecular Evolution. 16 (2): 111–120.

\subsubsection{K3ST}

The ``K3ST'' algorithm takes a Characters block as input and produces a Distances block as output. It calculates distances under the Kimura-3P model.

{\footnotesize\obeylines
\verb^TsTvRatio =  <Double>^ - ratio of transitions vs transversions
\verb^ACvATRatio =  <Double>^ - optionACvATRatio
\verb^Gamma =  <Double>^ - alpha value for the Gamma distribution
\verb^PropInvariableSites =  <Double>^ - proportion of invariable sites
\verb^SetSiteVarParams =  {fromChars | defaultValues}^ - set site variation parameters to default values, or to estimations from characters
\verb^UseML_Distances =  <Boolean>^ - use maximum likelihood estimation of distances (rather than exact distances)
}

See:  M. Kimura, Estimation of evolutionary sequences between homologous nucleotide sequences, Proc. Natl. Acad. Sci. USA 78 (1981) 454–45

\subsubsection{Log Det}

The ``Log Det'' algorithm takes a Characters block as input and produces a Distances block as output. It computes distances using the Log-Det method.

{\footnotesize\obeylines
\verb^PropInvariableSites =  <Double>^ - proportion of invariable sites
\verb^FudgeFactor =  <Boolean>^ - input missing matrix entries using LDDist method
\verb^FillZeros =  <Boolean>^ - replace zeros with small numbers in rows/columns with values
}

See:  M.A. Steel. Recovering a tree from the leaf colorations it generates under a Markov model. Appl. Math. Lett., 7(2):19–24, 1994.

\subsubsection{Median Joining}

The ``Median Joining'' algorithm takes a Characters block as input and produces a Network block as output. It computes a haplotype network using the median-joining method.

{\footnotesize\obeylines
\verb^Epsilon =  <Integer>^ - balances accuracy (smaller value) and efficiency (larger value)
}

See: H. -J. Bandelt, P. Forster, and A. Röhl. Median-joining networks for inferring intraspecific phylogenies. Molecular Biology and Evolution, 16:37–48, 1999.

\subsubsection{Nei Li Restriction Distance}

The ``Nei Li Restriction Distance'' algorithm takes a Characters block as input and produces a Distances block as output. It calculates distances for restriction data.

See: M Nei and W H Li. Mathematical model for studying genetic variation in terms of restriction endonucleases, PNAS 79(1):5269-5273, 1979.

\subsubsection{Nei Miller}

The ``Nei Miller'' algorithm takes a Characters block as input and produces a Distances block as output. It estimates the average number of nucleotide substitutions from restriction data.

See:  M. Nei and J.C. Miller. A simple method for estimating average number of nucleotide substitutions within and between populations from restriction data. Genetics, 125:873–879, 1990.

\subsubsection{Parsimony Splits}

The ``Parsimony Splits'' algorithm takes a Characters block as input and produces a Splits block as output. It computes weakly-compatible splits directly from DNA characters.

See:  H.-J.Bandelt and A.W.M.Dress. A canonical decomposition theory for metrics on a finite set. Advances in Mathematics, 92:47–105, 1992.

\subsubsection{Phi Test}

The ``Phi Test'' algorithm takes a Characters block as input and produces a Report block as output. It performs a statistical test for detecting the presence of recombination.

See:  Bruen TC, Philippe H, Bryant D. A simple and robust statistical test for detecting the presence of recombination. Genetics 17(4):2665-81, 2006

\subsubsection{Protein ML Dist}

The ``Protein ML Dist'' algorithm takes a Characters block as input and produces a Distances block as output. It computes distances for proteins using maximum-likelihood estimation.

{\footnotesize\obeylines
\verb^Model =  {cpREV45 | Dayhoff | JTT | mtMAM | mtREV24 | pmb | Rhodopsin | WAG}^ - choose an amino acid substitution model
\verb^PropInvariableSites =  <Double>^ - proportion of invariable sites
\verb^Gamma =  <Double>^ - alpha parameter for gamma distribution. Negative gamma = Equal rates
}

See:  D.L. Swofford, G.J. Olsen, P.J. Waddell, and  D.M. Hillis. Chapter  11:  Phylogenetic inference. In D. M. Hillis, C. Moritz, and B. K. Mable, editors, Molecular Systematics, pages 407–514. Sinauer Associates, Inc., 2nd edition, 1996.

\subsubsection{Uncorrected P}

The ``Uncorrected P'' algorithm takes a Characters block as input and produces a Distances block as output. It computes distances based on the number of character-state differences.

{\footnotesize\obeylines
\verb^Normalize =  <Boolean>^ - normalize distances
}

See:  Hamming, Richard W. Error detecting and error correcting codes. Bell System Technical Journal. 29 (2): 147–160. MR 0035935, 1950.

\subsubsection{Upholt}

The ``Upholt'' algorithm takes a Characters block as input and produces a Distances block as output. It calculates distances for restriction data.

See:  Upholt WB. Estimation of DNA sequence divergence from comparison of restriction endonuclease digests. Nucleic Acids Res. 1977;4(5):1257-65.

\subsection{Input Distances Block}

\subsubsection{Bio NJ}

The ``Bio NJ'' algorithm takes a Distances block as input and produces a Trees block as output. It computes an unrooted phylogenetic tree using the Bio-NJ method.

See:  O. Gascuel, BIONJ: an improved version of the NJ algorithm based on a simple model of sequence data. Molecular Biology and Evolution. 1997 14:685-695.

\subsubsection{Buneman Tree}

The ``Buneman Tree'' algorithm takes a Distances block as input and produces a Splits block as output. It computes a set of compatible splits using the Buneman tree method.

See:  H.-J. Bandelt and A.W.M.Dress. A canonical decomposition theory for metrics on a finite set. Advances in Mathematics, 92:47–105, 1992.

\subsubsection{Delta Score}

The ``Delta Score'' algorithm takes a Distances block as input and produces a Report block as output. It calculates the delta score.

See: B. R. Holland, K. T. Huber, A. Dress, V. Moulton, Delta Plots: A tool for analyzing phylogenetic distance data, Molecular Biology and Evolution, 19(12):2051–2059, 2002.

\subsubsection{Min Spanning Network}

The ``Min Spanning Network'' algorithm takes a Distances block as input and produces a Network block as output. It computes a minimum spanning network.

{\footnotesize\obeylines
\verb^Epsilon =  <Double>^ - weighted genetic distance measure. Low: MedianJoining, High: full median network
\verb^MinSpanningTree =  <Boolean>^ - calculate minimum spanning tree
}

See:  Excoffier L, Smouse PE. Using allele frequencies and geographic subdivision to reconstruct gene trees within a species: molecular variance parsimony (1994) Genetics.136(1):343-59.

\subsubsection{Min Spanning Tree}

The ``Min Spanning Tree'' algorithm takes a Distances block as input and produces a Trees block as output. It computes a minimum spanning tree.

See:  Excoffier L, Smouse PE. Using allele frequencies and geographic subdivision to reconstruct gene trees within a species: molecular variance parsimony (1994) Genetics.136(1):343-59.

\subsubsection{Neighbor Joining}

The ``Neighbor Joining'' algorithm takes a Distances block as input and produces a Trees block as output. It computes an unrooted phylogenetic tree using the neighbor-joining method.

See:  N. Saitou and M. Nei. The Neighbor-Joining method: a new method for reconstructing phylogenetic trees. Molecular Biology and Evolution, 4:406-425, 1987.

\subsubsection{Neighbor Net}

The ``Neighbor Net'' algorithm takes a Distances block as input and produces a Splits block as output. It computes a set of cyclic splits using the neighbor-net method.

{\footnotesize\obeylines
\verb^InferenceAlgorithm =  {GradientProjection | ActiveSet | APGD | SplitsTree4}^ - the inference algorithm to be used
}

See:  D. Bryant and V. Moulton. Neighbor-net: An agglomerative method for the construction of phylogenetic networks. Molecular Biology and Evolution, 21(2):255– 265, 2004.;Bryant \& Huson 2023;D. Bryant and D.H. Huson, NeighborNet- improved algorithms and implementation. Front. Bioinform. 3, 2023

\subsubsection{P Co A}

The ``P Co A'' algorithm takes a Distances block as input and produces a Network block as output. It performs principal coordinates analysis.

{\footnotesize\obeylines
\verb^FirstCoordinate =  <Integer>^ - choose principal component for the x Axis
\verb^SecondCoordinate =  <Integer>^ - choose principal component for the y Axis
}

See:  Gower, J. C. (1966). Some distance properties of latent root and vector methods used in multivariate analysis. Biometrika, 53(3-4), 325-338.

\subsubsection{Split Decomposition}

The ``Split Decomposition'' algorithm takes a Distances block as input and produces a Splits block as output. It computes a set of weakly-compatible splits using the split-decomposition method.

See:  H.-J.Bandelt and A.W.M.Dress. A canonical decomposition theory for metrics on a finite set. Advances in Mathematics, 92:47–105, 1992.

\subsubsection{UPGMA}

The ``UPGMA'' algorithm takes a Distances block as input and produces a Trees block as output. It computes a rooted phylogenetic tree using the UPGMA method.

See:  R.R. Sokal and C.D.Michener. A statistical method for evaluating systematic relationships. University of Kansas Scientific Bulletin, 28:1409-1438, 1958.

\subsection{Input Splits Block}

\subsubsection{Bootstrap Splits}

The ``Bootstrap Splits'' algorithm takes a Splits block as input and produces a Splits block as output. It performs bootstrapping on splits.

{\footnotesize\obeylines
\verb^Replicates =  <Integer>^ - number of bootstrap replicates
\verb^MinPercent =  <Double>^ - minimum percentage support for a split to be included
\verb^ShowAllSplits =  <Boolean>^ - show all bootstrap splits, not just the original splits
\verb^RandomSeed =  <Integer>^ - if non-zero, is used as seed for random number generator
\verb^HighDimensionFilter =  <Boolean>^ - heuristically remove splits causing high-dimensional network
}

See: Felsenstein J. Confidence limits on phylogenies: an approach using the bootstrap. Evolution. 1985;39(4):783-791

\subsubsection{Dimension Filter}

The ``Dimension Filter'' algorithm takes a Splits block as input and produces a Splits block as output. It heuristically removes splits that lead to high-dimensional boxes in a split network.

{\footnotesize\obeylines
\verb^MaxDimension =  <Integer>^ - heuristically remove splits that create configurations of a higher dimension than this threshold
}

\subsubsection{Greedy Tree}

The ``Greedy Tree'' algorithm takes a Splits block as input and produces a Trees block as output. It produces a phylogenetic tree based on greedily selecting a compatible set of splits.

See: D.H. Huson, R. Rupp and C. Scornavacca, Phylogenetic Networks, Cambridge, 2012.

\subsubsection{Phylogenetic Diversity}

The ``Phylogenetic Diversity'' algorithm takes a Splits block as input and produces a Report block as output. It calculates the phylogenetic diversity for selected taxa.

See: Volkmann L, Martyn I, Moulton V, Spillner A, Mooers AO. Prioritizing populations for conservation using phylogenetic networks. PLoS ONE 9(2):e88945 (2014)

\subsubsection{Shapley Values}

The ``Shapley Values'' algorithm takes a Splits block as input and produces a Report block as output. It calculates Shapley values on splits.

See: Volkmann L, Martyn I, Moulton V, Spillner A, Mooers AO. Prioritizing populations for conservation using phylogenetic networks. PLoS ONE 9(2):e88945 (2014)

\subsubsection{Show Splits}

The ``Show Splits'' algorithm takes a Splits block as input and produces a View block as output. It provides interactive visualizations of split networks.

{\footnotesize\obeylines
\verb^View =  {SplitsNetwork}^ - the type of splits viewer to use
}

\subsubsection{Splits Filter}

The ``Splits Filter'' algorithm takes a Splits block as input and produces a Splits block as output. It provides several ways of filtering splits.

{\footnotesize\obeylines
\verb^WeightThreshold =  <Float>^ - set minimum split weight threshold
\verb^ConfidenceThreshold =  <Float>^ - set the minimum split confidence threshold
\verb^MaximumDimension =  <Integer>^ - set maximum dimension threshold (necessary to avoid computational overload)
\verb^FilterAlgorithm =  {None | GreedyCompatible | GreedyCircular | GreedyWeaklyCompatible | BlobTree}^ - set the filter algorithm
\verb^RecomputeCycle =  <Boolean>^ - recompute circular ordering
}

\subsubsection{Weights Slider}

The ``Weights Slider'' algorithm takes a Splits block as input and produces a Splits block as output. It allows one to interactively filter splits by their weight.

{\footnotesize\obeylines
\verb^WeightThreshold =  <Double>^ - set minimum split weight threshold
}

\subsection{Input Trees Block}

\subsubsection{ALTS External}

The ``ALTS External'' algorithm takes a Trees block as input and produces a Trees block as output. It runs an external implementation of the ALTS algorithm.

{\footnotesize\obeylines
\verb^ALTSExecutableFile =  <String>^ - download and compile ALTSNetwork program from https://github.com/LX-Zhang/AAST, then set this parameter to the executable.
Note that the program requires fully resolved trees as input and any unresolved trees will be ignored
}

See:  Louxin Zhang, Niloufar Niloufar Abhari, Caroline Colijn and Yufeng Wu3. A fast and scalable method for inferring phylogenetic networks from trees by aligning lineage taxon strings. Genome Res. 2023

\subsubsection{ALTS Network}

The ``ALTS Network'' algorithm takes a Trees block as input and produces a Trees block as output. It computes one or more rooted networks that contain all input trees using the M-ALTS algorithm.

{\footnotesize\obeylines
\verb^MutualRefinement =  <Boolean>^ - mutually refine trees during preprocessing
\verb^RemoveDuplicates =  <Boolean>^ - remove duplicate networks in output
\verb^Kernelization =  <Boolean>^ - perform kernelization during preprocessing
}

See:  Louxin Zhang, Niloufar Niloufar Abhari, Caroline Colijn and Yufeng Wu3. A fast and scalable method for inferring phylogenetic networks from trees by aligning lineage taxon strings. Genome Res. 2023;Zhang et al, 2024; Louxin Zhang, Banu Cetinkaya and Daniel H Huson. Hybrization networks from multiple trees, in preparation.

\subsubsection{Autumn Algorithm}

The ``Autumn Algorithm'' algorithm takes a Trees block as input and produces a Trees block as output. It computes all minimum hybridization networks using the Autumn algorithm

{\footnotesize\obeylines
\verb^FirstTree =  <Integer>^ - index of the first tree
\verb^SecondTree =  <Integer>^ - index of the second tree
}

See:  D.H. Huson and S. Linz. Autumn Algorithm—Computation of Hybridization Networks for Realistic Phylogenetic Trees. IEEE/ACM Transactions on Computational Biology and Bioinformatics: 15:398-420, 2018.

\subsubsection{Average Consensus}

The ``Average Consensus'' algorithm takes a Trees block as input and produces a Splits block as output. It calculates average consensus tree.

See: Francois-Joseph Lapointe, Guy Cucumel, The Average Consensus Procedure: Combination of Weighted Trees Containing Identical or Overlapping Sets of Taxa. Systematic Biology, 46(2):306-312 (1997).

\subsubsection{Average Distances}

The ``Average Distances'' algorithm takes a Trees block as input and produces a Distances block as output. It calculates the average distances between taxa over a set of trees.

See: Francois-Joseph Lapointe, Guy Cucumel, The Average Consensus Procedure: Combination of Weighted Trees Containing Identical or Overlapping Sets of Taxa. Systematic Biology, 46(2):306-312 (1997).

\subsubsection{Bootstrap Tree}

The ``Bootstrap Tree'' algorithm takes a Trees block as input and produces a Trees block as output. It performs bootstrapping on trees.

{\footnotesize\obeylines
\verb^Replicates =  <Integer>^ - number of bootstrap replicates
\verb^TransferBootstrap =  <Boolean>^ - use transform bootstrapping (TBE), less susceptible to rouge taxa
\verb^MinPercent =  <Double>^ - minimum percentage support for a branch to be included
\verb^RandomSeed =  <Integer>^ - if non-zero, is used as seed for random number generator
}

See: Felsenstein J. Confidence limits on phylogenies: an approach using the bootstrap. Evolution, 39(4):783-791 (1985);

\subsubsection{Bootstrap Tree Splits}

The ``Bootstrap Tree Splits'' algorithm takes a Trees block as input and produces a Splits block as output. It performs bootstrapping on trees.

{\footnotesize\obeylines
\verb^Replicates =  <Integer>^ - number of bootstrap replicates
\verb^MinPercent =  <Double>^ - minimum percentage support for a split to be included
\verb^ShowAllSplits =  <Boolean>^ - show all bootstrap splits, not just the original splits
\verb^RandomSeed =  <Integer>^ - if non-zero, is used as seed for random number generator
\verb^HighDimensionFilter =  <Boolean>^ - heuristically remove splits causing high-dimensional network
}

See: Felsenstein J. Confidence limits on phylogenies: an approach using the bootstrap. Evolution, 39(4):783-791 (1985);

\subsubsection{Cluster Network}

The ``Cluster Network'' algorithm takes a Trees block as input and produces a Trees block as output. It computes the cluster network that contains all input trees (in the hardwired sense).

{\footnotesize\obeylines
\verb^EdgeWeights =  {Mean | Count | Sum | Uniform}^ - compute edge weights
\verb^ThresholdPercent =  <Double>^ - minimum percentage of trees that a cluster must appear in
}

See:  D.H. Huson and R. Rupp (2008) Summarizing multiple gene trees using cluster networks. In: Crandall, K.A., Lagergren, J. (eds) Algorithms in Bioinformatics. WABI 2008. Lecture Notes in Computer Science(), vol 5251.

\subsubsection{Consensus Network}

The ``Consensus Network'' algorithm takes a Trees block as input and produces a Splits block as output. It computes the consensus network.

{\footnotesize\obeylines
\verb^EdgeWeights =  {Mean | TreeSizeWeightedMean | Median | Count | Sum | Uniform | TreeNormalizedSum}^ - how to calculate edge weights in resulting network
\verb^ThresholdPercent =  <Double>^ - threshold for percent of input trees that split has to occur in for it to appear in the output
\verb^HighDimensionFilter =  <Boolean>^ - heuristically remove splits causing high-dimensional consensus network
}

See:  B. Holland and V. Moulton. Consensus networks:  A method for visualizing incompatibilities in  collections  of  trees. In  G.  Benson  and  R.  Page,  editors, Proceedings  of  “Workshop  on Algorithms in Bioinformatics, volume 2812 of LNBI, pages 165–176. Springer, 2003.

\subsubsection{Consensus Outline}

The ``Consensus Outline'' algorithm takes a Trees block as input and produces a Splits block as output. It computes the consensus outline.

{\footnotesize\obeylines
\verb^EdgeWeights =  {Mean | TreeSizeWeightedMean | Median | Count | Sum | Uniform | TreeNormalizedSum}^ - how to calculate edge weights in resulting network
\verb^ThresholdPercent =  <Double>^ - threshold for percent of input trees that split has to occur in for it to appear in the output
}

See: DH Huson and B Cetinkaya, Visualizing incompatibilities in phylogenetic trees using consensus outlines, Front. Bioinform. (2023) 

\subsubsection{Consensus Splits}

The ``Consensus Splits'' algorithm takes a Trees block as input and produces a Splits block as output. It provides several consensus methods.

{\footnotesize\obeylines
\verb^Consensus =  {Strict | Majority | GreedyCompatible | ConsensusOutline | GreedyWeaklyCompatible | ConsensusNetwork}^ - consensus method
\verb^EdgeWeights =  {Mean | TreeSizeWeightedMean | Median | Count | Sum | Uniform | TreeNormalizedSum}^ - how to calculate edge weights in resulting network
\verb^ThresholdPercent =  <Double>^ - threshold for percent of input trees that split has to occur in for it to appear in the output
\verb^HighDimensionFilter =  <Boolean>^ - heuristically remove splits causing high-dimensional consensus network
}

See: D.H. Huson, R. Rupp and C. Scornavacca, Phylogenetic Networks, Cambridge, 2012.

\subsubsection{Consensus Tree}

The ``Consensus Tree'' algorithm takes a Trees block as input and produces a Trees block as output. It provides several methods for computing a consensus tree.

{\footnotesize\obeylines
\verb^Consensus =  {Majority | Greedy | Strict}^ - consensus method to use
\verb^EdgeWeights =  {Mean | TreeSizeWeightedMean | Median | Count | Sum | Uniform | TreeNormalizedSum}^ - determine how to calculate edge weights in resulting network
}

See: D. Bryant, A classification of consensus methods for phylogenetics, in Bioconsensus, 2001

\subsubsection{Enumerate Contained Trees}

The ``Enumerate Contained Trees'' algorithm takes a Trees block as input and produces a Trees block as output. It enumerates all contained trees.

{\footnotesize\obeylines
\verb^RemoveDuplicates =  <Boolean>^ - suppress duplicate trees in output
}

\subsubsection{Filtered Super Network}

The ``Filtered Super Network'' algorithm takes a Trees block as input and produces a Splits block as output. It computes a super network using the Z-closure method.

{\footnotesize\obeylines
\verb^MinNumberTrees =  <Integer>^ - set the min number trees
\verb^MaxDistortionScore =  <Integer>^ - set the max distortion score
\verb^UseTotalScore =  <Boolean>^ - set the use total score
}

See: James B. Whitfield, Sydney A. Cameron, Daniel H. Huson, Mike A. Steel. Filtered Z-Closure Supernetworks for Extracting and Visualizing Recurrent Signal from Incongruent Gene Trees, Systematic Biology, Volume 57, Issue 6, 1 December 2008, Pages 939–947.

\subsubsection{List One RSPR Trees}

The ``List One RSPR Trees'' algorithm takes a Trees block as input and produces a Report block as output. It determines which trees are exactly on rSPR away from each other.

{\footnotesize\obeylines
\verb^ApplyTo =  {OneTree | AllTrees}^ - determine whether to apply to one or all trees
\verb^WhichTree =  <Integer>^ - the index of the tree that the method will be applied to
}

\subsubsection{Loose And Lacy}

The ``Loose And Lacy'' algorithm takes a Trees block as input and produces a Trees block as output. It computes the `loose' and `lacy' species for a given tree and taxon trait.

{\footnotesize\obeylines
\verb^SpeciesDefinition =  {Loose | Lacy | Both}^ - species definition to use
\verb^TraitNumber =  <Integer>^ - number of specific trait to use
\verb^UseAllTraits =  <Boolean>^ - use all traits
}

See: Anica Hoppe, Sonja Tuerpitz, Mike Steel. Species notions that combine phylogenetic trees and phenotypic partitions. arXiv:1711.08145v1.

\subsubsection{Phylogenetic Diversity}

The ``Phylogenetic Diversity'' algorithm takes a Trees block as input and produces a Report block as output. It calculates the phylogenetic diversity for selected taxa.

{\footnotesize\obeylines
\verb^Rooted =  <Boolean>^ - interpret trees as rooted?
\verb^ApplyTo =  {OneTree | AllTrees}^ - determine whether to apply to one or all trees
\verb^WhichTree =  <Integer>^ - the index of the tree that the method will be applied to
}

See: Faith, D.P. Conservation evaluation and phylogenetic diversity. Biological Conservation 61, 1–10 (1992)

\subsubsection{Reroot Or Reorder Trees}

The ``Reroot Or Reorder Trees'' algorithm takes a Trees block as input and produces a Trees block as output. It reroot, or change the order of children, on all trees.

{\footnotesize\obeylines
\verb^RootBy =  {Off | MidPoint | OutGroup}^ - determine how to reroot
\verb^RearrangeBy =  {Off | RotateChildren | RotateSubTrees | ReverseChildren | ReverseSubTrees}^ - determine how to rearrange
\verb^Reorder =  {Off | ByTaxa | Lexicographically | ReverseOrder | LadderizedUp | LadderizedDown | LadderizedRandom | Stabilize}^ - determine how to reorder
}

\subsubsection{Rooted Consensus Tree}

The ``Rooted Consensus Tree'' algorithm takes a Trees block as input and produces a Trees block as output. It provides several methods for computing a rooted consensus tree.

{\footnotesize\obeylines
\verb^Consensus =  {Majority | Strict | Greedy}^ - consensus method to use
}

\subsubsection{Show Trees}

The ``Show Trees'' algorithm takes a Trees block as input and produces a View block as output. It provides several types of interactive visualizations of trees.

{\footnotesize\obeylines
\verb^View =  {TreeView | TreePages | Tanglegram | DensiTree}^ - the type of viewer to use
}

\subsubsection{Super Network}

The ``Super Network'' algorithm takes a Trees block as input and produces a Splits block as output. It computes a super network using the Z-closure method.

{\footnotesize\obeylines
\verb^EdgeWeights =  {AverageRelative | Mean | TreeSizeWeightedMean | Sum | Min | None}^ - determine how to calculate edge weights in resulting network
\verb^SuperTree =  <Boolean>^ - enforce the strong induction property, which results in a super tree
\verb^NumberOfRuns =  <Integer>^ - number of runs using random permutations of the input splits
\verb^ApplyRefineHeuristic =  <Boolean>^ - apply a simple refinement heuristic
\verb^Seed =  <Integer>^ - set seed used for random permutations
\verb^HighDimensionFilter =  <Boolean>^ - heuristically remove splits causing high-dimensional network
}

See: D.H. Huson, T. Dezulian, T. Kloepper, and M. A. Steel. Phylogenetic super-networks from partial trees. IEEE/ACM Transactions in Computational Biology and Bioinformatics, 1(4):151–158, 2004.

\subsubsection{Tree Diversity Index}

The ``Tree Diversity Index'' algorithm takes a Trees block as input and produces a Report block as output. It calculates the fair-proportion and equal-splits values on trees.

{\footnotesize\obeylines
\verb^Method =  {FairProportions | EqualSplits}^ - choose the type of index calculation
\verb^ApplyTo =  {OneTree | AllTrees}^ - determine whether to apply to one or all trees
\verb^WhichTree =  <Integer>^ - the index of the tree that the method will be applied to
}

See: Redding, D. Incorporating genetic distinctness and reserve occupancy into a conservation priorisation approach. Master’s thesis. University of East Anglia (2003)

\subsubsection{Tree Selector}

The ``Tree Selector'' algorithm takes a Trees block as input and produces a Trees block as output. It allows the user to select one from a list of trees.

{\footnotesize\obeylines
\verb^Which =  <Integer>^ - which tree to use
}

\subsubsection{Tree Selector Splits}

The ``Tree Selector Splits'' algorithm takes a Trees block as input and produces a Splits block as output. It selects a single tree and extracts its splits.

{\footnotesize\obeylines
    Which =  <Integer>
}

\subsubsection{Trees Filter}

The ``Trees Filter'' algorithm takes a Trees block as input and produces a Trees block as output. It allows the user to interactively filter trees.

\subsubsection{Trees Filter 2}

The ``Trees Filter 2'' algorithm takes a Trees block as input and produces a Trees block as output. It provides several options for filtering trees.

{\footnotesize\obeylines
\verb^RequireAllTaxa =  <Boolean>^ - keep only trees that have the full set of taxa
\verb^MinNumberOfTaxa =  <Integer>^ - keep only trees that have at least this number of taxa
\verb^MinNumberOfTaxa =  <Integer>^ - keep only trees that have at least this number of taxa
\verb^MinEdgeLength =  <Double>^ - keep only edges that have this minimum length
\verb^MinConfidence =  <Double>^ - keep only edges that have this minimum confidence value
\verb^UniformEdgeLengths =  <Boolean>^ - change all edge weights to 1
}

\subsubsection{Unrooted Shapley Values}

The ``Unrooted Shapley Values'' algorithm takes a Trees block as input and produces a Report block as output. It calculates unrooted Shapley values on trees.

{\footnotesize\obeylines
\verb^ApplyTo =  {OneTree | AllTrees}^ - determine whether to apply to one or all trees
\verb^WhichTree =  <Integer>^ - the index of the tree that the method will be applied to
}

See: Haake C.J., Kashiwada A., Su F.E. The Shapley value of phylogenetic trees. J Math Biol 56:479–497 (2008) 

\subsection{Input Network Block}

\subsubsection{Show Network}

The ``Show Network'' algorithm takes a Network block as input and produces a View block as output. It provides interactive visualizations of networks.

{\footnotesize\obeylines
\verb^View =  {Network | Text}^ - the type of network viewer to use
}

\end{document}