<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
        "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head><title> SplitsTree App User Manual</title>
     <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
     <meta name="generator" content="TeX4ht (http://www.tug.org/tex4ht/)">
     <meta name="originator" content="TeX4ht (http://www.tug.org/tex4ht/)">
     <!-- html -->
     <meta name="src" content="manual.tex">
     <meta name="date" content="2025-10-16 12:19:00">
     <link rel="stylesheet" type="text/css" href="manual.css">
     <style> body {
          font-family: Arial, sans-serif;
     }</style>
</head>
<body
>
<div class="maketitle">


     <h2 class="titleHead"> SplitsTree App User Manual</h2>
     <div class="author"><span
             class="cmss-12">Daniel H. Huson and David Bryant</span></div>
     <br/>
     <div class="date"><span
             class="cmss-12">SplitsTree App (version 6.6.0, built 16 Oct 2025)  </span></div>


</div>


<h2 class="likechapterHead"><a
        id="x1-1000"></a>Contents</h2>
<div class="tableofcontents">
<span class="chapterToc">1 <a
        href="#x1-30001" id="QQ2-1-4">Using SplitsTree</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.1 <a
        href="#x1-40001.1" id="QQ2-1-5">Getting started</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.2 <a
        href="#x1-50001.2" id="QQ2-1-7">Layout of the main window</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.3 <a
        href="#x1-60001.3" id="QQ2-1-9">Main toolbar items</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.4 <a
        href="#x1-70001.4" id="QQ2-1-11">The main tabs</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.5 <a
        href="#x1-80001.5" id="QQ2-1-12">Alignment tab</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.6 <a
        href="#x1-90001.6" id="QQ2-1-14">Tree-View tab</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.7 <a
        href="#x1-100001.7" id="QQ2-1-16">Tree-Pages tab</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.8 <a
        href="#x1-110001.8" id="QQ2-1-18">Tanglegram tab</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.9 <a
        href="#x1-120001.9" id="QQ2-1-20">DensiTree tab</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.10 <a
        href="#x1-130001.10" id="QQ2-1-22">Split-Network tab</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.11 <a
        href="#x1-140001.11" id="QQ2-1-24">Network tab</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.12 <a
        href="#x1-150001.12" id="QQ2-1-26">World map tab</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.13 <a
        href="#x1-160001.13" id="QQ2-1-28">Workflow graph tab</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.14 <a
        href="#x1-170001.14" id="QQ2-1-30">How to cite tab</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.15 <a
        href="#x1-180001.15" id="QQ2-1-32">Input editor tab</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.16 <a
        href="#x1-190001.16" id="QQ2-1-34">Report tabs</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.17 <a
        href="#x1-200001.17" id="QQ2-1-36">Text tabs</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.18 <a
        href="#x1-210001.18" id="QQ2-1-38">The sidebar</a></span>
     <br/>&#x00A0;<span class="sectionToc">1.19 <a
        href="#x1-220001.19" id="QQ2-1-39">The draft genome dialog</a></span>
     <br/><span class="chapterToc">2 <a
        href="#x1-230002" id="QQ2-1-41">Building trees and networks</a></span>
     <br/>&#x00A0;<span class="sectionToc">2.1 <a
        href="#x1-240002.1" id="QQ2-1-42">Using the workflow</a></span>
     <br/>&#x00A0;<span class="sectionToc">2.2 <a
        href="#x1-250002.2" id="QQ2-1-46">Building trees</a></span>
     <br/>&#x00A0;<span class="sectionToc">2.3 <a
        href="#x1-260002.3" id="QQ2-1-47">Neighbor Net and other split network methods</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">2.3.1 <a
        href="#x1-270002.3.1" id="QQ2-1-48">Neighbor Net</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">2.3.2 <a
        href="#x1-280002.3.2" id="QQ2-1-49">Manipulating split networks</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">2.3.3 <a
        href="#x1-290002.3.3" id="QQ2-1-51">Split Decomposition</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">2.3.4 <a
        href="#x1-300002.3.4" id="QQ2-1-52">Splits in characters</a></span>


     <br/>&#x00A0;<span class="sectionToc">2.4 <a
        href="#x1-310002.4" id="QQ2-1-53">Haplotype networks</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">2.4.1 <a
        href="#x1-320002.4.1" id="QQ2-1-54">Minimum spanning network</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">2.4.2 <a
        href="#x1-330002.4.2" id="QQ2-1-55">Median Joining network</a></span>
     <br/>&#x00A0;<span class="sectionToc">2.5 <a
        href="#x1-340002.5" id="QQ2-1-56">Rooted phylogenetic networks</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">2.5.1 <a
        href="#x1-350002.5.1" id="QQ2-1-57">Implicit vs explicit trees and networks</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">2.5.2 <a
        href="#x1-360002.5.2" id="QQ2-1-58">Hybridization networks</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">2.5.3 <a
        href="#x1-370002.5.3" id="QQ2-1-60">How to compute a rooted network from rooted trees</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">2.5.4 <a
        href="#x1-380002.5.4" id="QQ2-1-62">Cluster networks</a></span>
     <br/><span class="chapterToc">3 <a
        href="#x1-390003" id="QQ2-1-63">Consensus trees and networks</a></span>
     <br/>&#x00A0;<span class="sectionToc">3.1 <a
        href="#x1-400003.1" id="QQ2-1-64">Consensus trees</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">3.1.1 <a
        href="#x1-410003.1.1" id="QQ2-1-65">Average consensus method</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">3.1.2 <a
        href="#x1-420003.1.2" id="QQ2-1-66">Strict-, majority- and greedy-consensus methods</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">3.1.3 <a
        href="#x1-430003.1.3" id="QQ2-1-67">Densi-tree consensus</a></span>
     <br/>&#x00A0;<span class="sectionToc">3.2 <a
        href="#x1-440003.2" id="QQ2-1-68">Networks representing trees</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">3.2.1 <a
        href="#x1-450003.2.1" id="QQ2-1-69">Consensus networks</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">3.2.2 <a
        href="#x1-460003.2.2" id="QQ2-1-70">Consensus outline</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">3.2.3 <a
        href="#x1-470003.2.3" id="QQ2-1-72">Confidence networks</a></span>
     <br/><span class="appendixToc">A <a
        href="#x1-48000A" id="QQ2-1-73">The main menu bar</a></span>
     <br/>&#x00A0;<span class="sectionToc">A.1 <a
        href="#x1-49000A.1" id="QQ2-1-74">The File menu</a></span>
     <br/>&#x00A0;<span class="sectionToc">A.2 <a
        href="#x1-50000A.2" id="QQ2-1-75">The Edit menu</a></span>
     <br/>&#x00A0;<span class="sectionToc">A.3 <a
        href="#x1-51000A.3" id="QQ2-1-76">The Select menu</a></span>
     <br/>&#x00A0;<span class="sectionToc">A.4 <a
        href="#x1-52000A.4" id="QQ2-1-77">The View menu</a></span>
     <br/>&#x00A0;<span class="sectionToc">A.5 <a
        href="#x1-53000A.5" id="QQ2-1-78">The Data menu</a></span>
     <br/>&#x00A0;<span class="sectionToc">A.6 <a
        href="#x1-54000A.6" id="QQ2-1-79">The Distances menu</a></span>
     <br/>&#x00A0;<span class="sectionToc">A.7 <a
        href="#x1-55000A.7" id="QQ2-1-80">The Tree menu</a></span>
     <br/>&#x00A0;<span class="sectionToc">A.8 <a
        href="#x1-56000A.8" id="QQ2-1-81">The Network menu</a></span>
     <br/>&#x00A0;<span class="sectionToc">A.9 <a
        href="#x1-57000A.9" id="QQ2-1-82">The Analysis menu</a></span>
     <br/>&#x00A0;<span class="sectionToc">A.10 <a
        href="#x1-58000A.10" id="QQ2-1-83">The Window menu</a></span>
     <br/>&#x00A0;<span class="sectionToc">A.11 <a
        href="#x1-59000A.11" id="QQ2-1-84">The Help menu</a></span>
     <br/><span class="appendixToc">B <a
        href="#x1-60000B" id="QQ2-1-85">Main data blocks</a></span>
     <br/>&#x00A0;<span class="sectionToc">B.1 <a
        href="#x1-61000B.1" id="QQ2-1-86">Taxa block</a></span>
     <br/>&#x00A0;<span class="sectionToc">B.2 <a
        href="#x1-62000B.2" id="QQ2-1-87">Traits block</a></span>
     <br/>&#x00A0;<span class="sectionToc">B.3 <a
        href="#x1-63000B.3" id="QQ2-1-88">Characters block</a></span>
     <br/>&#x00A0;<span class="sectionToc">B.4 <a
        href="#x1-64000B.4" id="QQ2-1-89">Distances block</a></span>
     <br/>&#x00A0;<span class="sectionToc">B.5 <a
        href="#x1-65000B.5" id="QQ2-1-90">Trees block</a></span>


     <br/>&#x00A0;<span class="sectionToc">B.6 <a
        href="#x1-66000B.6" id="QQ2-1-91">Splits block</a></span>
     <br/>&#x00A0;<span class="sectionToc">B.7 <a
        href="#x1-67000B.7" id="QQ2-1-92">Network block</a></span>
     <br/>&#x00A0;<span class="sectionToc">B.8 <a
        href="#x1-68000B.8" id="QQ2-1-93">View block</a></span>
     <br/>&#x00A0;<span class="sectionToc">B.9 <a
        href="#x1-69000B.9" id="QQ2-1-94">Algorithms block</a></span>
     <br/>&#x00A0;<span class="sectionToc">B.10 <a
        href="#x1-70000B.10" id="QQ2-1-95">Report block</a></span>
     <br/>&#x00A0;<span class="sectionToc">B.11 <a
        href="#x1-71000B.11" id="QQ2-1-96">Sets block</a></span>
     <br/>&#x00A0;<span class="sectionToc">B.12 <a
        href="#x1-72000B.12" id="QQ2-1-97">SplitsTree6 block</a></span>
     <br/>&#x00A0;<span class="sectionToc">B.13 <a
        href="#x1-73000B.13" id="QQ2-1-98">Genomes block</a></span>
     <br/><span class="appendixToc">C <a
        href="#x1-74000C" id="QQ2-1-99">Algorithms</a></span>
     <br/>&#x00A0;<span class="sectionToc">C.1 <a
        href="#x1-75000C.1" id="QQ2-1-100">Algorithms on a Characters Block</a></span>
     <br/>&#x00A0;<span class="sectionToc">C.2 <a
        href="#x1-102000C.2" id="QQ2-1-127">Algorithms on a Distances Block</a></span>
     <br/>&#x00A0;<span class="sectionToc">C.3 <a
        href="#x1-113000C.3" id="QQ2-1-138">Algorithms on a Splits Block</a></span>
     <br/>&#x00A0;<span class="sectionToc">C.4 <a
        href="#x1-123000C.4" id="QQ2-1-148">Algorithms on a Trees Block</a></span>
     <br/>&#x00A0;<span class="sectionToc">C.5 <a
        href="#x1-157000C.5" id="QQ2-1-182">Algorithms on a Network Block</a></span>
     <br/><span class="appendixToc">D <a
        href="#x1-159000D" id="QQ2-1-184">Supported import and export formats</a></span>
     <br/>&#x00A0;<span class="sectionToc">D.1 <a
        href="#x1-160000D.1" id="QQ2-1-185">Supported import formats</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.1.1 <a
        href="#x1-161000D.1.1" id="QQ2-1-186">Importers for a characters block</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.1.2 <a
        href="#x1-162000D.1.2" id="QQ2-1-187">Importers for a distances block</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.1.3 <a
        href="#x1-163000D.1.3" id="QQ2-1-188">Importers for a trees block</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.1.4 <a
        href="#x1-164000D.1.4" id="QQ2-1-189">Importers for a splits block</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.1.5 <a
        href="#x1-165000D.1.5" id="QQ2-1-190">Importers for a network block</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.1.6 <a
        href="#x1-166000D.1.6" id="QQ2-1-191">Importers for a genomes block</a></span>
     <br/>&#x00A0;<span class="sectionToc">D.2 <a
        href="#x1-167000D.2" id="QQ2-1-192">Supported output formats</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.2.1 <a
        href="#x1-168000D.2.1" id="QQ2-1-193">Exporters for a taxa block</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.2.2 <a
        href="#x1-169000D.2.2" id="QQ2-1-194">Exporters for a characters block</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.2.3 <a
        href="#x1-170000D.2.3" id="QQ2-1-195">Exporters for a distances block</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.2.4 <a
        href="#x1-171000D.2.4" id="QQ2-1-196">Exporters for a trees block</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.2.5 <a
        href="#x1-172000D.2.5" id="QQ2-1-197">Exporters for a splits block</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.2.6 <a
        href="#x1-173000D.2.6" id="QQ2-1-198">Exporters for a network block</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.2.7 <a
        href="#x1-174000D.2.7" id="QQ2-1-199">Exporters for a genomes block</a></span>
     <br/>&#x00A0;&#x00A0;<span class="subsectionToc">D.2.8 <a
        href="#x1-175000D.2.8" id="QQ2-1-200">Exporters for a view block</a></span>
     <br/>&#x00A0;<span class="sectionToc">D.3 <a
        href="#x1-176000D.3" id="QQ2-1-201">Taxon display labels import</a></span>
     <br/>&#x00A0;<span class="sectionToc">D.4 <a
        href="#x1-177000D.4" id="QQ2-1-202">Traits import</a></span>
     <br/><span class="appendixToc">E <a
        href="#x1-178000E" id="QQ2-1-203">Workflow</a></span>
     <br/>&#x00A0;<span class="sectionToc">E.1 <a
        href="#x1-179000E.1" id="QQ2-1-204">Input and working nodes</a></span>


     <br/>&#x00A0;<span class="sectionToc">E.2 <a
        href="#x1-180000E.2" id="QQ2-1-206">Data and algorithm nodes</a></span>
     <br/>&#x00A0;<span class="sectionToc">E.3 <a
        href="#x1-181000E.3" id="QQ2-1-210">Exporting the workflow</a></span>
     <br/>&#x00A0;<span class="sectionToc">E.4 <a
        href="#x1-182000E.4" id="QQ2-1-211">Running a workflow on multiple datasets</a></span>
     <br/><span class="appendixToc">F <a
        href="#x1-183000F" id="QQ2-1-212">Styling labels</a></span>
</div>
<!--l. 52--><p class="noindent">


<h3 class="likesectionHead"><a
        id="x1-2000"></a>Introduction</h3>
<!--l. 58--><p class="noindent">The SplitsTree App is new software for exploring and analyzing phylogenetic data, with
     an emphasis on phylogenetic
     networks. Offering a comprehensive set of features, the software provides over 100 algorithms for computing
     distances,
     phylogenetic trees, split networks, haplotype networks, rooted phylogenetic networks, tanglegrams, consensus trees
     and
     consensus networks.
     <!--l. 64-->
<p class="noindent">This new software [<a
        href="#XHusonBryant2024">Huson and Bryant</a>,&#x00A0;<a
        href="#XHusonBryant2024">2024</a>] is designed to accommodate the increasing scale and intricacy of modern
     data sets. It extends, integrates and supersedes our earlier applications SplitsTree4 [<a
             href="#XHusonBryant2006">Huson and Bryant</a>,&#x00A0;<a
             href="#XHusonBryant2006">2006</a>] for unrooted
     phylogenetic trees and networks, Dendroscope3 [<a
             href="#XHusonScornavacca2012">D.&#x00A0;H.&#x00A0;Huson</a>,&#x00A0;<a
             href="#XHusonScornavacca2012">2012</a>] for rooted trees and networks, and PopArt [<a
             href="#XLeighBryant2015">Leigh
          and Bryant</a>,&#x00A0;<a
             href="#XLeighBryant2015">2015</a>] for haplotype analysis.
     <!--l. 67-->
<p class="noindent">If you use this program, the please cite:
     <!--l. 69-->
<p class="noindent">Daniel H. Huson and David Bryant. The SplitsTree App: interactive analysis and visualization using
     phylogenetic trees and
     networks. Nature Methods (2024) <a
             href="https://doi.org/10.1038/s41592-024-02406-3" class="url"><span
             class="cmtt-10">https://doi.org/10.1038/s41592-024-02406-3</span></a>.
     <!--l. 72-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-20011"></a>


     <!--l. 73--><p class="noindent"><img
        src="figs/opening-example.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1: </span><span
             class="content">Example of SplitsTree analysis of primate mtDNA.</span></div>
     <!--tex4ht:label?: x1-20011 -->


     <!--l. 77--><p class="noindent"></div>
<hr class="endfigure">


<!--l. 2--><p class="noindent">


<h2 class="chapterHead"><span class="titlemark">Chapter&#x00A0;1</span><br/><a
        id="x1-30001"></a>Using SplitsTree</h2>
<a
        id="x1-3001"></a>
<!--l. 7--><p class="noindent">In this chapter we give an overview of how the interface of the SplitsTree app is
     organised. Later we provide more details
     on the actual methods and procedures.
<h3 class="sectionHead"><span class="titlemark">1.1   </span> <a
        id="x1-40001.1"></a>Getting started</h3>
<!--l. 11--><p class="noindent">To get started using this program, download the latest installer from <a
        href="https://software-ab.cs.uni-tuebingen.de/download/splitstree6" class="url"><span
        class="cmtt-10">https://software-ab.cs.uni-tuebingen.de/download/splitstree6</span></a>
     for Linux, Mac&#x00A0;OS&#x00A0;X or Windows, and install the program on your computer. Versions for iOS and
     Android are being
     tested.
     <!--l. 16-->
<p class="noindent">Launch the program by double-clicking the program icon or launch it from the command line (Linux).
     <!--l. 19-->
<p class="noindent">Use the <span class="obeylines-h"><span class="verb"><span
        class="cmtt-10">File-&#x003E;Open...</span></span></span> menu item to open a file containing data in one of the
     supported formats (see
     Chapter&#x00A0;<a
             href="#x1-159000D">D<!--tex4ht:ref: chapter:formats --></a>).
     <!--l. 22-->
<p class="noindent">If the data you provide is set of characters (or multiple sequence alignment), then by default, the
     SplitsTree app will
     compute P-distances (see Section&#x00A0;<a
             href="#x1-76000C.1">C.1<!--tex4ht:ref: alg:PDistance --></a>) and then will run Neighbor Net (see Section&#x00A0;<a
             href="#x1-106000C.2">C.2<!--tex4ht:ref: alg:NeighborNet --></a>) to obtain a split network (see
     Section&#x00A0;<a
             href="#x1-130001.10">1.10<!--tex4ht:ref: sec:split-network-tab --></a>). If you provide a distance matrix,
     then this will also result in a split network being displayed. If you supply
     trees, then the first tree will be displayed (see Section&#x00A0;<a
             href="#x1-90001.6">1.6<!--tex4ht:ref: sec:tree-view-tab --></a>).
     <!--l. 29-->
<p class="noindent">Here is a toy example of characters data. You can copy this text from the manual and then paste it
     onto the import data
     button (see Section&#x00A0;<a
             href="#x1-60001.3">1.3<!--tex4ht:ref: sec:tool-bar --></a>) to obtain the network show in the figure (see
     Fig.&#x00A0;<a
             href="#x1-40011">1.1<!--tex4ht:ref: fig:toy-network --></a>).


<div class="verbatim" id="verbatim-1">
     6&#x00A0;64
     &#x00A0;<br/>Taxon1&#x00A0;TAAGTAGATCGGAGTTTTTACTCGTGTGATTTTGGGTATTTTTTATTTAGATTATGAAATTATA
     &#x00A0;<br/>Taxon2&#x00A0;CTTAATATATAATGATATTACTTAAACATTATTAAATGATACACTAACTATAATTATTGAACAT
     &#x00A0;<br/>Taxon3&#x00A0;AAAATTATATAATATAACATTATATTCATTACCACAAGATTATATTATAAAATATATTGTACAC
     &#x00A0;<br/>Taxon4&#x00A0;TTCAATATATAATGAAACTTTATAAATAACTTTAGAAATCTTATAAAAAAATATCGACGAACAA
     &#x00A0;<br/>Taxon5&#x00A0;TAAGTAGATCTGAGTTTTTACTCGTGTGATTTTGGGTATTTTTTATTTAGATTATGAAATTATA
     &#x00A0;<br/>Taxon6&#x00A0;TAAATTGGATAATATTTTATTATGTGTTACTACAGAAATCATTAATTATATAAACGATATACAA
</div>
<!--l. 42--><p class="nopar">
     <!--l. 44-->
<p class="noindent">


     <!--l. 46-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-40011"></a>


     <!--l. 47--><p class="noindent"><img
        src="figs/toy-characters.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.1: </span><span
             class="content">Neighbor net computed for toy characters data.</span></div><!--tex4ht:label?: x1-40011 -->


     <!--l. 52--><p class="noindent"></div>
<hr class="endfigure">
<h3 class="sectionHead"><span class="titlemark">1.2   </span> <a
        id="x1-50001.2"></a>Layout of the main window</h3>
<!--l. 57--><p class="noindent">In the SplitsTree app, you can open one or more documents and each one has its own main
     window. Different analyses of
     the same data in the same document are shown in different tabs in that window (see Fig.&#x00A0;<a
             href="#x1-50012">1.2<!--tex4ht:ref: fig:main-window --></a>). Tabs can be laid out
     side-by-side.
     <!--l. 60-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-50012"></a>


     <!--l. 61--><p class="noindent"><img
        src="figs/window.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.2: </span><span
             class="content">The main window. (a) Text and graphic output are presented in the main tabs on the right-hand side
of the window. (b) There is a side-bar on the left hand side that shows (c) the workflow at the top and (d) tabs
for setting algorithm parameters at the bottom. At the top left (e) there are some document-specific tool-bar
items whereas the items at the top right (f) apply to the current main tab. While all program features can be
directly accessed from the main window, the menu bar (g) provides alternative access to many of the features. <a
                  id="x1-5002"></a></span></div><!--tex4ht:label?: x1-50012 -->


     <!--l. 67--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 69--><p class="noindent">Each document opened in SplitsTree has its own main window. The main window has the
     following parts:
<ul class="itemize1">
     <li class="itemize">A menu bar providing menu items to access features of the program. Note that all features or
          the program
          can also be accessed from within the main window as well (see Fig.&#x00A0;<a
                  href="#x1-50012">1.2<!--tex4ht:ref: fig:main-window --></a>) (figure part g).
     </li>
     <li class="itemize">A toolbar containing both document- and tab-specific items (figure parts e,f).
     </li>
     <li class="itemize">A main tab pane that contains all the text and data tabs (figure part a).
     </li>
     <li class="itemize">A sidebar providing access to the workflow (figure parts b,c).
     </li>
     <li class="itemize">An algorithms tab pane (inside the sidebar) for parameterizing and running algorithms (figure
          parts b,d).
     </li>
</ul>
<h3 class="sectionHead"><span class="titlemark">1.3   </span> <a
        id="x1-60001.3"></a>Main toolbar items</h3>
<!--l. 88--><p class="noindent">There are three <span
        class="cmssi-10">document-specific toolbar items</span><a
        id="x1-6001"></a> on the left side of the main toolbar and several <span
        class="cmssi-10">tab-specific toolbar items</span><a
        id="x1-6002"></a>
     ones on the right (see Fig.&#x00A0;<a
             href="#x1-50012">1.2<!--tex4ht:ref: fig:main-window --></a>) (see Fig.&#x00A0;<a
             href="#x1-60033">1.3<!--tex4ht:ref: fig:toolbar --></a>).
     <!--l. 93-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-60033"></a>


     <!--l. 94--><p class="noindent"><img
        src="figs/toolbar.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.3: </span><span
             class="content">The main toolbar has three document-specific items on the left: Files menu button, import data
button and sidebar toggle button. It has several tab-specific ones on the right: Undo and redo, font resizing, select
all/none, find/replace and export.</span></div><!--tex4ht:label?: x1-60033 -->


     <!--l. 99--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 101--><p class="noindent">Document specific toolbar items:
<ul class="itemize1">
     <li class="itemize">The <span
             class="cmssi-10">files menu button</span><a
             id="x1-6004"></a> - provides a menu that contains file-related items from the main files menu and a
          list of recently opened documents.
     </li>
     <li class="itemize">The <span
             class="cmssi-10">import data button</span><a
             id="x1-6005"></a> - use this open any string or file from the system clipboard, or via drag-and-drop,
          into a new SplitsTree document.
     </li>
     <li class="itemize">The <span
             class="cmssi-10">sidebar toggle button</span><a
             id="x1-6006"></a> - use this to show or hide the sidebar.
     </li>
</ul>
<!--l. 109--><p class="noindent">Tab specific toolbar items (that apply only to the currently selected main tab):
<ul class="itemize1">
     <li class="itemize">An <span
             class="cmssi-10">undo button</span><a
             id="x1-6007"></a> and <span
             class="cmssi-10">redo button</span><a
             id="x1-6008"></a>.
     </li>
     <li class="itemize">An <span
             class="cmssi-10">increase font size button</span><a
             id="x1-6009"></a> and <span
             class="cmssi-10">increase font size button</span><a
             id="x1-6010"></a>.
     </li>
     <li class="itemize">A <span
             class="cmssi-10">selection button</span><a
             id="x1-6011"></a> to select all or none.
     </li>
     <li class="itemize">A <span
             class="cmssi-10">find button</span><a
             id="x1-6012"></a> that you press to open the find dialog, press again to open the replace dialog (if
          available)
          and once again to close the dialog.
     </li>
     <li class="itemize">An <span
             class="cmssi-10">export button</span><a
             id="x1-6013"></a> that gives access to items for copying, exporting or printing the data or image
          associated
          with the current tab. An additional item is provided for showing or hiding the <span
                  class="cmssi-10">QR code</span><a
                  id="x1-6014"></a> associated with a
          given tree or network.
     </li>
</ul>
<h3 class="sectionHead"><span class="titlemark">1.4   </span> <a
        id="x1-70001.4"></a>The main tabs</h3>
<!--l. 125--><p class="noindent">The main window uses a tabbed pane to present all text and visualizations (see Fig.&#x00A0;<a
        href="#x1-50012">1.2<!--tex4ht:ref: fig:main-window --></a>). Here is an overview of the
     supported tabs:
<ul class="itemize1">
     <li class="itemize">Alignment tab - provides a visualization of the input multiple sequence alignment.
     </li>
     <li class="itemize">Tree-View tab - shows a phylogenetic tree or rooted network.
     </li>
     <li class="itemize">Tree-Pages tab - shows pages of phylogenetic trees or rooted networks.
     </li>
     <li class="itemize">Tanglegram tab - shows a tanglegram of trees or rooted networks [<a
             href="#XScornavaccaetal2011">Scornavacca et&#x00A0;al.</a>,&#x00A0;<a
             href="#XScornavaccaetal2011">2011</a>].
     </li>
     <li class="itemize">Densi-Tree tab - shows a densi-tree visualization of a profile of trees [<a
             href="#XBouckaert2010">Bouckaert</a>,&#x00A0;<a
             href="#XBouckaert2010">2010</a>].


     </li>
     <li class="itemize">Split-Network tab - show a split network [<a
             href="#XDressHuson2004">Dress and Huson</a>,&#x00A0;<a
             href="#XDressHuson2004">2004</a>] or phylogenetic outline [<a
             href="#XBagcietal2021">Bagci
          et&#x00A0;al.</a>,&#x00A0;<a
             href="#XBagcietal2021">2021</a>].
     </li>
     <li class="itemize">Network tab - shows a network such as a haplotype network [<a
             href="#XBandeltetal1999">Bandelt et&#x00A0;al.</a>,&#x00A0;<a
             href="#XBandeltetal1999">1999</a>].
     </li>
     <li class="itemize">World map tab - shows a map of the world and displays traits data that have associated and
          latitude and
          longitude coordinates.
     </li>
     <li class="itemize">Workflow tab - provides access to the workflow graph<a
             id="x1-7001"></a>.
     </li>
     <li class="itemize">How to cite tab - provides a description of the data and algorithms used, and provides the
          necessary citations.
     </li>
     <li class="itemize">Input editor tab - provides an interactive editor for entering and parsing input data.
     </li>
     <li class="itemize">Report tabs - these are used to present the results of analyses such as phylogenetic diversity.
     </li>
     <li class="itemize">Text tabs - any of the data blocks can be displayed in such a text tab, they provide several
          different formats.
     </li>
</ul>
<!--l. 149--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">1.5   </span> <a
        id="x1-80001.5"></a>Alignment tab</h3>
<!--l. 152--><p class="noindent">The <span
        class="cmssi-10">Alignment tab</span><a
        id="x1-8001"></a> provides a visualization of the input characters or multiple sequence alignment (see
     Fig.&#x00A0;<a
             href="#x1-80024">1.4<!--tex4ht:ref: fig:alignment-tab --></a>).
     <!--l. 155-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-80024"></a>


     <!--l. 156--><p class="noindent"><img
        src="figs/alignment-tab.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.4: </span><span
             class="content">The alignment tab for displaying and working with a multiple sequence alignment.</span>
     </div><!--tex4ht:label?: x1-80024 -->


     <!--l. 158--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 160--><p class="noindent">The alignment tab has a drop-down menu button at the left that contains items for
     selecting sites. There is a button for
     selecting a color scheme. There is a button for toggling between a close-up view and a total view of the alignment,
     and
     buttons for zooming in and out both vertically and horizontally.
<h3 class="sectionHead"><span class="titlemark">1.6   </span> <a
        id="x1-90001.6"></a>Tree-View tab</h3>
<!--l. 171--><p class="noindent">The <span
        class="cmssi-10">Tree-View tab</span><a
        id="x1-9001"></a> shows a phylogenetic tree or a rooted network (see Fig.&#x00A0;<a
        href="#x1-90025">1.5<!--tex4ht:ref: fig:treeview-tab --></a>).
     <!--l. 174-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-90025"></a>


     <!--l. 175--><p class="noindent"><img
        src="figs/treeview-tab.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.5: </span><span
             class="content">The tree-view tab for drawing a phylogenetic tree or rooted network.</span></div>
     <!--tex4ht:label?: x1-90025 -->


     <!--l. 177--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 179--><p class="noindent">The tree view tab has a toolbar and side panel that are hidden by default, but can be
     opened using two toggle buttons at
     the top right of the tab.
     <!--l. 183-->
<p class="noindent">The toolbar provides items for selecting how to draw the tree (or rooted network), the choices are
     between rectangular,
     circular and radial cladogram or phylogram. There is a button that toggles the scale bar and addition information
     (such as
     name of the tree, if any, and number of nodes, edges and leaves). In addition, there are buttons for rotating,
     flipping and
     zooming.
     <!--l. 191-->
<p class="noindent">The side panel (on the right) contains items for styling the taxon labels and for adding marks to
     the taxa. These are
     colored shapes that appear next to the taxon label. There are items for displaying taxon traits, if present.
     Moreover, there
     are items for setting the line width of edges and to choose whether to label edges by associated weights or
     confidence
     values.
<h3 class="sectionHead"><span class="titlemark">1.7   </span> <a
        id="x1-100001.7"></a>Tree-Pages tab</h3>
<!--l. 198--><p class="noindent">The <span
        class="cmssi-10">Tree-Pages tab</span><a
        id="x1-10001"></a> shows pages of phylogenetic trees (see Fig.&#x00A0;<a
        href="#x1-100026">1.6<!--tex4ht:ref: fig:treepages-tab --></a>).
     <!--l. 200-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-100026"></a>


     <!--l. 201--><p class="noindent"><img
        src="figs/treepages-tab.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.6: </span><span
             class="content">The tree-pages tab for displaying a collection of phylogenetic trees or rooted networks.</span>
     </div><!--tex4ht:label?: x1-100026 -->


     <!--l. 203--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 205--><p class="noindent">The tree-pages tab has a toolbar and side panel that are hidden by default, but can be
     opened using two toggle buttons
     at the top right of the tab.
     <!--l. 209-->
<p class="noindent">The toolbar provides items for selecting how to draw the tree (or rooted network), the choices are
     between rectangular,
     circular and radial cladogram or phylogram. There is a button that toggles addition information (such as name of
     the tree,
     if any, and number of nodes, edges and leaves). In addition, there are buttons for rotating, flipping and zooming.
     At the
     right side of the toolbar, there is a text input field for setting the dimensions of a page in the format <span
             class="cmssi-10">rows x</span>
     <span
             class="cmssi-10">cols</span>.
     <!--l. 220-->
<p class="noindent">The side panel contains items for styling the taxon labels and for adding marks to the taxa.
     <!--l. 222-->
<p class="noindent">At the bottom of the tree-pages tab there is a row of buttons that can be used to navigate through
     the
     pages.
<h3 class="sectionHead"><span class="titlemark">1.8   </span> <a
        id="x1-110001.8"></a>Tanglegram tab</h3>
<!--l. 232--><p class="noindent">The <span
        class="cmssi-10">Tanglegram tab</span><a
        id="x1-11001"></a> shows a tanglegram of trees or rooted networks, which is calculated based on the objective of
     displacement optimization [<a
             href="#XHuson2025Tanglegrams">Huson</a>,&#x00A0;<a
             href="#XHuson2025Tanglegrams">2025</a>] (see Fig.&#x00A0;<a
             href="#x1-110027">1.7<!--tex4ht:ref: fig:tanglegram-tab --></a>).
     <!--l. 235-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-110027"></a>


     <!--l. 236--><p class="noindent"><img
        src="figs/tanglegram-tab.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.7: </span><span
             class="content">The tanglegram tab for comparing two phylogenetic trees or rooted networks.</span></div>
     <!--tex4ht:label?: x1-110027 -->


     <!--l. 238--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 240--><p class="noindent">The tanglegram tab has a toolbar and side panel that are hidden by default, but can be
     opened using two toggle buttons
     at the top right of the tab.
     <!--l. 244-->
<p class="noindent">The toolbar provides items to determine the first and second trees (from the same file) and for
     selecting how to draw
     either tree (or rooted network), the choices are between rectangular phylogram, rectangular cladogram
     and triangular cladogram (trees only). There is a button that toggles addition information (such as name
     of the tree, if any, and number of nodes, edges and leaves). There are buttons for rotating, flipping and
     zooming.
     <!--l. 253-->
<p class="noindent">There is a menu that allows one to select precisely what the target of displacement optimization
     should be, namely taxa
     or reticulate edges in which of the two phylogenies. In addition, if one of the two input phylogenies has a very
     unresolved
     node, that using the PQ-tree cluster heuristic may help to untangle that situation (but only when both sides are
     being
     untangled.)
     <!--l. 258-->
<p class="noindent">The side panel contains items for styling the taxon labels and for adding marks to the taxa.


<h3 class="sectionHead"><span class="titlemark">1.9   </span> <a
        id="x1-120001.9"></a>DensiTree tab</h3>
<!--l. 267--><p class="noindent">The <span
        class="cmssi-10">Densi-Tree tab</span><a
        id="x1-12001"></a> shows a densi-tree visualization of a Bayesian profile of trees [<a
        href="#XBouckaert2010">Bouckaert</a>,&#x00A0;<a
        href="#XBouckaert2010">2010</a>] (see
     Fig.&#x00A0;<a
             href="#x1-120028">1.8<!--tex4ht:ref: fig:densitree-tab --></a>).
     <!--l. 269-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-120028"></a>


     <!--l. 270--><p class="noindent"><img
        src="figs/densitree-tab.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.8: </span><span
             class="content">The densi-tree tab for displaying a Bayesian profile of phylogenetic trees.</span></div>
     <!--tex4ht:label?: x1-120028 -->


     <!--l. 272--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 274--><p class="noindent">The densi-tree tab has a toolbar and side panel that are hidden by default, but can be
     opened using two toggle buttons at
     the top right of the tab.
     <!--l. 278-->
<p class="noindent">The toolbar provides items for determining how to draw the trees, choices are between rectangular,
     triangular, rounded
     and radial phylogram. (The rounded phylograms are time-consuming to draw and can cause problems.) In addition,
     there
     are buttons for rotating, flipping and zooming.
     <!--l. 283-->
<p class="noindent">The side panel contains items for styling the taxon labels and for adding marks to the taxa. The
     line width can be set
     here. In addition, two colors can be set. The first is used for edges that in the tree profile that are compatible
     with the
     displayed greedy consensus tree (default color is black), and the second is used for incompatible edges (default
     color is
     red).
<h3 class="sectionHead"><span class="titlemark">1.10   </span> <a
        id="x1-130001.10"></a>Split-Network tab</h3>
<!--l. 296--><p class="noindent">The <span
        class="cmssi-10">Split-Network tab</span><a
        id="x1-13001"></a> shows a split network [<a
        href="#XDressHuson2004">Dress and Huson</a>,&#x00A0;<a
        href="#XDressHuson2004">2004</a>] or phylogenetic outline [<a
        href="#XBagcietal2021">Bagci et&#x00A0;al.</a>,&#x00A0;<a
        href="#XBagcietal2021">2021</a>] (see
     Fig.&#x00A0;<a
             href="#x1-130029">1.9<!--tex4ht:ref: fig:splits-tab --></a>).
     <!--l. 299-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-130029"></a>


     <!--l. 300--><p class="noindent"><img
        src="figs/splits-tab.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.9: </span><span
             class="content">The split-network tab for displaying a collection of splits as a split network or phylogenetic outline.
</span></div><!--tex4ht:label?: x1-130029 -->


     <!--l. 305--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 307--><p class="noindent">To reshape the layout of the network by rotating the edges associated with one or more
     selected splits, press-and-drag on
     the network.
     <!--l. 310-->
<p class="noindent">The split-network tab has a toolbar and side panel that are hidden by default, but can be opened
     using two toggle buttons
     at the top right of the tab.
     <!--l. 314-->
<p class="noindent">The toolbar provides a choice box to determine how to draw the splits. The choices are as a split
     network, to-scale or as a
     topology (with all edges of uniform length), or as a phylogenetic outline, again, either to-scale or as a topology
     (with all
     edges of uniform length). There is a second choice box to determine whether the network is to be drawn unrooted<a
             id="x1-13003"></a><a
             id="x1-13004"></a> or
     rooted<a
             id="x1-13005"></a><a
             id="x1-13006"></a>, using either <span
             class="cmssi-10">mid-point rooting</span><a
             id="x1-13007"></a> or <span
             class="cmssi-10">outgroup rooting</span><a
             id="x1-13008"></a>. The latter requires that some taxa have been selected; these
     are treated as the outgroup.
     <!--l. 324-->
<p class="noindent">In addition, there are buttons for rotating, flipping, zooming and for setting the <span
        class="cmssi-10">scale ratio</span> <a
        id="x1-13009"></a>to a specific value to
     ensure that different networks are drawn to the same scale. Use the rotate buttons to rotate the entire
     network.
     <!--l. 329-->
<p class="noindent">The right side panel contains items for styling the taxon labels and for adding marks to the taxa.
     The line width can be
     set here. In addition, the line width and color can be set. The color of the inner area of an outline can be set.
     You can
     request to have the splits labeled by their weight, their confidence values (if available) or their internal split
     ids.
<h3 class="sectionHead"><span class="titlemark">1.11   </span> <a
        id="x1-140001.11"></a>Network tab</h3>
<!--l. 340--><p class="noindent">The <span
        class="cmssi-10">Network tab</span><a
        id="x1-14001"></a> shows a network (see Fig.&#x00A0;<a
        href="#x1-1400210">1.10<!--tex4ht:ref: fig:network-tab --></a>). The network tab has a toolbar and side panel
     that are hidden by
     default, but can be opened using two toggle buttons at the top right of the tab.
     <!--l. 346-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-1400210"></a>


     <!--l. 347--><p class="noindent"><img
        src="figs/network-tab.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.10: </span><span
             class="content">The network tab for displaying a haplotype networks and related on constructs. </span>
     </div><!--tex4ht:label?: x1-1400210 -->


     <!--l. 351--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 354--><p class="noindent">The toolbar provides a choice box to determine how to draw the network. In addition,
     there are buttons for rotating,
     flipping and zooming.
     <!--l. 358-->
<p class="noindent">The side panel contains items for styling the taxon labels and for adding marks to the taxa. The
     line width can be set
     here. In addition, there are items to determine which traits are to be shown in pie charts and whether to a
     legend. To change the colors used in pie charts, press on the items in the legend. These are then stored in
     the traits block in the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">TRAITCOLOR</span></span></span> entry. Also, there is a menu button for determining how to
     represent
     character-state changes along an edge. The choices are has hatches (short marks), labels, compact labels and
     counts.
<h3 class="sectionHead"><span class="titlemark">1.12   </span> <a
        id="x1-150001.12"></a>World map tab</h3>
<!--l. 372--><p class="noindent">The <span
        class="cmssi-10">World Map tab</span><a
        id="x1-15001"></a> shows a map of the world and places any traits data that comes with latitude and longitude
     assignments on the map (see Fig.&#x00A0;<a
             href="#x1-1500211">1.11<!--tex4ht:ref: fig:world-map-tab --></a>). To change the colors used in pie charts,
     press on the items in the
     legend. They are stored in the traits block in the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">TAXONCOLOR</span></span></span> entry. The world tab has a toolbar and side
     panel, the latter is hidden by default, but can be opened using the toggle button at the top right of the
     tab.
     <!--l. 382-->
<p class="noindent">This tab appears when the input data contains a traits block that has latitude and longitude
     specifications (see
     Section&#x00A0;<a
             href="#x1-62000B.2">B.2<!--tex4ht:ref: sec:traits-block --></a>).
     <!--l. 385-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-1500211"></a>


     <!--l. 386--><p class="noindent"><img
        src="figs/world-map.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.11: </span><span
             class="content">The world map tab for displaying haplotype locations of origin. </span></div>
     <!--tex4ht:label?: x1-1500211 -->


     <!--l. 390--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 393--><p class="noindent">There is a <span class="obeylines-h"><span class="verb"><span
        class="cmtt-10">Show</span></span></span> menu button to determine whether country names, continent names and/or
     oceans should appear as
     labels. There is a button to determine whether to show two copies of the map side-by-side for Pacific-centric data.
     There
     is a button to zoom to the shown haplotype data.
<h3 class="sectionHead"><span class="titlemark">1.13   </span> <a
        id="x1-160001.13"></a>Workflow graph tab</h3>
<!--l. 406--><p class="noindent">The <span
        class="cmssi-10">workflow tab</span><a
        id="x1-16001"></a> provides access to the <span
        class="cmssi-10">workflow graph</span><a
        id="x1-16002"></a> (see Fig.&#x00A0;<a
        href="#x1-1600312">1.12<!--tex4ht:ref: fig:workflow-tab --></a>).
     <!--l. 410-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-1600312"></a>


     <!--l. 411--><p class="noindent"><img
        src="figs/workflow-tab.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.12: </span><span
             class="content">The workflow tab provides access to the workflow tab, for advanced users. </span></div>
     <!--tex4ht:label?: x1-1600312 -->


     <!--l. 415--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 417--><p class="noindent">The workflow tab has a toolbar that contains a number of items, whose purpose and
     enabled state depends on which
     nodes in the workflow graph are currently selected.
     <!--l. 422-->
<p class="noindent">The first toolbar item will open the corresponding algorithm, text display or view tab, depending on
     whether
     the selected node is an algorithm node, data node or view node. Double-clicking on a node has the same
     effect.
     <!--l. 427-->
<p class="noindent">When a data node is selected, then the second toolbar item can be used to attach an additional
     algorithm to the data
     node.
     <!--l. 430-->
<p class="noindent">When an algorithm node is selected, then the next two items can be used either to duplicate the
     analysis, or to delete it,
     respectively. Each algorithm node also carries a similar menu button.
     <!--l. 433-->
<p class="noindent">There are two items for zooming in and out.
<h3 class="sectionHead"><span class="titlemark">1.14   </span> <a
        id="x1-170001.14"></a>How to cite tab</h3>
<!--l. 441--><p class="noindent">The <span
        class="cmssi-10">How to cite tab</span><a
        id="x1-17001"></a> provides a description of the data and algorithms used, and provides the necessary citations
     (see
     Fig.&#x00A0;<a
             href="#x1-1700213">1.13<!--tex4ht:ref: fig:cite-tab --></a>).
     <!--l. 443-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-1700213"></a>


     <!--l. 444--><p class="noindent"><img
        src="figs/cite-tab.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.13:  </span><span
             class="content">The  how-to-cite  tab  provides  methods  summary  of  the  data  and  algorithms,  and  provides  all
suggested references for the methods used. </span></div><!--tex4ht:label?: x1-1700213 -->


     <!--l. 448--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 450--><p class="noindent">The toolbar of the input tab contains a button to copy the complete or selected content
     of the tab. There are buttons to
     turn line-wrapping and lines numbers on and off.
     <!--l. 454-->
<p class="noindent">If the input is a Nexus or SplitsTree file that contains a comment at the beginning of the file
     describing the source of the
     data, then this will be reported at the top of the text area. This is followed by a description of the methods
     used. Finally,
     all suggested references are listed.
<h3 class="sectionHead"><span class="titlemark">1.15   </span> <a
        id="x1-180001.15"></a>Input editor tab</h3>
<!--l. 463--><p class="noindent">The <span
        class="cmssi-10">Input editor tab</span><a
        id="x1-18001"></a> provides an interactive editor for entering and parsing input data (see Fig.&#x00A0;<a
        href="#x1-1800214">1.14<!--tex4ht:ref: fig:editor-tab --></a>).
     <!--l. 465-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-1800214"></a>


     <!--l. 466--><p class="noindent"><img
        src="figs/editor-tab.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.14: </span><span
             class="content">The editor tab is used to enter data into a new document. Pressing the run button will parse the
data and launch an analysis of the entered data. </span></div><!--tex4ht:label?: x1-1800214 -->


     <!--l. 469--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 471--><p class="noindent">The toolbar of the input tab contains a button to copy the complete or selected content
     of the tab. There buttons to turn
     line-wrapping and lines numbers on and off.
     <!--l. 475-->
<p class="noindent">The program will try to guess to which input format the entered text adheres to and will indicate
     the name of the format
     in the toolbar. When a valid format has been detected, then the run button will be enabled. Pressing the run button
     will
     parse the data and launch an analysis of the entered data.
     <!--l. 478-->
<p class="noindent">The input editor can be opened from the File menu and is automatically open when the user imports a
     text or file into the
     program that is not in one of the recognized input formats.
<h3 class="sectionHead"><span class="titlemark">1.16   </span> <a
        id="x1-190001.16"></a>Report tabs</h3>
<!--l. 489--><p class="noindent"><span
        class="cmssi-10">Report tabs</span><a
        id="x1-19001"></a> are used to present the results of analyses such as Tajima&#8217; D, phylogenetic diversity
     or Shapely values as a
     text (see Fig.&#x00A0;<a
             href="#x1-1900215">1.15<!--tex4ht:ref: fig:report-tab --></a>).
     <!--l. 493-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-1900215"></a>


     <!--l. 494--><p class="noindent"><img
        src="figs/report-tab.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.15: </span><span
             class="content">Report tabs are text tabs that are used to provide the result of an analysis, here the Shapely values
for a set of taxa based on splits. </span></div><!--tex4ht:label?: x1-1900215 -->


     <!--l. 499--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 501--><p class="noindent">The toolbar of the report tab contains a button to copy the complete or selected
     content of the tab. There are buttons to
     turn line-wrapping and lines numbers on and off.
<h3 class="sectionHead"><span class="titlemark">1.17   </span> <a
        id="x1-200001.17"></a>Text tabs</h3>
<!--l. 509--><p class="noindent"><span
        class="cmssi-10">Text tabs</span><a
        id="x1-20001"></a> are used to show the content of data blocks, in a choice of several different formats (see
     Fig.&#x00A0;<a
             href="#x1-2000216">1.16<!--tex4ht:ref: fig:text-tab --></a>).
     <!--l. 513-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-2000216"></a>


     <!--l. 514--><p class="noindent"><img
        src="figs/text-tab.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.16: </span><span
             class="content">Text tabs are used to display the content of any of the data nodes in the workflow. Here we show
three such tabs, one for characters data, one for distance data and one for splits data. </span></div>
     <!--tex4ht:label?: x1-2000216 -->


     <!--l. 518--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 520--><p class="noindent">The toolbar of any text tab contains a button to copy the complete or selected content
     of the tab. There are buttons to
     turn line-wrapping and lines numbers on and off. There is a format pane that can be used to select the desired
     display
     format and to specify any options associated with the format.
     <!--l. 525-->
<p class="noindent">Such a text tab can be opened by selecting a data node item in the sidebar and then pressing the
     show/edit button at the
     top of the sidebar, or by double-clicking on the item.
<h3 class="sectionHead"><span class="titlemark">1.18   </span> <a
        id="x1-210001.18"></a>The sidebar</h3>
<!--l. 537--><p class="noindent">The sidebar (see Fig.&#x00A0;<a
        href="#x1-50012">1.2<!--tex4ht:ref: fig:main-window --></a>) (figure part b) contains a representation of the
     workflow as a tree at the top, and the
     algorithms tab pane at the bottom (figure parts c,d).
     <!--l. 540-->
<p class="noindent">The <span
        class="cmssi-10">workflow tree view</span><a
        id="x1-21001"></a> contains a representation of all input data, computed data and algorithms used in the
     computation. There are three types of nodes:
<ul class="itemize1">
     <li class="itemize"><span
             class="cmssi-10">data nodes</span><a
             id="x1-21002"></a> that represent data blocks and result blocks,
     </li>
     <li class="itemize"><span
             class="cmssi-10">algorithm nodes</span><a
             id="x1-21003"></a> that represent algorithms, and
     </li>
     <li class="itemize"><span
             class="cmssi-10">view nodes</span><a
             id="x1-21004"></a> that represent visualizations.
     </li>
</ul>
<!--l. 548--><p class="noindent">Double-clicking on a data node will open a text tab displaying the corresponding data,
     or analysis result, if
     the data block is a report. Double-clicking on an algorithm node will open the corresponding algorithm
     tab.
     <!--l. 551-->
<p class="noindent">An algorithm tab has a run button (at the right) to execute the algorithm and may contain some
     optional input items
     (below) to set parameters of the algorithm.
     <!--l. 554-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">1.19   </span> <a
        id="x1-220001.19"></a>The draft genome dialog</h3>
<!--l. 557--><p class="noindent">The SplitsTree App supports the calculation of the <span
        class="cmssi-10">phylogenetic context </span>of a draft prokaryotic genome [<a
        href="#XBagcietal2021">Bagci
     et&#x00A0;al.</a>,&#x00A0;<a
        href="#XBagcietal2021">2021</a>].<a
        id="x1-22001"></a> One or more files (FastA format) each containing one or more sequences representing draft
     genomes (or
     metagenomic assembly bins) can be imported into the program and then compared against a set of GTDB reference
     genomes [<a
             href="#XParksetal2018">Parks et&#x00A0;al.</a>,&#x00A0;<a
             href="#XParksetal2018">2018</a>] <a
             id="x1-22002"></a>using mash distances [<a
             href="#XOndovetal2016">Ondov et&#x00A0;al.</a>,&#x00A0;<a
             href="#XOndovetal2016">2016</a>] <a
             id="x1-22003"></a>and then represented as a phylogenetic outline.
     <a
             id="x1-22004"></a>
     <!--l. 564-->
<p class="noindent">The dialog is opened using the <span class="obeylines-h"><span class="verb"><span
        class="cmtt-10">File-&#x003E;Analyze</span><span
        class="cmtt-10">&#x00A0;Draft</span><span
        class="cmtt-10">&#x00A0;Genomes...</span></span></span> menu item and is set up using three tabs, as shown
     in the Figure (see Fig.&#x00A0;<a
             href="#x1-2200517">1.17<!--tex4ht:ref: fig:draft-genomes-dialog --></a>).
     <!--l. 567-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-2200517"></a>


     <!--l. 568--><p class="noindent"><img
        src="figs/draft-genomes.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;1.17: </span><span
             class="content">The first tab is used to specify the input files, the type of input (DNA or protein sequences) and
whether to use files or FastA records as input genomes. Also, specify the output file (and whether to store input
sequences as sequences or as references to files). The second tab is used to edit the labels of genomes. The third
tab is used to specify the database to compare against (downloaded from the SplitsTree page), the distance to
search in, and distance within which to include references. </span></div><!--tex4ht:label?: x1-2200517 -->


     <!--l. 579--><p class="noindent"></div>
<hr class="endfigure">


<h2 class="chapterHead"><span class="titlemark">Chapter&#x00A0;2</span><br/><a
        id="x1-230002"></a>Building trees and networks</h2>
<h3 class="sectionHead"><span class="titlemark">2.1   </span> <a
        id="x1-240002.1"></a>Using the workflow</h3>
<!--l. 7--><p class="noindent">In SplitsTree4, data analysis was based on a simple linear sequence. To construct a
     Neighbor Net, for example, one might
     input character data, apply a transform to infer a distance matrix, apply another transform to produce
     the set of splits in the Neighbor Net and another transform for convert those splits into a network on the
     screen.
     <!--l. 9-->
<p class="noindent">That simplicity came with limitations. For example, to compare the result of analyses using
     different parameters or
     distance methods it was necessary to duplicate the whole file and start again.
     <!--l. 11-->
<p class="noindent">The SplitsTree App implements a far more sophisticated system for workflows. It is still
     straightforward to run a simple
     linear workflow as in SplitsTree4, but it is now possible to branch that workflow at any point, exploring
     alternative parameters or methods. The use of frames make it easy to view the results of different analyses
     side-by-side.
     <!--l. 13-->
<p class="noindent">The branching structure of a document&#8217;s workflow can be viewed in the side panel (as a
     hierarchy) or in the workflow
     panel (as a graph). To illustrate, open the example file <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">ungulates.nex</span></span></span> which can be found in the directory
     <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">publications/WelkerEtal2015</span></span></span> in the Examples directory. By default, the
     SplitsTree App creates a network by running
     Neighbor Net and using the <span
             class="cmmi-10">p</span>-distance. Switching to the workflow panel displays the (linear) workflow for this
     initial
     analysis (see Fig.&#x00A0;<a
             href="#x1-240011">2.1<!--tex4ht:ref: fig:ungulate1 --></a>).
     <!--l. 15-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-240011"></a>


     <!--l. 17--><p class="noindent"><img
        src="figs/Ungulate1.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;2.1: </span><span
             class="content">Right hand side of the workflow created when <span
             class="cmtt-10">ungulates.nex </span>is opened.</span></div><!--tex4ht:label?: x1-240011 -->


     <!--l. 20--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 23--><p class="noindent">In this graph, nodes correspond to algorithms (indicated by a <img
        src="figs/algorithm-icon.png" alt="PIC"
> icon) or data (indicated by an <img
        src="figs/data-icon.png" alt="PIC"
> icon). From the
     algorithm nodes you can edit the parameters of the method. Selecting an algorithm node and clicking the delete
     button
     (top of pane) removes that node and any descendants of that node.
     <!--l. 25-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-240022"></a>


     <!--l. 26--><p class="noindent"><img
        src="figs/Ungulate2.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;2.2: </span><span
             class="content">Attaching a BioNJ algorithm to an existing Distances block.</span></div>
     <!--tex4ht:label?: x1-240022 -->


     <!--l. 28--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 30--><p class="noindent">Suppose we want to compare a network computed by the Neighbor Net algorithm with a tree
     obtained using
     BioNJ. Assuming both are to be computed from the same distance matrix, we can select the corresponding
     node and choose BioNJ from the popup menu marked with a plus (either on the node or in the toolbar)
     (see Fig.&#x00A0;<a
             href="#x1-240022">2.2<!--tex4ht:ref: fig:attach-bionj --></a>). SplitsTree then constructs and displays the
     BioNJ tree. Switching back to the workflow
     panel we see that a new sequence of nodes has branched off the distances node, indicating the revised
     analysis.
     <!--l. 36-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-240033"></a>


     <!--l. 38--><p class="noindent"><img
        src="figs/Ungulate3.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;2.3: </span><span
             class="content">Right hand side of the workflow after adding a BioNJ analysis.</span></div>
     <!--tex4ht:label?: x1-240033 -->


     <!--l. 41--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 43--><p class="noindent">This analysis also creates a new window tab. Right-click on a tab to get a context menu
     that allows you to split the main
     tab pane into two parts, then drag the tabs to the left or right panes to view both the Neighbor Net network and
     the
     BioNJ tree side-by-side.
<h3 class="sectionHead"><span class="titlemark">2.2   </span> <a
        id="x1-250002.2"></a>Building trees</h3>
<!--l. 51--><p class="noindent">SplitsTree implements four standard tree construction methods:
<ul class="itemize1">
     <li class="itemize">NJ (Neighbor-Joining), the original method of Saitou and Nei [<a
             href="#XSaitouNei1987">Saitou and Nei</a>,&#x00A0;<a
             href="#XSaitouNei1987">1987</a>].
     </li>
     <li class="itemize">BioNJ, the modification of NJ introduced by Gascuel to reduce variance of the node-to-node
          estimates
          [<a
                  href="#XGascuel1997">Gascuel</a>,&#x00A0;<a
                  href="#XGascuel1997">1997</a>].
     </li>
     <li class="itemize">UPGMA, the agglomerative method for constructing ultrametric (molecular clock) trees,
          introduced by
          Sokal and Sneath [<a
                  href="#XSokalMichener1958">Sokal and Michener</a>,&#x00A0;<a
                  href="#XSokalMichener1958">1958</a>].
     </li>
     <li class="itemize">Buneman, a method for inferring compatible splits (and therefore trees) from distances which
          tends to
          produce trees with large multifurcations [<a
                  href="#XBandeltDress1992">Bandelt and Dress</a>,&#x00A0;<a
                  href="#XBandeltDress1992">1992</a>].
     </li>
</ul>
<!--l. 60--><p class="noindent">Each of these can be called from the <span class="obeylines-h"><span class="verb"><span
        class="cmtt-10">Trees</span></span></span> menu, or added as an algorithm in the workflow. There are several
     options for
     displaying trees, available by clicking on one of the two buttons on the right-hand-side of the tree window:
<div
        class="centerline"><img
        src="figs/treeButtons.png" alt="PIC"
></div>
. (see
Section&#x00A0;<a
        href="#x1-90001.6">1.6<!--tex4ht:ref: sec:tree-view-tab --></a>).
<!--l. 65--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">2.3   </span> <a
        id="x1-260002.3"></a>Neighbor Net and other split network methods</h3>
<!--l. 67--><p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">2.3.1   </span> <a
        id="x1-270002.3.1"></a>Neighbor Net</h4>
<!--l. 69--><p class="noindent">Given a distance matrix as input, the <span
        class="cmssi-10">Neighbor Net</span><a
        id="x1-27001"></a> algorithm operates in three stages. First, an agglomerative method is
     used to identify a circular ordering of the taxa. The splits computed by the algorithm are a subset of the set of
     all splits
     that can be formed from consecutive sets of taxa in that ordering. Second, a heavily customized algorithm is
     used to efficiently compute split weights. Those with zero weight are removed (use a split filter to remove
     splits with larger weight). Finally, a planar split network algorithm takes the weighted splits and produces
     the split network representation. A complete description of the entire process is available in <a
             href="#XBryantHuson2023">Bryant and
          Huson</a>&#x00A0;[<a
             href="#XBryantHuson2023">2023</a>].
     <!--l. 73-->
<p class="noindent">There is a single option available in Neighbor Net, the method used to infer split weights. We found
     that the Active
     Set method performed better than the other methods, and this is the default and recommended option.
     We have left the other algorithms as options in order to enable a repeat of the analysis in <a
             href="#XBryantHuson2023">Bryant and


          Huson</a>&#x00A0;[<a
             href="#XBryantHuson2023">2023</a>].
     <!--l. 75-->
<p class="noindent">When Neighbor Net is called, SplitsTree produces a split block and a split network block in the
     workflow. As we stress in
     <a
             href="#XHusonBryant2006">Huson and Bryant</a>&#x00A0;[<a
             href="#XHusonBryant2006">2006</a>], the main information in the network is the set of weighted splits.
     Think of the network as a
     means of visualising the splits, in the sense that the same set of splits can be represented in several different
     ways.
     <!--l. 78-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">2.3.2   </span> <a
        id="x1-280002.3.2"></a>Manipulating split networks</h4>
<!--l. 80--><p class="noindent">To rotate or flip the entire network, use the toolbar revealed by pressing the
     preferences button at the top right
     of the split network panel, making sure that none of the nodes or edges in the network are selected (see
     Section&#x00A0;<a
             href="#x1-130001.10">1.10<!--tex4ht:ref: sec:split-network-tab --></a>).
     <!--l. 82-->
<p class="noindent">Click on an edge in the split network to select that split. The edges associated to that split can
     be rotated using the
     rotate buttons in the toolbar or the arrows in the side panel which appears when you click the button on the right
     (see
     Fig.&#x00A0;<a
             href="#x1-280014">2.4<!--tex4ht:ref: fig:rotateButtons --></a>).
     <!--l. 84-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-280014"></a>


     <!--l. 86--><p class="noindent"><img
        src="figs/rotateButtons.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;2.4: </span><span
             class="content">If one or more splits are selected, then highlighted buttons can be used to change the angles of the
selected splits.</span></div><!--tex4ht:label?: x1-280014 -->


     <!--l. 89--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 91--><p class="noindent">The traditional approach to displaying split networks marks out the splits with a mesh
     of quadrilaterals and polygons. The
     outline representation <a
             href="#XBagcietal2021">Bagci et&#x00A0;al.</a>&#x00A0;[<a
             href="#XBagcietal2021">2021</a>] constructs just the outer perimeter of the network. This is sufficient to
     represent all the split weights, and is generally much faster to compute and draw. To switch back and
     forward between the graph mode and the outline mode use the pop-up menu at the left of the toolbar (see
     Section&#x00A0;<a
             href="#x1-130001.10">1.10<!--tex4ht:ref: sec:split-network-tab --></a>).
<h4 class="subsectionHead"><span class="titlemark">2.3.3   </span> <a
        id="x1-290002.3.3"></a>Split Decomposition</h4>
<!--l. 95--><p class="noindent">Given a distance matrix, the <span
        class="cmssi-10">Split Decomposition</span><a
        id="x1-29001"></a> method [<a
        href="#XBandeltDress1992">Bandelt and Dress</a>,&#x00A0;<a
        href="#XBandeltDress1992">1992</a>] can be selected in the
     Network menu, or on a distances node in the workflow. <span
             class="cmssi-10">Split Decomposition </span>is a predecessor of Neighbor
     Net, though the structures of the two methods are quite different. Split Decomposition works by inferring
     a set of splits satisfying a quartet condition in the distance matrix. Split Decomposition produces a set
     of weakly compatible splits and, as such, can produce more complex split networks than those produced
     by Neighbor Net. The resulting split network will not necessarily be planar. In practice, the conservative
     nature of the selection criteria means that Split Decomposition produces far fewer splits than Neighbor
     Net.
     <!--l. 99-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">2.3.4   </span> <a
        id="x1-300002.3.4"></a>Splits in characters</h4>
<!--l. 101--><p class="noindent">SplitsTree includes several methods for extracting splits directly from character data.
     These methods do not assume any
     explicit model for sequence evolution. As such they do not correct for hidden mutations. However, they can reveal
     important structure within sequences from closely related organisms, as well as artefacts resulting from data
     handling
     problems.
     <!--l. 104-->
<p class="noindent">The simplest is <span class="obeylines-h"><span class="verb"><span
        class="cmtt-10">BinaryToSplits</span></span></span> (see Section&#x00A0;<a
        href="#x1-93000C.1">C.1<!--tex4ht:ref: alg:BinaryToSplits --></a>), which applies to binary data only. Each
     binary character determines a
     split separating those with allele/state 0 and those with allele/state 1. The weight assigned to a split equals the
     summed
     weight for all characters inducing that split, defaulting to a count of those characters if weights are not
     specified. The <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">BinaryToSplits</span></span></span> algorithm is available via the workflow graph or
     workflow hierarchy. The user
     can specify a weight/count threshold on the splits, a cap on the maximum dimension of the split network
     (see Section&#x00A0;<a
             href="#x1-116000C.3">C.3<!--tex4ht:ref: alg:DimensionFilter --></a>) and the option to include all &#8216;trivial&#8217;
     splits separating one taxon from the remainder
     automatically.
     <!--l. 108-->
<p class="noindent">The <span class="obeylines-h"><span class="verb"><span
        class="cmtt-10">DNAtoSplits</span></span></span> method (see Section&#x00A0;<a
        href="#x1-94000C.1">C.1<!--tex4ht:ref: alg:DnaToSplits --></a>) carries out a similar analysis but on nucleotide
     data. Splits are either
     determined via an RY coding (AG vs CT) or by splitting the most frequent state (assumed ancestral) from the other
     states (assumed derived) at each site.
     <!--l. 110-->
<p class="noindent">The <span
        class="cmssi-10">Parsimony Splits </span>method (see Section&#x00A0;<a
        href="#x1-96000C.1">C.1<!--tex4ht:ref: alg:ParsimonySplits --></a>), introduced by <a
        href="#XBandeltDress1992">Bandelt and Dress</a>&#x00A0;[<a
        href="#XBandeltDress1992">1992</a>], produces a set of weakly
     compatible splits directly from character taxa. The method is quartet based, like Split Decomposition, but for each
     four
     taxa, it determines the two most frequent pairings of two taxa versus the other two taxa.
     <!--l. 112-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">2.4   </span> <a
        id="x1-310002.4"></a>Haplotype networks</h3>


<!--l. 114--><p class="noindent">A haplotype network is an elegant and efficient way to represent character or sequence
     data. Each node corresponds to a
     particular sequence with the size of the node proportional to the number of copies of that sequence in the data.
     Sequences which differ in one position are connected by an edge which is (optionally) labelled by the exact
     difference.
     Different methods for constructing haplotype networks generate different graphs for connecting sequences at larger
     distances. For them all, a key property is that given one sequence, the network, and the mutations along each edge,
     the
     entire alignment can be reconstructed.
     <!--l. 116-->
<p class="noindent">SplitsTree provides implementations of two widely-used haplotype network methods, MinSpanningNetwork
     [<a
             href="#XExcoffierSmouse1994">Excoffier and
          Smouse</a>,&#x00A0;<a
             href="#XExcoffierSmouse1994">1994</a>] and MedianJoining [<a
             href="#XBandeltetal1999">Bandelt et&#x00A0;al.</a>,&#x00A0;<a
             href="#XBandeltetal1999">1999</a>]. Haplotype networks are drawn as graphs with each edge
     labelled by marks indicating the number of mutations/differences along that edge. This can be modified
     using the side panel which appears when clicking the preferences button at the top right of the network
     panel.
     <!--l. 119-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">2.4.1   </span> <a
        id="x1-320002.4.1"></a>Minimum spanning network</h4>
<!--l. 121--><p class="noindent">A minimum spanning tree for a graph is a connected subgraph of minimum weight.
     Sometimes there is a unique minimum
     spanning tree; other times there are multiple.
     <!--l. 123-->
<p class="noindent">In this context, the graph contains a node for each input sequence and edge between every pair of
     nodes. The length of
     each edge is the Hamming distance between the corresponding sequences. Other distance measures can be used, but the
     Hamming distance is appropriate for Haplotype Network construction.
     <!--l. 125-->
<p class="noindent">The minimum spanning network is formed from all those edges in the graph which appear in every
     minimum spanning tree
     (see Section&#x00A0;<a
             href="#x1-109000C.2">C.2<!--tex4ht:ref: alg:MinSpanningNetwork --></a>).
     <!--l. 127-->
<p class="noindent">A minimum spanning network is constructed from a characters block by first determining Hamming
     distances (right-click
     on the characters block and select <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">Add</span><span
             class="cmtt-10">&#x00A0;Algorithm</span><span
             class="cmtt-10">&#x00A0;-&#x003E;</span><span
             class="cmtt-10">&#x00A0;Hamming</span><span
             class="cmtt-10">&#x00A0;distance</span></span></span>). Then right-click on the distance block
     produced and add the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">Min</span><span
             class="cmtt-10">&#x00A0;Spanning</span><span
             class="cmtt-10">&#x00A0;Network</span></span></span> algorithm.
     <!--l. 129-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">2.4.2   </span> <a
        id="x1-330002.4.2"></a>Median Joining network</h4>
<!--l. 131--><p class="noindent">Median Joining (see Section&#x00A0;<a
        href="#x1-95000C.1">C.1<!--tex4ht:ref: alg:MedianJoining --></a>) is probably the most highly-cited method for
     constructing phylogenetic networks. The
     implementation in SplitsTree is based on the method described in <a
             href="#XBandeltetal1999">Bandelt et&#x00A0;al.</a>&#x00A0;[<a
             href="#XBandeltetal1999">1999</a>]. The Median-Joining network
     method makes repeated use of minimum spanning networks, each time augmenting the set of observed sequences with
     putative ancestral sequences.
     <!--l. 134-->
<p class="noindent">A Median-Joining network is constructed from a characters block via the Network menu, or by adding
     an
     algorithm to the workflow. The method comes with a single option <span
             class="cmmi-10">&#x03F5; </span>that is an integer controlling a threshold
     determining when two sequences are considered adjacent. In <a
             href="#XBandeltetal1999">Bandelt et&#x00A0;al.</a>&#x00A0;[<a
             href="#XBandeltetal1999">1999</a>], <span
             class="cmmi-10">&#x03F5; </span>varies between <span
             class="cmr-10">0</span>, <span
             class="cmr-10">1 </span>and
     <span
             class="cmr-10">2</span>.
     <!--l. 136-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">2.5   </span> <a
        id="x1-340002.5"></a>Rooted phylogenetic networks</h3>


<!--l. 138--><p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">2.5.1   </span> <a
        id="x1-350002.5.1"></a>Implicit vs explicit trees and networks</h4>
<!--l. 141--><p class="noindent">A haplotype network is a direct representation of the input data and a split network
     represents groupings or splits between
     taxa. Both are examples of so-called <span
             class="cmssi-10">implicit</span><a
             id="x1-35001"></a> or <span
             class="cmssi-10">data-display</span><a
             id="x1-35002"></a> networks that aim at visualizing evolutionary data. In
     contrast, an <span
             class="cmssi-10">explicit network</span><a
             id="x1-35003"></a> is a representation of the putative evolutionary history, including reticulate events
     such as
     speciation-by-hybridization or horizontal gene transfer.
     <!--l. 149-->
<p class="noindent">Strictly speaking, unrooted phylogenetic trees, too, are implicit representations of evolutionary
     data, whereas rooted
     phylogenetic trees have a direction (away from the root) and this allows branching nodes to be explicitly
     interpreted as
     representing speciation events.
     <!--l. 154-->
<p class="noindent">Explicit phylogenetic networks are necessarily rooted. The Autumn algorithm [<a
        href="#XHusonLinz2018">Huson and Linz</a>,&#x00A0;<a
        href="#XHusonLinz2018">2018</a>] (see Section&#x00A0;<a
        href="#x1-124000C.4">C.4<!--tex4ht:ref: alg:AutumnAlgorithm --></a>)
     produces an explicit rooted phylogenetic network in which reticulations may be interpreted as putative
     hybridization or
     HGT events. However, just because a phylogenetic tree has a root does not mean that it is explicit. For example,
     the
     Cluster Network algorithm (see Section&#x00A0;<a
             href="#x1-130000C.4">C.4<!--tex4ht:ref: alg:ClusterNetwork --></a>) takes as input a set of rooted trees
     and aims at displaying all their clusters
     as a rooted network (in the hardwired sense [<a
             href="#XHusonetal2012">Huson et&#x00A0;al.</a>,&#x00A0;<a
             href="#XHusonetal2012">2012</a>]). Here, the reticulate nodes do not have a direct
     biological interpretation.
     <!--l. 164-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">2.5.2   </span> <a
        id="x1-360002.5.2"></a>Hybridization networks</h4>
<!--l. 166--><p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-360015"></a>


     <!--l. 168--><p class="noindent"><img
        src="figs/phylofusion-ndh-genes.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;2.5: </span><span
             class="content">On the left we show 10 different gene trees for the NADH dehydrogenase-like complex in waterlilies
[<a
                  href="#XGruenstaeudl2019">Gruenstaeudl</a>,&#x00A0;<a
                  href="#XGruenstaeudl2019">2019</a>] and on the right we show a network that contains all 10 trees, with hybridization number
<span
        class="cmmi-10">h </span><span
                  class="cmr-10">= 5</span>, computed using the PhyloFusion algorithm.</span></div>
     <!--tex4ht:label?: x1-360015 -->


     <!--l. 175--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 177--><p class="noindent">In mathematical phylogenetics, a <span
        class="cmssi-10">hybrization network</span><a
        id="x1-36002"></a> is a rooted phylogenetic network that contains or displays an input
     set of rooted phylogenetic trees. Usually, the requirement is that such a network minimizes the &#8220;hybridization
     number&#8221;,
     that is, the number of reticulations. (To be precise, a reticulation node of indegree <span
             class="cmmi-10">k </span>contributes <span
             class="cmmi-10">k </span><span
             class="cmsy-10">- </span><span
             class="cmr-10">1 </span>toward the
     hybridization number.)
     <!--l. 184-->
<p class="noindent">SplitsTree currently offers two algorithms for computing such networks for real world data. The
     Autumn algorithm [<a
             href="#XHusonLinz2018">Huson
          and Linz</a>,&#x00A0;<a
             href="#XHusonLinz2018">2018</a>] (see Section&#x00A0;<a
             href="#x1-124000C.4">C.4<!--tex4ht:ref: alg:AutumnAlgorithm --></a>) takes as input two rooted phylogenetic
     trees and computes, as output the list of
     all different hybridization networks that contain the two trees. The input trees may have multifurcations
     and unequal taxon sets. This algorithm aims at providing an exact solution (networks that minimize the
     hybridization) of a computational hard problem, so it might not terminate if the input trees have too many
     conflicts.
     <!--l. 193-->
<p class="noindent">The PhyloFusion algorithm [<a
        href="#XZhangetal2023">Zhang et&#x00A0;al.</a>,&#x00A0;<a
        href="#XZhangetal2023">2023</a>,&#x00A0;<a
        href="#XZhangetal2024">2024</a>] takes as input multiple rooted trees and computes one or
     more rooted phylogenetic networks that display all the input trees. Again, we allow multifurcations and
     missing taxa. This very fast heuristic aims at minimizing the hybridization number. With this, we provide a
     versatile method for exploring the practical use of rooted networks in phylogenetics (see Section&#x00A0;<a
             href="#x1-125000C.4">C.4<!--tex4ht:ref: alg:PhyloFusion --></a>) (see
     Fig.&#x00A0;<a
             href="#x1-360015">2.5<!--tex4ht:ref: fig:phylo-fusion-example --></a>).
<h4 class="subsectionHead"><span class="titlemark">2.5.3   </span> <a
        id="x1-370002.5.3"></a>How to compute a rooted network from rooted trees</h4>
<a
        id="x1-37001"></a>
<!--l. 205--><p class="noindent">Assume that you have a collection of phylogenetic trees for which you would like to
     explore the use of rooted phylogenetic
     networks to represent them. To obtain a useful network, you must setup a pipeline consisting of several steps (see
     Fig.&#x00A0;<a
             href="#x1-370026">2.6<!--tex4ht:ref: fig:phylo-fusion-full --></a>). In this analysis, incorrect edges are
     particularly harmful because they generate unnecessary reticulations and so
     it is important that the input trees have confidence values (such as bootstrap support values, say) associated with
     the
     edges so that low-confidence can be ignored.
     <!--l. 214-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-370026"></a>


     <!--l. 216--><p class="noindent"><img
        src="figs/phylofusion-full.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;2.6: </span><span
             class="content">For a set of 48 genes in waterlilies [<a
             href="#XGruenstaeudl2019">Gruenstaeudl</a>,&#x00A0;<a
             href="#XGruenstaeudl2019">2019</a>], on the left we see that 10 different gene
trees for the NADH dehydrogenase-like complex have bee selected. On the right we see a hybridization network
computed using the PhyloFusion algorithm.</span></div><!--tex4ht:label?: x1-370026 -->


     <!--l. 219--><p class="noindent"></div>
<hr class="endfigure">
<ul class="itemize1">
     <li class="itemize">First, use the <span
             class="cmssi-10">Reroot or Reorder </span>algorithm (see Section&#x00A0;<a
             href="#x1-139000C.4">C.4<!--tex4ht:ref: alg:RerootOrReorderTrees --></a>) to ensure that all trees are
          correctly rooted
          (using either midpoint- or outgroup rooting).
     </li>
     <li class="itemize">Second, use the <span
             class="cmssi-10">Trees Filter </span>(see Section&#x00A0;<a
             href="#x1-142000C.4">C.4<!--tex4ht:ref: alg:TreesFilter --></a>) to select the subset of trees that you
          would like to place into
          the network.
     </li>
     <li class="itemize">Third, use the <span
             class="cmssi-10">Trees Edges Filter </span>(see Section&#x00A0;<a
             href="#x1-144000C.4">C.4<!--tex4ht:ref: alg:TreesEdgesFilter --></a>) to contract any low-confidence edges.
          By default, the
          confidence threshold is set to 70.
     </li>
     <li class="itemize">Finally, use the <span
             class="cmssi-10">Phylo Fusion </span>algorithm (see Section&#x00A0;<a
             href="#x1-125000C.4">C.4<!--tex4ht:ref: alg:PhyloFusion --></a>) to compute a rooted network that contains
          all
          input trees.
     </li>
</ul>
<h4 class="subsectionHead"><span class="titlemark">2.5.4   </span> <a
        id="x1-380002.5.4"></a>Cluster networks</h4>
<!--l. 237--><p class="noindent">The Cluster Network algorithm extracts all clusters from an input set of rooted
     phylogenetic trees and
     computes a network using the cluster-popping algorithm [<a
             href="#XHusonetal2012">Huson et&#x00A0;al.</a>,&#x00A0;<a
             href="#XHusonetal2012">2012</a>]. This is a fast algorithm that
     provides a network that contains all input trees. However, it does not aim at minimizing the hybridization
     number.


<h2 class="chapterHead"><span class="titlemark">Chapter&#x00A0;3</span><br/><a
        id="x1-390003"></a>Consensus trees and networks</h2>
<!--l. 3--><p class="noindent">The methods in this chapter all attempt to summarise information contained in a set of
     trees. (Most also work if the input
     contains rooted phylogenetic networks, in which case the calculations are based on &#8220;hardwired clusters&#8221;
     contained in the
     networks.) There are several possible sources:
<ol class="enumerate1">
     <li
             class="enumerate" id="x1-39002x1">Trees returned from different genes or loci.
     </li>
     <li
             class="enumerate" id="x1-39004x2">Trees produced from different methods.
     </li>
     <li
             class="enumerate" id="x1-39006x3">Trees produced from different bootstrap replicates.
     </li>
     <li
             class="enumerate" id="x1-39008x4">Trees sampled from the posterior distribution in a Bayesian analysis.
     </li>
</ol>
<!--l. 14--><p class="noindent">One of the big improvements with the most recent version of SplitsTree is that the
     routines for reading in files of trees can
     now cope with large tree files or large trees.
<h3 class="sectionHead"><span class="titlemark">3.1   </span> <a
        id="x1-400003.1"></a>Consensus trees</h3>
<!--l. 18--><p class="noindent">A consensus method summarises a set of trees (on the same set of taxa) with a single
     tree. It can be thought of as
     analogous to an average tree or median tree.
     <!--l. 20-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">3.1.1   </span> <a
        id="x1-410003.1.1"></a>Average consensus method</h4>
<!--l. 22--><p class="noindent">The average consensus method implements an idea of <a
        href="#XLapointeCucumel1997">Lapointe and Cucumel</a>&#x00A0;[<a
        href="#XLapointeCucumel1997">1997</a>]. Additive (leaf to leaf) distance
     matrices are constructed for each tree. This can take some time on larger files. The average of these matrices are
     then
     used to construct either a Neighbor-Joining tree or a NeighborNet.
     <!--l. 24-->
<p class="noindent">The method can be called from the workflow by selecting a trees block and adding the algorithm
     &#8216;Average Consensus&#8217;
     (see Section&#x00A0;<a
             href="#x1-126000C.4">C.4<!--tex4ht:ref: alg:AverageConsensus --></a>). Alternatively, add an &#8216;Average
     Distances&#8217; algorithm to the tree block. This creates a new distance
     block which can be output or analysed using a method of choice.
     <!--l. 26-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">3.1.2   </span> <a
        id="x1-420003.1.2"></a>Strict-, majority- and greedy-consensus methods</h4>
<!--l. 28--><p class="noindent">The strict consensus, majority rule consensus and greedy consensus are three of the
     oldest and most widely used
     consensus methods in phylogenetics.
<ul class="itemize1">
     <li class="itemize">The <span
             class="cmssi-10">strict consensus tree </span>is formed from all splits appearing in all trees;
     </li>
     <li class="itemize">The <span
             class="cmssi-10">majority rule tree </span>is formed from all splits appearing in over half the trees;


     </li>
     <li class="itemize">The <span
             class="cmssi-10">greedy consensus tree  </span>is constructed using a greedy algorithm aimed at producing a
          collection of splits
          with maximal weight, the weight of each split given by the number of trees containing it.
     </li>
</ul>
<!--l. 35--><p class="noindent">These methods are available from the Trees menu in the menu bar or by adding an
     algorithm to the trees block in the
     workflow.
     <!--l. 37-->
<p class="noindent">Note that there is a slight difference in the consensus tree depending on whether the input trees
     are to be considered
     rooted or unrooted. For example the two trees
     <!--l. 39-->
<p class="noindent"><!--tex4ht:inline-->
<div class="tabular">
     <table id="TBL-2" class="tabular"
            cellspacing="0" cellpadding="0"
     >
          <colgroup id="TBL-2-1g">
               <col
                       id="TBL-2-1">
               <col
                       id="TBL-2-2">
               <col
                       id="TBL-2-3">
          </colgroup>
          <tr
                  style="vertical-align:baseline;" id="TBL-2-1-">
               <td style="white-space:nowrap; text-align:center;" id="TBL-2-1-1"
                   class="td11"><span
                       class="cmtt-10">((a,b),c,d)</span></td>
               <td style="white-space:nowrap; text-align:center;" id="TBL-2-1-2"
                   class="td11"> and
               </td>
               <td style="white-space:nowrap; text-align:center;" id="TBL-2-1-3"
                   class="td11"><span
                       class="cmtt-10">(a,b,(c,d));</span></td>
          </tr>
          <tr
                  style="vertical-align:baseline;" id="TBL-2-2-">
               <td style="white-space:nowrap; text-align:center;" id="TBL-2-2-1"
                   class="td11"></td>
          </tr>
     </table>
</div>
<!--l. 45--><p class="noindent">share a split <span
        class="cmmi-10">ab</span><span
        class="cmsy-10">|</span><span
        class="cmmi-10">cd </span>which would appear in their unrooted consensus tree, but they share no clusters, so
     their rooted
     consensus tree would be completely unresolved.
     <!--l. 47-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">3.1.3   </span> <a
        id="x1-430003.1.3"></a>Densi-tree consensus</h4>
<!--l. 49--><p class="noindent">The densi-tree consensus<a
        id="x1-43001"></a> [<a
        href="#XBouckaert2010">Bouckaert</a>,&#x00A0;<a
        href="#XBouckaert2010">2010</a>] shows the greedy consensus tree together with a rendering of all input trees
     (see Section&#x00A0;<a
             href="#x1-120001.9">1.9<!--tex4ht:ref: sec:densitree-tab --></a>).
     <!--l. 52-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">3.2   </span> <a
        id="x1-440003.2"></a>Networks representing trees</h3>
<!--l. 54--><p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">3.2.1   </span> <a
        id="x1-450003.2.1"></a>Consensus networks</h4>
<!--l. 56--><p class="noindent">Consensus networks are based on the idea of using split networks to represent more
     splits than can appear in a single tree
     <a
             href="#XBandelt1995">Bandelt</a>&#x00A0;[<a
             href="#XBandelt1995">1995</a>],&#x00A0;<a
             href="#XHollandetal2004">Holland et&#x00A0;al.</a>&#x00A0;[<a
             href="#XHollandetal2004">2004</a>]. They can be constructed using the menu command Network<span
             class="cmmi-10">&#x003E;</span>Consensus network,
     or by adding an algorithm to a trees block in the workflow. Note that, with the menu command, if there is more than
     one
     trees block then SplitsTree will ask the user to select one.
     <!--l. 58-->
<p class="noindent">SplitsTree implements several weighting methods for the splits. These are used to determine the
     split weights used in the
     output tree or network. A standard analysis consensus network analysis will use the frequency (or count) of a split
     as the
     weight used for selecting and displaying splits.
<ul class="itemize1">
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">Mean</span></span></span> - use the mean of the weights in the input trees. This treats
          different trees as estimations of the
          distances between taxa.
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">TreeSizeWeightedMean</span></span></span> - use the mean of the weights in the input trees
          after normalizing each of the
          input trees to total length <span
                  class="cmr-10">1</span>. This should be used if the different trees are on different scales, e.g.
          because
          they were computed using different methods.


     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">Median</span></span></span> - use the median weight. Use as an alternative to mean weights.
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">Count</span></span></span> - use the number of trees that contain a split as its weight.
          This is useful to emphasize the conflicts
          in different trees when using a network for consensus.
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">Sum</span></span></span> - use the sum over all weights in the input trees. Similar
          use-case to counts.
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">Uniform</span></span></span> - give all splits weight 1. This emphasizes the topology of
          the consensus tree or network.
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">TreeNormalizedSum</span></span></span> - use the sum over all weights in the normalized
          input trees. Not sure when you would
          want to use this.
     </li>
</ul>
<!--l. 80--><p class="noindent">The threshold percent controls how many splits are included in the network. When the
     weight is computed from split
     counts it specifies the percentage of trees which a split needs to be contained in for the split to be included in
     the
     network. Reducing this threshold will increase the number of splits, giving a more complex network. The High
     Dimension
     Filter is the same as that used in the split weight filter (see Section&#x00A0;<a
             href="#x1-116000C.3">C.3<!--tex4ht:ref: alg:DimensionFilter --></a>), greedily removing splits which
     generate high
     dimensional boxes in the diagram.
     <!--l. 82-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">3.2.2   </span> <a
        id="x1-460003.2.2"></a>Consensus outline</h4>
<!--l. 84--><p class="noindent">The consensus outline method (see Section&#x00A0;<a
        href="#x1-133000C.4">C.4<!--tex4ht:ref: alg:ConsensusOutline --></a>) takes as input a set of trees and produces
     as output a set of circular
     splits that are displayed either as a planar split network<a
             id="x1-46001"></a> or as a phylogenetic outline<a
             id="x1-46002"></a>. It operates by greedily selecting a
     subset of input splits that are compatible with some circular ordering of the input tree, computed using the
     PQ-tree
     algorithm [<a
             href="#XBoothLueker1976">Booth and Lueker</a>,&#x00A0;<a
             href="#XBoothLueker1976">1976</a>]. One possible application is as an alternative to the densi-tree
     visualization (see
     Fig.&#x00A0;<a
             href="#x1-460031">3.1<!--tex4ht:ref: fig:densi-vs-outline --></a>).
     <!--l. 92-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-460031"></a>


     <!--l. 93--><p class="noindent"><img
        src="figs/densi-vs-outline.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;3.1: </span><span
             class="content">For a Bayesian profile of trees (from the Beast examples directory), we show the densi-tree consensus
on the left and the outline consensus on the right (rooted by the three taxa at the top of both diagrams). </span></div>
     <!--tex4ht:label?: x1-460031 -->


     <!--l. 98--><p class="noindent"></div>
<hr class="endfigure">
<h4 class="subsectionHead"><span class="titlemark">3.2.3   </span> <a
        id="x1-470003.2.3"></a>Confidence networks</h4>
<!--l. 103--><p class="noindent">The idea behind a <span
        class="cmssi-10">confidence network </span>(see Section&#x00A0;<a
        href="#x1-148000C.4">C.4<!--tex4ht:ref: alg:ConfidenceNetwork --></a>) is to choose the threshold in a consensus
     network so that at
     least <span
             class="cmr-10">95% </span>of the trees have <span
             class="cmssi-10">all </span>their splits contained in that network. The method was originally designed
     as a way to create confidence intervals from bootstrap distributions <a
             href="#XHusonBryant2006">Huson and Bryant</a>&#x00A0;[<a
             href="#XHusonBryant2006">2006</a>], however
     the dimensionality of the problem, and shortcomings of empirical bootstrap distributions, meant that the
     confidence sets produced were massive. The same machinery can be readily applied to samples from the
     posterior distribution of trees in a Bayesian analysis, in which case the network represents a confidence
     set.
     <!--l. 105-->
<p class="noindent">The main option in a confidence network is the level, which is <span
        class="cmr-10">0</span><span
        class="cmmi-10">.</span><span
        class="cmr-10">95 </span>by default. This is the proportion of
     input trees which will have their splits contained in the network. Decreasing this number produces smaller
     networks.


     <a
             id="x1-47001r63"></a>
<h2 class="appendixHead"><span class="titlemark">Appendix&#x00A0;A</span><br/><a
        id="x1-48000A"></a>The main menu bar</h2>
<!--l. 5--><p class="noindent">All functionality of the program can be used directly from the main window. In addition,
     the program provides menus to
     access the most often used features.
<h3 class="sectionHead"><span class="titlemark">A.1   </span> <a
        id="x1-49000A.1"></a>The File menu</h3>
<a
        id="x1-49001"></a>
<!--l. 13--><p class="noindent">This menu has the following items:
<ul class="itemize1">
     <li class="itemize">New&#x2026;<a
             id="x1-49002"></a> - Create a new document and open it
     </li>
     <li class="itemize">Open&#x2026;<a
             id="x1-49003"></a> - Open an existing file and process it
     </li>
     <li class="itemize">Replace Data&#x2026;<a
             id="x1-49004"></a> - Replace the current input data
     </li>
     <li class="itemize">Edit Input&#x2026;<a
             id="x1-49005"></a> - Open the input editor tab
     </li>
     <li class="itemize">Analyze Draft Genomes&#x2026;<a
             id="x1-49006"></a> - Open the microbial draft genome analyzer
     </li>
     <li class="itemize">Save <a
             id="x1-49007"></a>- Save the current document
     </li>
     <li class="itemize">Save As&#x2026;<a
             id="x1-49008"></a> - Save the current document to a chosen file
     </li>
     <li class="itemize">Page Setup&#x2026;<a
             id="x1-49009"></a> - Page setup for printing
     </li>
     <li class="itemize">Print&#x2026;<a
             id="x1-49010"></a> - Print the current tab
     </li>
     <li class="itemize">Close <a
             id="x1-49011"></a>- Close the current document
     </li>
</ul>
<!--l. 28--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">A.2   </span> <a
        id="x1-50000A.2"></a>The Edit menu</h3>
<a
        id="x1-50001"></a>
<!--l. 31--><p class="noindent">This menu has the following items:
<ul class="itemize1">
     <li class="itemize">Undo <a
             id="x1-50002"></a>- Undo


     </li>
     <li class="itemize">Redo <a
             id="x1-50003"></a>- Redo
     </li>
     <li class="itemize">Cut <a
             id="x1-50004"></a>- Copy selected text in a text tab or nodes in a workflow tab
     </li>
     <li class="itemize">Copy <a
             id="x1-50005"></a>- Copy selected text in a text tab or nodes in a workflow tab
     </li>
     <li class="itemize">Copy Newick <a
             id="x1-50006"></a>- Copy tree(s) and networks(s) from the current tab to the system clipboard
     </li>
     <li class="itemize">Copy Image <a
             id="x1-50007"></a>- Copy an image of the current tab to the system clipboard
     </li>
     <li class="itemize">Paste <a
             id="x1-50008"></a>- Paste text in a text tab or nodes in a workflow tab
     </li>
     <li class="itemize">Duplicate <a
             id="x1-50009"></a>- Duplicate selected nodes in a workflow tab
     </li>
     <li class="itemize">Delete <a
             id="x1-50010"></a>- Delete selected text in a text tab or nodes in a workflow tab
     </li>
     <li class="itemize">Find&#x2026;<a
             id="x1-50011"></a> - Open the find dialog for the current tab
     </li>
     <li class="itemize">Find Again <a
             id="x1-50012"></a>- Find next match in the current tab
     </li>
     <li class="itemize">Replace&#x2026;<a
             id="x1-50013"></a> - Open the replace dialog for the current tab
     </li>
     <li class="itemize">Goto Line&#x2026;<a
             id="x1-50014"></a> - Go to a specific line in current text tab
     </li>
</ul>
<!--l. 49--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">A.3   </span> <a
        id="x1-51000A.3"></a>The Select menu</h3>
<a
        id="x1-51001"></a>
<!--l. 52--><p class="noindent">This menu has the following items:
<ul class="itemize1">
     <li class="itemize">Select All <a
             id="x1-51002"></a>- Select all taxa or text in the current tab
     </li>
     <li class="itemize">Select None <a
             id="x1-51003"></a>- De-select all taxa or text in the current tab
     </li>
     <li class="itemize">Select Inverse <a
             id="x1-51004"></a>- Invert the selection in the current tab
     </li>
     <li class="itemize">From Previous Window <a
             id="x1-51005"></a>- Select taxa in current tab that were selected in the previously focused window
     </li>
     <li class="itemize">Brackets <a
             id="x1-51006"></a>- Select text between matching brackets


     </li>
     <li class="itemize">Compatible Sites <a
             id="x1-51007"></a>- Select all compatible sites in the alignment tab
     </li>
</ul>
<!--l. 63--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">A.4   </span> <a
        id="x1-52000A.4"></a>The View menu</h3>
<a
        id="x1-52001"></a>
<!--l. 66--><p class="noindent">This menu has the following items:
<ul class="itemize1">
     <li class="itemize">Use Dark Theme <a
             id="x1-52002"></a>- Toggle dark mode
     </li>
     <li class="itemize">Enter Full Screen <a
             id="x1-52003"></a>- Toggle full-screen mode
     </li>
     <li class="itemize">Increase Font Size <a
             id="x1-52004"></a>- Increase the font size in the current tab
     </li>
     <li class="itemize">Decrease Font Size <a
             id="x1-52005"></a>- Decrease the font size in the current tab
     </li>
     <li class="itemize">Zoom In <a
             id="x1-52006"></a>- Increase the scale factor in the current tab
     </li>
     <li class="itemize">Zoom Out <a
             id="x1-52007"></a>- Decrease the scale factor in the current tab
     </li>
     <li class="itemize">Zoom In Horizontal <a
             id="x1-52008"></a>- Increase the horizontal scale factor in the current tab
     </li>
     <li class="itemize">Zoom Out Horizontal <a
             id="x1-52009"></a>- Decrease the horizontal scale factor in the current tab
     </li>
     <li class="itemize">Reset <a
             id="x1-52010"></a>- Reset the scale factor in the current tab
     </li>
     <li class="itemize">Rotate Left <a
             id="x1-52011"></a>- Left rotate the tree or network, or selected splits, in the current tab
     </li>
     <li class="itemize">Rotate Right <a
             id="x1-52012"></a>- Right rotate the tree or network, or selected splits, in the current tab
     </li>
     <li class="itemize">Flip <a
             id="x1-52013"></a>- Flip the tree or network in the current tab
     </li>
     <li class="itemize">Show Scale Bar <a
             id="x1-52014"></a>- Show a scale bar in the current tab
     </li>
     <li class="itemize">Show QR Code <a
             id="x1-52015"></a>- Show or hide a QR code for the tree or network in the current tab
     </li>
     <li class="itemize">Layout Labels <a
             id="x1-52016"></a>- Rerun label layout
     </li>
</ul>


<!--l. 86--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">A.5   </span> <a
        id="x1-53000A.5"></a>The Data menu</h3>
<a
        id="x1-53001"></a>
<!--l. 89--><p class="noindent">This menu has the following items:
<ul class="itemize1">
     <li class="itemize">Filter Taxa <a
             id="x1-53002"></a>- Show the filter taxa tab
     </li>
     <li class="itemize">Filter Characters&#x2026;<a
             id="x1-53003"></a> - Shows the alignment tab
     </li>
     <li class="itemize">Group Identical Haplotypes&#x2026;<a
             id="x1-53004"></a> - Group identical haplotypes and open as new document
     </li>
     <li class="itemize">Filter Trees <a
             id="x1-53005"></a>- Show the filter trees tab
     </li>
     <li class="itemize">Filter Splits <a
             id="x1-53006"></a>- Show the filter splits tab
     </li>
     <li class="itemize">Splits Slider <a
             id="x1-53007"></a>- Show the splits slider (filter by weight) tab
     </li>
     <li class="itemize">Edit Traits <a
             id="x1-53008"></a>- Show the edit traits tab
     </li>
</ul>
<!--l. 101--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">A.6   </span> <a
        id="x1-54000A.6"></a>The Distances menu</h3>
<a
        id="x1-54001"></a>
<!--l. 104--><p class="noindent">This menu has the following items:
<ul class="itemize1">
     <li class="itemize">P Distances <a
             id="x1-54002"></a>- Compute distances based on the normalized number of character-state differences
     </li>
     <li class="itemize">Log Det <a
             id="x1-54003"></a>- Compute distances using the Log-Det method
     </li>
     <li class="itemize">Jukes Cantor <a
             id="x1-54004"></a>- Compute distances using the Jukes-Cantor model
     </li>
     <li class="itemize">K2P <a
             id="x1-54005"></a>- Compute distances using the Kimura-2P model
     </li>
     <li class="itemize">F81 <a
             id="x1-54006"></a>- Compute distances using the Felsenstein-1981 model
     </li>
     <li class="itemize">HKY 85 <a
             id="x1-54007"></a>- Compute distances using the Hasegawa-Kishino-Yano model
     </li>
     <li class="itemize">F84 <a
             id="x1-54008"></a>- Compute distances using the Felsenstein-1984 model


     </li>
     <li class="itemize">Protein ML Dist <a
             id="x1-54009"></a>- Compute distances for proteins using maximum-likelihood estimation
     </li>
     <li class="itemize">Gene Content Distance <a
             id="x1-54010"></a>- Compute distances based on the presence/absence of genes
     </li>
</ul>
<!--l. 118--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">A.7   </span> <a
        id="x1-55000A.7"></a>The Tree menu</h3>
<a
        id="x1-55001"></a>
<!--l. 121--><p class="noindent">This menu has the following items:
<ul class="itemize1">
     <li class="itemize">NJ <a
             id="x1-55002"></a>- Compute a tree from distances using the Neighbor-Joining method
     </li>
     <li class="itemize">BioNJ <a
             id="x1-55003"></a>- Compute a tree from distances using the Bio-NJ method
     </li>
     <li class="itemize">UPGMA <a
             id="x1-55004"></a>- Compute a rooted tree from distances using the UPGMA method
     </li>
     <li class="itemize">Buneman Tree <a
             id="x1-55005"></a>- Compute a tree from distances using the Buneman tree method
     </li>
     <li class="itemize">Consensus Tree <a
             id="x1-55006"></a>- Compute a consensus tree
     </li>
     <li class="itemize">Minimum Spanning Tree <a
             id="x1-55007"></a>- Compute a minimum spanning tree from distances
     </li>
     <li class="itemize">Reroot Or Reorder Trees <a
             id="x1-55008"></a>- Open the reroot or reorder tab
     </li>
     <li class="itemize">Show Single Tree <a
             id="x1-55009"></a>- Show a single tree in a tab
     </li>
     <li class="itemize">Show Tree Pages <a
             id="x1-55010"></a>- Show multiple trees in a tab
     </li>
     <li class="itemize">Show Tanglegram <a
             id="x1-55011"></a>- Show two trees as a tanglegram
     </li>
     <li class="itemize">Show DensiTree <a
             id="x1-55012"></a>- Show a Bayesian chain of trees as a densi-tree
     </li>
</ul>
<!--l. 137--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">A.8   </span> <a
        id="x1-56000A.8"></a>The Network menu</h3>
<a
        id="x1-56001"></a>
<!--l. 140--><p class="noindent">This menu has the following items:


<ul class="itemize1">
     <li class="itemize">Neighbor Net <a
             id="x1-56002"></a>- Compute a planar split network from distances using the Neighbor-Net method
     </li>
     <li class="itemize">Split Decomposition <a
             id="x1-56003"></a>- Compute a split network from distances using the split-decomposition method
     </li>
     <li class="itemize">Parsimony Splits <a
             id="x1-56004"></a>- Compute a split network from DNA sequences using the parsimony-splits method
     </li>
     <li class="itemize">Consensus Splits <a
             id="x1-56005"></a>- Open the consensus splits tab
     </li>
     <li class="itemize">Consensus Network <a
             id="x1-56006"></a>- Compute a consensus network from trees
     </li>
     <li class="itemize">Consensus Outline <a
             id="x1-56007"></a>- Compute a consensus outline from trees
     </li>
     <li class="itemize">Super Network <a
             id="x1-56008"></a>- Compute a super network from trees on unequal taxon sets
     </li>
     <li class="itemize">Median Joining Network <a
             id="x1-56009"></a>- Compute a haplotype network from DNA sequences using the median-joining
          algorithm
     </li>
     <li class="itemize">Min Spanning Network <a
             id="x1-56010"></a>- Compute a minimum spanning network from distances
     </li>
     <li class="itemize">World Map <a
             id="x1-56011"></a>- Show the world map tab
     </li>
     <li class="itemize">Hybridization Network <a
             id="x1-56012"></a>- Compute a minimum hybridization network for two rooted trees using the autumn
          algorithm
     </li>
     <li class="itemize">Cluster Network <a
             id="x1-56013"></a>- Compute a cluster network from trees
     </li>
</ul>
<!--l. 157--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">A.9   </span> <a
        id="x1-57000A.9"></a>The Analysis menu</h3>
<a
        id="x1-57001"></a>
<!--l. 160--><p class="noindent">This menu has the following items:
<ul class="itemize1">
     <li class="itemize">Bootstrap Tree <a
             id="x1-57002"></a>- Perform bootstrapping on a tree calculation from sequences
     </li>
     <li class="itemize">Bootstrap Tree as Network <a
             id="x1-57003"></a>- Perform bootstrapping on a tree calculation from sequences, producing a
          network
     </li>
     <li class="itemize">Bootstrap Splits Network <a
             id="x1-57004"></a>- Perform bootstrapping on a split network calculation from sequences


     </li>
     <li class="itemize">Estimate Invariable Sites <a
             id="x1-57005"></a>- Estimate the proportion of invariable sites
     </li>
     <li class="itemize">Compute Delta Score <a
             id="x1-57006"></a>- Compute the delta score from distances
     </li>
     <li class="itemize">Run Phi Test for Recombination <a
             id="x1-57007"></a>- Run the Phi Test to detect recombination in sequences
     </li>
     <li class="itemize">Compute Tajima&#8217;s D <a
             id="x1-57008"></a>- Run the Phi Test to detect recombination in sequences
     </li>
     <li class="itemize">Splits Phylogenetic Diversity <a
             id="x1-57009"></a>- Compute the phylogenetic diversity for selected taxa on splits
     </li>
     <li class="itemize">Splits Shapley Values <a
             id="x1-57010"></a>- Compute Shapely values on splits
     </li>
     <li class="itemize">Tree Phylogenetic Diversity <a
             id="x1-57011"></a>- Compute the phylogenetic diversity for selected taxa on trees
     </li>
     <li class="itemize">Rooted Tree Fair Proportion Diversity Index <a
             id="x1-57012"></a>- Compute rooted-tree fair-proportion diversity index
     </li>
     <li class="itemize">Rooted Tree Equal Splits Diversity Index <a
             id="x1-57013"></a>- Compute rooted-tree equal-splits diversity index
     </li>
     <li class="itemize">Unrooted Tree Shapley Values <a
             id="x1-57014"></a>- Compute Shapely values on unrooted trees
     </li>
     <li class="itemize">PCoA <a
             id="x1-57015"></a>- Perform principal coordinate analysis on distances
     </li>
     <li class="itemize">Show Workflow <a
             id="x1-57016"></a>- Show the workflow tab
     </li>
</ul>
<!--l. 180--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">A.10   </span> <a
        id="x1-58000A.10"></a>The Window menu</h3>
<a
        id="x1-58001"></a>
<!--l. 183--><p class="noindent">This menu has the following items:
<ul class="itemize1">
     <li class="itemize">Show Message Window&#x2026;<a
             id="x1-58002"></a> - Show or hide the message window
     </li>
     <li class="itemize">Set Window Size&#x2026;<a
             id="x1-58003"></a> - Set the window size
     </li>
     <li class="itemize">Untitled <a
             id="x1-58004"></a></li>
</ul>


<!--l. 191--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">A.11   </span> <a
        id="x1-59000A.11"></a>The Help menu</h3>
<a
        id="x1-59001"></a>
<!--l. 194--><p class="noindent">This menu has the following items:
<ul class="itemize1">
     <li class="itemize">Check For Updates&#x2026;<a
             id="x1-59002"></a> - Check whether an update is available for download
     </li>
     <li class="itemize">About&#x2026;<a
             id="x1-59003"></a> - Show splash screen (authors, version number and license).
     </li>
</ul>


<a
        id="x1-59004r63"></a>
<h2 class="appendixHead"><span class="titlemark">Appendix&#x00A0;B</span><br/><a
        id="x1-60000B"></a>Main data blocks</h2>
<!--l. 6--><p class="noindent">SplitsTree is organized around data blocks that correspond to &#8220;Nexus&#8221; blocks
     [<a
             href="#XMaddisonetal1997">Maddison et&#x00A0;al.</a>,&#x00A0;<a
             href="#XMaddisonetal1997">1997</a>].
<h3 class="sectionHead"><span class="titlemark">B.1   </span> <a
        id="x1-61000B.1"></a>Taxa block</h3>
<a
        id="x1-61001"></a>
<!--l. 10--><p class="noindent">This block maintains the list of all taxa in the analysis. There is a fixed number
     (nTax) of taxa and each has an id
     1..nTax and an unique label. Optionally, an info string can be provided for each taxon. Also, a display
     label may be provided for each taxon. This can include certain HTML tags that are used to render the
     label.


<div class="verbatim" id="verbatim-2">
     BEGIN&#x00A0;TAXA;
     &#x00A0;<br/>&#x00A0;&#x00A0;[TITLE&#x00A0;title;]
     &#x00A0;<br/>&#x00A0;&#x00A0;DIMENSIONS&#x00A0;NTAX=number-of-taxa;
     &#x00A0;<br/>&#x00A0;&#x00A0;[TAXLABELS
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;list-of-labels
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[TAXINFO
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;list-of-info-items&#x00A0;(use&#x00A0;&#8217;null&#8217;&#x00A0;for&#x00A0;missing&#x00A0;item)
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[DISPLAYLABELS
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;list-of-html-strings&#x00A0;(use&#x00A0;&#8217;null&#8217;&#x00A0;for&#x00A0;missing&#x00A0;item)
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>END;
</div>
<!--l. 30--><p class="nopar">
     <!--l. 32-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">B.2   </span> <a
        id="x1-62000B.2"></a>Traits block</h3>
<a
        id="x1-62001"></a>
<!--l. 34--><p class="noindent">This block contains traits associated with the input taxa. Each trait has a label,
     optional latitude and longitude, and a
     value, which can either be a number or a string.


<div class="verbatim" id="verbatim-3">
     BEGIN&#x00A0;TRAITS;
     &#x00A0;<br/>&#x00A0;&#x00A0;[TITLE&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;DIMENSIONS&#x00A0;[NTAX=number-of-taxa]&#x00A0;NTRAITS=number-of-traits;
     &#x00A0;<br/>&#x00A0;&#x00A0;[FORMAT
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[LABELS={YES|NO}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[MISSING=symbol]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[SEPARATOR={COMMA|SEMICOLON|WHITESPACE}]
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[TRAITLATITUDE&#x00A0;&#x00A0;latitude-trait-1&#x00A0;&#x00A0;latitude-trait-2&#x00A0;...&#x00A0;&#x00A0;latitude-trait-n;
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;TRAITLONGITUDE&#x00A0;longitude-trait-1&#x00A0;longitude-trait-2&#x00A0;...&#x00A0;longitude-trait-n;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[TRAITCOLOR&#x00A0;color-trait-1&#x00A0;&#x00A0;color-trait-2&#x00A0;...&#x00A0;&#x00A0;color-trait-n;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[TAXONCOLOR&#x00A0;color-taxon-1&#x00A0;&#x00A0;color-taxon-2&#x00A0;...&#x00A0;&#x00A0;color-taxon-ntax;]
     &#x00A0;<br/>
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;TRAITLABELS&#x00A0;label-trait-1&#x00A0;label-trait-2&#x00A0;...&#x00A0;label-trait-n;
     &#x00A0;<br/>&#x00A0;&#x00A0;MATRIX
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;trait&#x00A0;data&#x00A0;in&#x00A0;specified&#x00A0;format
     &#x00A0;<br/>&#x00A0;&#x00A0;;
     &#x00A0;<br/>END;
</div>
<!--l. 56--><p class="nopar">
     <!--l. 58-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">B.3   </span> <a
        id="x1-63000B.3"></a>Characters block</h3>
<a
        id="x1-63001"></a>
<!--l. 60--><p class="noindent">This block maintains a set of characters or a multiple sequence alignment. There is a
     fixed number of taxa and characters
     or positions. Several different formats are supported. Characters can have weights and both characters and their
     states
     can have labels.


<div class="verbatim" id="verbatim-4">
     BEGIN&#x00A0;CHARACTERS;
     &#x00A0;<br/>&#x00A0;&#x00A0;[TITLE&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;[LINK&#x00A0;{type}&#x00A0;=&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;DIMENSIONS&#x00A0;[NTAX=number-of-taxa]&#x00A0;NCHAR=number-of-characters;
     &#x00A0;<br/>&#x00A0;&#x00A0;[FORMAT
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[DATATYPE={STANDARD|DNA|RNA|PROTEIN|MICROSAT}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[RESPECTCASE]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[MISSING=symbol]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[GAP=symbol]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[MatchChar=symbol]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[SYMBOLS="symbol&#x00A0;symbol&#x00A0;..."]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[LABELS={NO|LEFT}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[TRANSPOSE={NO|YES}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[INTERLEAVE={NO|YES}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[TOKENS=NO]
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[CHARWEIGHTS&#x00A0;wgt_1&#x00A0;wgt_2&#x00A0;...&#x00A0;wgt_nchar;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[CHARSTATELABELS&#x00A0;character-number&#x00A0;[&#x00A0;character-name&#x00A0;][&#x00A0;/state-name&#x00A0;[&#x00A0;state-name...&#x00A0;]&#x00A0;],&#x00A0;...;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[CHARLABELS&#x00A0;character-name&#x00A0;[character-name...];]
     &#x00A0;<br/>&#x00A0;&#x00A0;MATRIX
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;sequence&#x00A0;data&#x00A0;in&#x00A0;specified&#x00A0;format
     &#x00A0;<br/>&#x00A0;&#x00A0;;
     &#x00A0;<br/>END;
</div>
<!--l. 89--><p class="nopar">
     <!--l. 91-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">B.4   </span> <a
        id="x1-64000B.4"></a>Distances block</h3>
<a
        id="x1-64001"></a>
<!--l. 93--><p class="noindent">This block maintains a distance matrix of size nTax times nTax.


<div class="verbatim" id="verbatim-5">
     BEGIN&#x00A0;DISTANCES;
     &#x00A0;<br/>&#x00A0;&#x00A0;[TITLE&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;[LINK&#x00A0;{type}&#x00A0;=&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;[DIMENSIONS&#x00A0;NTAX=number-of-taxa;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[FORMAT
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[TRIANGLE={LOWER|UPPER|BOTH}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[[NO]&#x00A0;DIAGONAL]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[LABELS={LEFT|NO}]
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>&#x00A0;&#x00A0;MATRIX
     &#x00A0;<br/>distance&#x00A0;data&#x00A0;in&#x00A0;specified&#x00A0;format
     &#x00A0;<br/>&#x00A0;&#x00A0;;
     &#x00A0;<br/>END;
</div>
<!--l. 109--><p class="nopar">
     <!--l. 111-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">B.5   </span> <a
        id="x1-65000B.5"></a>Trees block</h3>
<a
        id="x1-65001"></a>
<!--l. 113--><p class="noindent">This block maintains a list of trees. These can be rooted or unrooted phylogenetic
     trees, or rooted phylogenetic networks.
     Trees can partial in the sense that they need to contain all taxa.


<div class="verbatim" id="verbatim-6">
     BEGIN&#x00A0;TREES;
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[TITLE&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[LINK&#x00A0;{type}&#x00A0;=&#x00A0;{title};]
     &#x00A0;<br/>[PROPERTIES&#x00A0;[PARTIALTREES={YES|NO}]&#x00A0;[ROOTED={YES|NO}]&#x00A0;[RETICULATED={YES|NO}];]
     &#x00A0;<br/>[TRANSLATE
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;nodeLabel1&#x00A0;taxon1,
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;nodeLabel2&#x00A0;taxon2,
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;...
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;nodeLabelN&#x00A0;taxonN
     &#x00A0;<br/>;]
     &#x00A0;<br/>[TREE&#x00A0;name1&#x00A0;=&#x00A0;tree1-in-Newick-format;]
     &#x00A0;<br/>[TREE&#x00A0;name2&#x00A0;=&#x00A0;tree2-in-Newick-format;]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;...
     &#x00A0;<br/>[TREE&#x00A0;nameM&#x00A0;=&#x00A0;treeM-in-Newick-format;]
     &#x00A0;<br/>END;
</div>
<!--l. 133--><p class="nopar">
     <!--l. 135-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">B.6   </span> <a
        id="x1-66000B.6"></a>Splits block</h3>
<a
        id="x1-66001"></a>
<!--l. 137--><p class="noindent">This block maintains a set of splits, usually with weights, sometimes with confidence
     values, and usually with a suitable
     cycle for layout purposes.


<div class="verbatim" id="verbatim-7">
     BEGIN&#x00A0;SPLITS;
     &#x00A0;<br/>&#x00A0;&#x00A0;[TITLE&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;[LINK&#x00A0;{type}&#x00A0;=&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;[DIMENSIONS&#x00A0;[NTAX=number-of-taxa]&#x00A0;[NSPLITS=number-of-splits];]
     &#x00A0;<br/>&#x00A0;&#x00A0;[FORMAT
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[Labels={LEFT|NO}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[Weights={YES|NO}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[Confidences={YES|NO}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[Intervals={YES|NO}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[ShowBothSides={NO|YES}]
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[Threshold=non-negative-number;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[PROPERTIES
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[Fit=non-negative-number]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[{Compatible|Cyclic|Weakly&#x00A0;Compatible|Incompatible]
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[CYCLE&#x00A0;[taxon_i_1&#x00A0;taxon_i_2&#x00A0;...&#x00A0;taxon_i_ntax];]
     &#x00A0;<br/>&#x00A0;&#x00A0;[SPLITSLABELS&#x00A0;label_1&#x00A0;label_2&#x00A0;...&#x00A0;label_nsplits;]
     &#x00A0;<br/>MATRIX
     &#x00A0;<br/>[label_1]&#x00A0;[weight_1]&#x00A0;[confidence_1]&#x00A0;split_1,
     &#x00A0;<br/>[label_2]&#x00A0;[weight_2]&#x00A0;[confidence_2]&#x00A0;split_2,
     &#x00A0;<br/>&#x00A0;....
     &#x00A0;<br/>[label_nsplits]&#x00A0;[weight_nsplits]&#x00A0;[confidence_nsplits]&#x00A0;split_nsplits[,]
     &#x00A0;<br/>;
     &#x00A0;<br/>END;
</div>
<!--l. 166--><p class="nopar">
     <!--l. 168-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">B.7   </span> <a
        id="x1-67000B.7"></a>Network block</h3>
<a
        id="x1-67001"></a>
<!--l. 170--><p class="noindent">Maintain a network, such as a haplotype network or just a set of points (for PCoA).


<div class="verbatim" id="verbatim-8">
     BEGIN&#x00A0;NETWORK;
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[TITLE&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;[LINK&#x00A0;{type}&#x00A0;=&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;[DIMENSIONS&#x00A0;[NVertices=number-of-nodes]&#x00A0;[NEdges=number-of-edges];]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[TYPE&#x00A0;{HaplotypeNetwork|Points|Other};]
     &#x00A0;<br/>&#x00A0;&#x00A0;[FORMAT
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[PROPERTIES
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[info&#x00A0;=&#8217;&#x00A0;information&#x00A0;string&#x00A0;to&#x00A0;be&#x00A0;shown&#x00A0;with&#x00A0;plot&#8217;]
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>&#x00A0;&#x00A0;VERTICES
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;ID=number&#x00A0;[LABEL=label]&#x00A0;[x=number]&#x00A0;[y=number]&#x00A0;[key=value&#x00A0;...],
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;...
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;ID=number&#x00A0;[LABEL=label]&#x00A0;[x=number]&#x00A0;[y=number]&#x00A0;[key=value&#x00A0;...]
     &#x00A0;<br/>&#x00A0;&#x00A0;;
     &#x00A0;<br/>&#x00A0;&#x00A0;EDGES
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;ID=number&#x00A0;SID=number&#x00A0;TID=number&#x00A0;[LABEL=label]&#x00A0;[key=value&#x00A0;...],
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;...
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;ID=number&#x00A0;SID=number&#x00A0;TID=number&#x00A0;[LABEL=label]&#x00A0;[key=value&#x00A0;...]
     &#x00A0;<br/>&#x00A0;&#x00A0;;
     &#x00A0;<br/>END;
</div>
<!--l. 194--><p class="nopar">
     <!--l. 196-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">B.8   </span> <a
        id="x1-68000B.8"></a>View block</h3>
<a
        id="x1-68001"></a>
<!--l. 198--><p class="noindent">This block represents a viewer.


<div class="verbatim" id="verbatim-9">
     BEGIN&#x00A0;VIEW;
     &#x00A0;<br/>&#x00A0;&#x00A0;[TITLE&#x00A0;title;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[LINK&#x00A0;{type}&#x00A0;=&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;NAME&#x00A0;&#x003C;name&#x003E;;
     &#x00A0;<br/>&#x00A0;&#x00A0;INPUT&#x00A0;&#x003C;input-block-name&#x003E;;
     &#x00A0;<br/>&#x00A0;&#x00A0;[OPTIONS
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x003C;name&#x003E;=&#x003C;value&#x003E;,
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;...
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x003C;name&#x003E;=&#x003C;value&#x003E;
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>END;
</div>
<!--l. 212--><p class="nopar">
     <!--l. 214-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">B.9   </span> <a
        id="x1-69000B.9"></a>Algorithms block</h3>
<a
        id="x1-69001"></a>
<!--l. 216--><p class="noindent">This block represents an algorithm.


<div class="verbatim" id="verbatim-10">
     BEGIN&#x00A0;ALGORITHM;
     &#x00A0;<br/>&#x00A0;&#x00A0;[TITLE&#x00A0;&#x003C;title&#x003E;;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[LINK&#x00A0;&#x003C;parent-block-type&#x003E;&#x00A0;=&#x00A0;&#x003C;parent-title&#x003E;;]
     &#x00A0;<br/>&#x00A0;&#x00A0;NAME&#x00A0;&#x003C;name&#x003E;;
     &#x00A0;<br/>&#x00A0;&#x00A0;[OPTIONS
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x003C;name&#x003E;=&#x003C;value&#x003E;,
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;...
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x003C;name&#x003E;=&#x003C;value&#x003E;
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>END;
</div>
<!--l. 229--><p class="nopar">
     <!--l. 231-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">B.10   </span> <a
        id="x1-70000B.10"></a>Report block</h3>
<a
        id="x1-70001"></a>
<!--l. 233--><p class="noindent">This block represents a textual report.


<div class="verbatim" id="verbatim-11">
     BEGIN&#x00A0;REPORT;
     &#x00A0;<br/>&#x00A0;&#x00A0;[TITLE&#x00A0;title;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[LINK&#x00A0;{type}&#x00A0;=&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;TEXT
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;text...
     &#x00A0;<br/>&#x00A0;&#x00A0;;
     &#x00A0;<br/>END;
</div>
<!--l. 243--><p class="nopar">
     <!--l. 245-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">B.11   </span> <a
        id="x1-71000B.11"></a>Sets block</h3>
<a
        id="x1-71001"></a>
<!--l. 247--><p class="noindent">This block represents a collection of taxon sets and/or character sets.


<div class="verbatim" id="verbatim-12">
     BEGIN&#x00A0;SETS;
     &#x00A0;<br/>&#x00A0;&#x00A0;[TITLE&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;[TAXSET&#x00A0;{name}={list&#x00A0;of&#x00A0;names&#x00A0;and/or&#x00A0;IDS};]
     &#x00A0;<br/>&#x00A0;&#x00A0;...
     &#x00A0;<br/>&#x00A0;&#x00A0;[CHARSET&#x00A0;{name}={list&#x00A0;of&#x00A0;positions};]
     &#x00A0;<br/>&#x00A0;&#x00A0;...
     &#x00A0;<br/>END;
</div>
<!--l. 257--><p class="nopar">
     <!--l. 259-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">B.12   </span> <a
        id="x1-72000B.12"></a>SplitsTree6 block</h3>
<a
        id="x1-72001"></a>
<!--l. 261--><p class="noindent">This block holds program-specific data and its presence in a file indicates that the
     file was generated by SplitsTree6 and
     represents a complete analysis.


<div class="verbatim" id="verbatim-13">
     BEGIN&#x00A0;SPLITSTREE6;
     &#x00A0;<br/>&#x00A0;&#x00A0;DIMENSIONS&#x00A0;nDataNodes=number&#x00A0;nAlgorithms=number;
     &#x00A0;<br/>&#x00A0;&#x00A0;PROGRAM&#x00A0;version=version-string;
     &#x00A0;<br/>&#x00A0;&#x00A0;WORKFLOW&#x00A0;creationDate=long;
     &#x00A0;<br/>END;
</div>
<!--l. 270--><p class="nopar">
     <!--l. 272-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">B.13   </span> <a
        id="x1-73000B.13"></a>Genomes block</h3>
<a
        id="x1-73001"></a>
<!--l. 274--><p class="noindent">This block represents a collection of genomes.


<div class="verbatim" id="verbatim-14">
     BEGIN&#x00A0;GENOMES;
     &#x00A0;<br/>&#x00A0;&#x00A0;[TITLE&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;[LINK&#x00A0;{type}&#x00A0;=&#x00A0;{title};]
     &#x00A0;<br/>&#x00A0;&#x00A0;[DIMENSIONS&#x00A0;NTAX=number-of-taxa;]
     &#x00A0;<br/>&#x00A0;&#x00A0;[FORMAT
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[LABELS={YES|NO}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[ACCESSIONS={YES|NO}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[MULTIPART={YES|NO}]
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;[FILES={YES|NO}]
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>&#x00A0;&#x00A0;MATRIX
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;[label]&#x00A0;[accession]&#x00A0;length&#x00A0;{sequence&#x00A0;|&#x00A0;[#parts]&#x00A0;length&#x00A0;{sequence|{file:..&#x00A0;offset}}&#x00A0;..&#x00A0;length&#x00A0;{sequence|{file&#x00A0;offset}}},
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;...
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;[label]&#x00A0;[accession]&#x00A0;length&#x00A0;{sequence&#x00A0;|&#x00A0;[#parts]&#x00A0;length&#x00A0;{sequence|{file:..&#x00A0;offset}}&#x00A0;..&#x00A0;length&#x00A0;{sequence|{file&#x00A0;offset}}}
     &#x00A0;<br/>&#x00A0;&#x00A0;;]
     &#x00A0;<br/>END;
</div>
<!--l. 293--><p class="nopar">


     <a
             id="x1-73002r63"></a>
<h2 class="appendixHead"><span class="titlemark">Appendix&#x00A0;C</span><br/><a
        id="x1-74000C"></a>Algorithms</h2>
<!--l. 6--><p class="noindent">Here we list of all provided algorithms, organized by input data.
<h3 class="sectionHead"><span class="titlemark">C.1   </span> <a
        id="x1-75000C.1"></a>Algorithms on a Characters Block</h3>
<!--l. 10--><p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-76000C.1"></a>P Distance</h5>
<a
        id="x1-76001"></a>
<!--l. 12--><p class="noindent">The <span
        class="cmssi-10">P Distance </span>algorithm takes a Characters block as input and produces a Distances block as
     output. It computes the
     normalized Hamming distance. The algorithm has the following options:
     <!--l. 16-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">HandleAmbiguousStates</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Ignore</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;AverageStates</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;MatchStates}</span></span></span> <span
        class="cmss-8">- choose how to handle ambiguous states (nucleotide data</span>
     <span
             class="cmss-8">only)</span>
     <!--l. 19-->
<p class="noindent">Reference: [<a
        href="#XHamming1950">Hamming</a>,&#x00A0;<a
        href="#XHamming1950">1950</a>]
     <!--l. 21-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-77000C.1"></a>Hamming Distance</h5>
<a
        id="x1-77001"></a>
<!--l. 23--><p class="noindent">The <span
        class="cmssi-10">Hamming Distance </span>algorithm takes a Characters block as input and produces a Distances
     block as output. It
     computes the Hamming distance, that is the number of differences between sequences The algorithm has the following
     options:
     <!--l. 27-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">HandleAmbiguousStates</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Ignore</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;AverageStates</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;MatchStates}</span></span></span> <span
        class="cmss-8">- choose how to handle ambiguous states (nucleotide data</span>
     <span
             class="cmss-8">only)</span>
     <!--l. 30-->
<p class="noindent">Reference: [<a
        href="#XHamming1950">Hamming</a>,&#x00A0;<a
        href="#XHamming1950">1950</a>]
     <!--l. 32-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-78000C.1"></a>Log Det</h5>
<a
        id="x1-78001"></a>
<!--l. 34--><p class="noindent">The <span
        class="cmssi-10">Log Det </span>algorithm takes a Characters block as input and produces a Distances block as
     output. It computes distances
     using the Log-Det method. The algorithm has the following options:
     <!--l. 38-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">PropInvariableSites</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- proportion of invariable sites</span>
     <!--l. 40-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">FudgeFactor</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- input missing matrix entries using LDDist method</span>
     <!--l. 42-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">FillZeros</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- replace zeros with small numbers in rows/columns with values</span>
     <!--l. 45-->
<p class="noindent">Reference: [<a
        href="#XSteel1994">Steel</a>,&#x00A0;<a
        href="#XSteel1994">1994</a>]


     <!--l. 47-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-79000C.1"></a>Jukes Cantor Distance</h5>
<a
        id="x1-79001"></a>
<!--l. 49--><p class="noindent">The <span
        class="cmssi-10">Jukes Cantor Distance </span>algorithm takes a Characters block as input and produces a
     Distances block as output. It
     calculates distances under the Jukes-Cantor model. The algorithm has the following options:
     <!--l. 53-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">PropInvariableSites</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- proportion of invariable sites</span>
     <!--l. 55-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SetSiteVarParams</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{fromChars</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;defaultValues}</span></span></span> <span
        class="cmss-8">- set site variation parameters to default values, or to estimations from</span>
     <span
             class="cmss-8">characters</span>
     <!--l. 58-->
<p class="noindent">Reference: [<a
        href="#XJukesCantor1969">Jukes and Cantor</a>,&#x00A0;<a
        href="#XJukesCantor1969">1969</a>]
     <!--l. 60-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-80000C.1"></a>K2P Distance</h5>
<a
        id="x1-80001"></a>
<!--l. 62--><p class="noindent">The <span
        class="cmssi-10">K2P Distance </span>algorithm takes a Characters block as input and produces a Distances block
     as output. It calculates
     distances under the Kimura-2P model. The algorithm has the following options:
     <!--l. 66-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">TsTvRatio</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- ratio of transitions vs transversions</span>
     <!--l. 68-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Gamma</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- alpha value for the Gamma distribution</span>
     <!--l. 70-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">PropInvariableSites</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- proportion of invariable sites</span>
     <!--l. 72-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SetSiteVarParams</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{fromChars</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;defaultValues}</span></span></span> <span
        class="cmss-8">- set site variation parameters to default values, or to estimations from</span>
     <span
             class="cmss-8">characters</span>
     <!--l. 74-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">UseML_Distances</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- use maximum likelihood estimation of distances (rather than exact distances)</span>
     <!--l. 77-->
<p class="noindent">Reference: [<a
        href="#XKimura1980">Kimura</a>,&#x00A0;<a
        href="#XKimura1980">1980</a>]
     <!--l. 79-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-81000C.1"></a>F81 Distance</h5>
<a
        id="x1-81001"></a>
<!--l. 81--><p class="noindent">The <span
        class="cmssi-10">F81 Distance </span>algorithm takes a Characters block as input and produces a Distances block
     as output. It calculates
     distances under the Felsenstein-81 model. The algorithm has the following options:
     <!--l. 85-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">BaseFrequencies</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;doubleArray&#x003E;</span></span></span> <span
        class="cmss-8">- base frequencies (in order ACGT/U)</span>
     <!--l. 87-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SetBaseFrequencies</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{fromChars</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;defaultValues}</span></span></span> <span
        class="cmss-8">- set base frequencies to default values, or to estimations from characters (using</span>
     <span
             class="cmss-8">Capture-recapture for invariable sites)</span>
     <!--l. 89-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">PropInvariableSites</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- proportion of invariable sites</span>
     <!--l. 91-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SetSiteVarParams</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{fromChars</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;defaultValues}</span></span></span> <span
        class="cmss-8">- set site variation parameters to default values, or to estimations from</span>
     <span
             class="cmss-8">characters</span>
     <!--l. 93-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">UseML_Distances</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- use maximum likelihood estimation of distances (rather than exact distances)</span>
     <!--l. 96-->
<p class="noindent">Reference: [<a
        href="#XFelsenstein1981">Felsenstein</a>,&#x00A0;<a
        href="#XFelsenstein1981">1981</a>]


     <!--l. 98-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-82000C.1"></a>HKY85 Distance</h5>
<a
        id="x1-82001"></a>
<!--l. 100--><p class="noindent">The <span
        class="cmssi-10">HKY85 Distance </span>algorithm takes a Characters block as input and produces a Distances
     block as output. It calculates
     distances under the Hasegawa-Kishino-Yano model. The algorithm has the following options:
     <!--l. 104-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">TsTvRatio</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- ratio of transitions vs transversions</span>
     <!--l. 106-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">BaseFrequencies</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;doubleArray&#x003E;</span></span></span> <span
        class="cmss-8">- base frequencies (in order ACGT/U)</span>
     <!--l. 108-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SetBaseFrequencies</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{fromChars</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;defaultValues}</span></span></span> <span
        class="cmss-8">- set base frequencies to default values, or to estimations from characters (using</span>
     <span
             class="cmss-8">Capture-recapture for invariable sites)</span>
     <!--l. 110-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">PropInvariableSites</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- proportion of invariable sites</span>
     <!--l. 112-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SetSiteVarParams</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{fromChars</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;defaultValues}</span></span></span> <span
        class="cmss-8">- set site variation parameters to default values, or to estimations from</span>
     <span
             class="cmss-8">characters</span>
     <!--l. 115-->
<p class="noindent">Reference: [<a
        href="#XHasegawaetal1985">Hasegawa et&#x00A0;al.</a>,&#x00A0;<a
        href="#XHasegawaetal1985">1985</a>]
     <!--l. 117-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-83000C.1"></a>F84 Distance</h5>
<a
        id="x1-83001"></a>
<!--l. 119--><p class="noindent">The <span
        class="cmssi-10">F84 Distance </span>algorithm takes a Characters block as input and produces a Distances block
     as output. It calculates
     distances under the Felsenstein-84 model. The algorithm has the following options:
     <!--l. 123-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">BaseFrequencies</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;doubleArray&#x003E;</span></span></span> <span
        class="cmss-8">- base frequencies (in order ACGT/U)</span>
     <!--l. 125-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SetBaseFrequencies</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{fromChars</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;defaultValues}</span></span></span> <span
        class="cmss-8">- set base frequencies to default values, or to estimations from characters (using</span>
     <span
             class="cmss-8">Capture-recapture for invariable sites)</span>
     <!--l. 127-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">PropInvariableSites</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- proportion of invariable sites</span>
     <!--l. 129-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SetSiteVarParams</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{fromChars</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;defaultValues}</span></span></span> <span
        class="cmss-8">- set site variation parameters to default values, or to estimations from</span>
     <span
             class="cmss-8">characters</span>
     <!--l. 131-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">UseML_Distances</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- use maximum likelihood estimation of distances (rather than exact distances)</span>
     <!--l. 134-->
<p class="noindent">Reference: [<a
        href="#XFelsensteinChurchill1996">Felsenstein and Churchill</a>,&#x00A0;<a
        href="#XFelsensteinChurchill1996">1996</a>]
     <!--l. 136-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-84000C.1"></a>GTR Distance</h5>
<a
        id="x1-84001"></a>
<!--l. 138--><p class="noindent">The <span
        class="cmssi-10">GTR Distance </span>algorithm takes a Characters block as input and produces a Distances block
     as output. It calculates
     distances under the general time-reversible model. The algorithm has the following options:
     <!--l. 142-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">PropInvariableSites</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- proportion of invariable sites</span>
     <!--l. 144-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SetSiteVarParams</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{fromChars</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;defaultValues}</span></span></span> <span
        class="cmss-8">- set site variation parameters to default values, or to estimations from</span>
     <span
             class="cmss-8">characters</span>
     <!--l. 146-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">RateMatrix</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;doubleSquareMatrix&#x003E;</span></span></span> <span
        class="cmss-8">- rate matrix for GTRDistance (in order ACGT/U)</span>
     <!--l. 148-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">UseML_Distances</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- use maximum likelihood estimation of distances (rather than exact distances)</span>
     <!--l. 151-->
<p class="noindent">Reference: [<a
        href="#XTavar1986">Tavar</a>,&#x00A0;<a
        href="#XTavar1986">1986</a>]


     <!--l. 153-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-85000C.1"></a>Protein ML Distance</h5>
<a
        id="x1-85001"></a>
<!--l. 155--><p class="noindent">The <span
        class="cmssi-10">Protein ML Distance </span>algorithm takes a Characters block as input and produces a Distances
     block as output. It
     computes distances for proteins using maximum-likelihood estimation. The algorithm has the following
     options:
     <!--l. 159-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Model</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{cpREV45</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Dayhoff</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;JTT</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;mtMAM</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;mtREV24</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;pmb</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Rhodopsin</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;WAG}</span></span></span> <span
        class="cmss-8">- choose an amino acid substitution</span>
     <span
             class="cmss-8">model</span>
     <!--l. 161-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">PropInvariableSites</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- proportion of invariable sites</span>
     <!--l. 163-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Gamma</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- alpha parameter for gamma distribution. Negative gamma = Equal rates</span>
     <!--l. 166-->
<p class="noindent">Reference: [<a
        href="#XSwoffordetal1996">Swofford et&#x00A0;al.</a>,&#x00A0;<a
        href="#XSwoffordetal1996">1996</a>]
     <!--l. 168-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-86000C.1"></a>Dice Distance</h5>
<a
        id="x1-86001"></a>
<!--l. 170--><p class="noindent">The <span
        class="cmssi-10">Dice Distance </span>algorithm takes a Characters block as input and produces a Distances block
     as output. It computes
     distances using the DiceDistance coefficient distance.
     <!--l. 172-->
<p class="noindent">Reference: [<a
        href="#XDice1945">Dice</a>,&#x00A0;<a
        href="#XDice1945">1945</a>]
     <!--l. 174-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-87000C.1"></a>Jaccard Distance</h5>
<a
        id="x1-87001"></a>
<!--l. 176--><p class="noindent">The <span
        class="cmssi-10">Jaccard Distance </span>algorithm takes a Characters block as input and produces a Distances
     block as output. It computes
     distances based on the JaccardDistance index.
     <!--l. 178-->
<p class="noindent">Reference: [<a
        href="#XJaccard1901">Jaccard</a>,&#x00A0;<a
        href="#XJaccard1901">1901</a>]
     <!--l. 180-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-88000C.1"></a>Gene Content Distance</h5>
<a
        id="x1-88001"></a>
<!--l. 182--><p class="noindent">The <span
        class="cmssi-10">Gene Content Distance </span>algorithm takes a Characters block as input and produces a
     Distances block as output. It
     computes distances based on the presence/absence of genes.
     <!--l. 184-->
<p class="noindent">Reference: [<a
        href="#XHusonSteel2004">Huson and Steel</a>,&#x00A0;<a
        href="#XHusonSteel2004">2004</a>]


     <!--l. 186-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-89000C.1"></a>Gene Sharing Distance</h5>
<a
        id="x1-89001"></a>
<!--l. 188--><p class="noindent">The <span
        class="cmssi-10">Gene Sharing Distance </span>algorithm takes a Characters block as input and produces a
     Distances block as output. It
     computes distances using the gene-sharing distance.
     <!--l. 190-->
<p class="noindent">Reference: [<a
        href="#XSneletal1997">Snel et&#x00A0;al.</a>,&#x00A0;<a
        href="#XSneletal1997">1997</a>]
     <!--l. 192-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-90000C.1"></a>Upholt Restriction Distance</h5>
<a
        id="x1-90001"></a>
<!--l. 194--><p class="noindent">The <span
        class="cmssi-10">Upholt Restriction Distance </span>algorithm takes a Characters block as input and produces a
     Distances block as output. It
     calculates distances for restriction data.
     <!--l. 196-->
<p class="noindent">Reference: [<a
        href="#XUpholt1977">Upholt</a>,&#x00A0;<a
        href="#XUpholt1977">1977</a>]
     <!--l. 198-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-91000C.1"></a>Nei Li Restriction Distance</h5>
<a
        id="x1-91001"></a>
<!--l. 200--><p class="noindent">The <span
        class="cmssi-10">Nei Li Restriction Distance </span>algorithm takes a Characters block as input and produces a
     Distances block as output. It
     calculates distances for restriction data.
     <!--l. 202-->
<p class="noindent">Reference: [<a
        href="#XNeiLi1979">Nei and Li</a>,&#x00A0;<a
        href="#XNeiLi1979">1979</a>]
     <!--l. 204-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-92000C.1"></a>Base Freq Distance</h5>
<a
        id="x1-92001"></a>
<!--l. 206--><p class="noindent">The <span
        class="cmssi-10">Base Freq Distance </span>algorithm takes a Characters block as input and produces a Distances
     block as output. It
     calculates distances from differences in the base composition.
     <!--l. 208-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-93000C.1"></a>Binary To Splits</h5>
<a
        id="x1-93001"></a>
<!--l. 210--><p class="noindent">The <span
        class="cmssi-10">Binary To Splits </span>algorithm takes a Characters block as input and produces a Splits block
     as output. It converts binary
     characters directly into splits. The algorithm has the following options:
     <!--l. 214-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MinSplitWeight</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- minimum split weight threshold</span>
     <!--l. 216-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">HighDimensionFilter</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- activate high-dimensional filter to avoid exponential graph size</span>
     <!--l. 218-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">AddAllTrivial</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- ensure all trival splits are present</span>


     <!--l. 221-->
<p class="noindent">Reference: [<a
        href="#XHusonetal2012">Huson et&#x00A0;al.</a>,&#x00A0;<a
        href="#XHusonetal2012">2012</a>]
     <!--l. 223-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-94000C.1"></a>DNA To Splits</h5>
<a
        id="x1-94001"></a>
<!--l. 225--><p class="noindent">The <span
        class="cmssi-10">DNA To Splits </span>algorithm takes a Characters block as input and produces a Splits block as
     output. It converts DNA
     characters directly into splits. The algorithm has the following options:
     <!--l. 229-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Method</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{MajorityState</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;RYAlphabet}</span></span></span> <span
        class="cmss-8">- use either majority-state-vs-others or RY alphabet</span>
     <!--l. 231-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MinSplitWeight</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- minimum split weight threshold</span>
     <!--l. 233-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">HighDimensionFilter</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- activate high-dimensional filter to avoid exponential graph size</span>
     <!--l. 236-->
<p class="noindent">Reference: [<a
        href="#XHusonetal2012">Huson et&#x00A0;al.</a>,&#x00A0;<a
        href="#XHusonetal2012">2012</a>]
     <!--l. 238-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-95000C.1"></a>Median Joining</h5>
<a
        id="x1-95001"></a>
<!--l. 240--><p class="noindent">The <span
        class="cmssi-10">Median Joining </span>algorithm takes a Characters block as input and produces a Network block
     as output. It computes a
     haplotype network using the median-joining method. The algorithm has the following options:
     <!--l. 244-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Epsilon</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- balances accuracy (smaller value) and efficiency (larger value)</span>
     <!--l. 247-->
<p class="noindent">Reference: [<a
        href="#XBandeltetal1999">Bandelt et&#x00A0;al.</a>,&#x00A0;<a
        href="#XBandeltetal1999">1999</a>]
     <!--l. 249-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-96000C.1"></a>Parsimony Splits</h5>
<a
        id="x1-96001"></a>
<!--l. 251--><p class="noindent">The <span
        class="cmssi-10">Parsimony Splits </span>algorithm takes a Characters block as input and produces a Splits block
     as output. It computes
     weakly-compatible splits directly from DNA characters.
     <!--l. 253-->
<p class="noindent">Reference: [<a
        href="#XBandeltDress1992">Bandelt and Dress</a>,&#x00A0;<a
        href="#XBandeltDress1992">1992</a>]
     <!--l. 255-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-97000C.1"></a>Characters Filter</h5>
<a
        id="x1-97001"></a>
<!--l. 257--><p class="noindent">The <span
        class="cmssi-10">Characters Filter </span>algorithm takes a Characters block as input and produces a Characters
     block as output. It filter
     characters. The algorithm has the following options:
     <!--l. 261-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ExcludeGapSites</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- exclude all sites that contain a gap</span>
     <!--l. 263-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ExcludeParsimonyUninformativeSites</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- exclude all sites that are parsimony uninformative</span>
     <!--l. 265-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ExcludeConstantSites</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- exclude all sites that are constant</span>


     <!--l. 267-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ExcludeFirstCodonPosition</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- exclude first and then every third site</span>
     <!--l. 269-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ExcludeSecondCodonPosition</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- exclude second and then every third site</span>
     <!--l. 271-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ExcludeThirdCodonPosition</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- exclude third and then every third site</span>
     <!--l. 274-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-98000C.1"></a>External Program</h5>
<a
        id="x1-98001"></a>
<!--l. 276--><p class="noindent">The <span
        class="cmssi-10">External Program </span>algorithm takes a Characters block as input and produces a Trees block
     as output. It runs an
     external program. The algorithm has the following options:
     <!--l. 280-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Name</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;String&#x003E;</span></span></span> <span
        class="cmss-8">- specify a name for this calculation</span>
     <!--l. 282-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ProgramCall</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;String&#x003E;</span></span></span> <span
        class="cmss-8">- specification of external program: replace &#8217;path-to-program&#8217; by path to program and</span>
     <!--l. 284-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">use</span><span
        class="cmtt-8">&#x00A0;&#8217;%i&#8217;</span><span
        class="cmtt-8">&#x00A0;and</span><span
        class="cmtt-8">&#x00A0;&#8217;%o&#8217;</span><span
        class="cmtt-8">&#x00A0;as</span><span
        class="cmtt-8">&#x00A0;place</span></span></span> <span
        class="cmss-8">- holders for the program&#8217;s input and output files</span>
     <!--l. 286-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">CharactersFormat</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Phylip</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Nexus</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;FastA}</span></span></span> <span
        class="cmss-8">- specify the format to write out the current data in</span>
     <!--l. 289-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-99000C.1"></a>Estimate Invariable Sites</h5>
<a
        id="x1-99001"></a>
<!--l. 291--><p class="noindent">The <span
        class="cmssi-10">Estimate Invariable Sites </span>algorithm takes a Characters block as input and produces a
     Report block as output. It
     estimates the proportion of invariant sites using capture-recapture.
     <!--l. 293-->
<p class="noindent">Reference: [<a
        href="#XSteeletal2000">Steel et&#x00A0;al.</a>,&#x00A0;<a
        href="#XSteeletal2000">2000</a>]
     <!--l. 295-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-100000C.1"></a>Phi Test</h5>
<a
        id="x1-100001"></a>
<!--l. 297--><p class="noindent">The <span
        class="cmssi-10">Phi Test </span>algorithm takes a Characters block as input and produces a Report block as
     output. It performs a statistical
     test for detecting the presence of recombination.
     <!--l. 299-->
<p class="noindent">Reference: [<a
        href="#XBruenetal2006">Bruen et&#x00A0;al.</a>,&#x00A0;<a
        href="#XBruenetal2006">2006</a>]
     <!--l. 301-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-101000C.1"></a>Tajima D</h5>
<a
        id="x1-101001"></a>
<!--l. 303--><p class="noindent">The <span
        class="cmssi-10">Tajima D </span>algorithm takes a Characters block as input and produces a Report block as
     output. It performs
     Tajima&#8217;s D test to determine whether a DNA sequence is evolving neutrally. The algorithm has the following
     options:
     <!--l. 307-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ExcludeGapSites</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- exclude gapped sites from calculation.</span>


     <!--l. 310-->
<p class="noindent">Reference: [<a
        href="#XTajima1989">Tajima</a>,&#x00A0;<a
        href="#XTajima1989">1989</a>]
     <!--l. 312-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">C.2   </span> <a
        id="x1-102000C.2"></a>Algorithms on a Distances Block</h3>
<!--l. 314--><p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-103000C.2"></a>Neighbor Joining</h5>
<a
        id="x1-103001"></a>
<!--l. 316--><p class="noindent">The <span
        class="cmssi-10">Neighbor Joining </span>algorithm takes a Distances block as input and produces a Trees block
     as output. It computes an
     unrooted phylogenetic tree using the neighbor-joining method.
     <!--l. 318-->
<p class="noindent">Reference: [<a
        href="#XSaitouNei1987">Saitou and Nei</a>,&#x00A0;<a
        href="#XSaitouNei1987">1987</a>]
     <!--l. 320-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-104000C.2"></a>Bio NJ</h5>
<a
        id="x1-104001"></a>
<!--l. 322--><p class="noindent">The <span
        class="cmssi-10">Bio NJ </span>algorithm takes a Distances block as input and produces a Trees block as output.
     It computes an unrooted
     phylogenetic tree using the Bio-NJ method.
     <!--l. 324-->
<p class="noindent">Reference: [<a
        href="#XGascuel1997">Gascuel</a>,&#x00A0;<a
        href="#XGascuel1997">1997</a>]
     <!--l. 326-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-105000C.2"></a>UPGMA</h5>
<a
        id="x1-105001"></a>
<!--l. 328--><p class="noindent">The <span
        class="cmssi-10">UPGMA </span>algorithm takes a Distances block as input and produces a Trees block as output.
     It computes a rooted
     phylogenetic tree using the UPGMA method.
     <!--l. 330-->
<p class="noindent">Reference: [<a
        href="#XSokalMichener1958">Sokal and Michener</a>,&#x00A0;<a
        href="#XSokalMichener1958">1958</a>]
     <!--l. 332-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-106000C.2"></a>Neighbor Net</h5>
<a
        id="x1-106001"></a>
<!--l. 334--><p class="noindent">The <span
        class="cmssi-10">Neighbor Net </span>algorithm takes a Distances block as input and produces a Splits block as
     output. It computes a set of
     cyclic splits using the neighbor-net method. The algorithm has the following options:
     <!--l. 338-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">InferenceAlgorithm</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{GradientProjection</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;ActiveSet</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;APGD</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;SplitsTree4}</span></span></span> <span
        class="cmss-8">- the inference algorithm to be</span>
     <span
             class="cmss-8">used</span>


     <!--l. 341-->
<p class="noindent">References: [<a
        href="#XBryantMoulton2004">Bryant and Moulton</a>,&#x00A0;<a
        href="#XBryantMoulton2004">2004</a>,&#x00A0;<a
        href="#XBryantHuson2023">Bryant and Huson</a>,&#x00A0;<a
        href="#XBryantHuson2023">2023</a>]
     <!--l. 343-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-107000C.2"></a>Split Decomposition</h5>
<a
        id="x1-107001"></a>
<!--l. 345--><p class="noindent">The <span
        class="cmssi-10">Split Decomposition </span>algorithm takes a Distances block as input and produces a Splits
     block as output. It computes a
     set of weakly-compatible splits using the split-decomposition method.
     <!--l. 347-->
<p class="noindent">Reference: [<a
        href="#XBandeltDress1992">Bandelt and Dress</a>,&#x00A0;<a
        href="#XBandeltDress1992">1992</a>]
     <!--l. 349-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-108000C.2"></a>Buneman Tree</h5>
<a
        id="x1-108001"></a>
<!--l. 351--><p class="noindent">The <span
        class="cmssi-10">Buneman Tree </span>algorithm takes a Distances block as input and produces a Splits block as
     output. It computes a set of
     compatible splits using the Buneman tree method.
     <!--l. 353-->
<p class="noindent">Reference: [<a
        href="#XBandeltDress1992">Bandelt and Dress</a>,&#x00A0;<a
        href="#XBandeltDress1992">1992</a>]
     <!--l. 355-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-109000C.2"></a>Min Spanning Network</h5>
<a
        id="x1-109001"></a>
<!--l. 357--><p class="noindent">The <span
        class="cmssi-10">Min Spanning Network </span>algorithm takes a Distances block as input and produces a Network
     block as output. It
     computes a minimum spanning network. The algorithm has the following options:
     <!--l. 361-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Epsilon</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- weighted genetic distance measure. Low: MedianJoining, High: full median network</span>
     <!--l. 363-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MinSpanningTree</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- calculate minimum spanning tree</span>
     <!--l. 366-->
<p class="noindent">Reference: [<a
        href="#XExcoffierSmouse1994">Excoffier and Smouse</a>,&#x00A0;<a
        href="#XExcoffierSmouse1994">1994</a>]
     <!--l. 368-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-110000C.2"></a>Min Spanning Tree</h5>
<a
        id="x1-110001"></a>
<!--l. 370--><p class="noindent">The <span
        class="cmssi-10">Min Spanning Tree </span>algorithm takes a Distances block as input and produces a Trees block
     as output. It computes a
     minimum spanning tree.
     <!--l. 372-->
<p class="noindent">Reference: [<a
        href="#XExcoffierSmouse1994">Excoffier and Smouse</a>,&#x00A0;<a
        href="#XExcoffierSmouse1994">1994</a>]


     <!--l. 374-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-111000C.2"></a>PCOA</h5>
<a
        id="x1-111001"></a>
<!--l. 376--><p class="noindent">The <span
        class="cmssi-10">PCOA </span>algorithm takes a Distances block as input and produces a Network block as output.
     It performs principal
     coordinates analysis. The algorithm has the following options:
     <!--l. 380-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">FirstCoordinate</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- choose principal component for the x Axis</span>
     <!--l. 382-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SecondCoordinate</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- choose principal component for the y Axis</span>
     <!--l. 385-->
<p class="noindent">Reference: [<a
        href="#XGower1966">Gower</a>,&#x00A0;<a
        href="#XGower1966">1966</a>]
     <!--l. 387-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-112000C.2"></a>Delta Score</h5>
<a
        id="x1-112001"></a>
<!--l. 389--><p class="noindent">The <span
        class="cmssi-10">Delta Score </span>algorithm takes a Distances block as input and produces a Report block as
     output. It calculates the delta
     score.
     <!--l. 391-->
<p class="noindent">Reference: [<a
        href="#XHollandetal2002">Holland et&#x00A0;al.</a>,&#x00A0;<a
        href="#XHollandetal2002">2002</a>]
     <!--l. 393-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">C.3   </span> <a
        id="x1-113000C.3"></a>Algorithms on a Splits Block</h3>
<!--l. 395--><p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-114000C.3"></a>Bootstrap Splits</h5>
<a
        id="x1-114001"></a>
<!--l. 397--><p class="noindent">The <span
        class="cmssi-10">Bootstrap Splits </span>algorithm takes a Splits block as input and produces a Splits block as
     output. It performs
     bootstrapping on splits. The algorithm has the following options:
     <!--l. 401-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Replicates</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- number of bootstrap replicates</span>
     <!--l. 403-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MinPercent</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- minimum percentage support for a split to be included</span>
     <!--l. 405-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ShowAllSplits</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- show all bootstrap splits, not just the original splits</span>
     <!--l. 407-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">RandomSeed</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- if non-zero, is used as seed for random number generator</span>
     <!--l. 409-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">HighDimensionFilter</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- heuristically remove splits causing high-dimensional network</span>
     <!--l. 412-->
<p class="noindent">Reference: [<a
        href="#XFelsenstein1985">Felsenstein</a>,&#x00A0;<a
        href="#XFelsenstein1985">1985</a>]


     <!--l. 414-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-115000C.3"></a>Greedy Tree</h5>
<a
        id="x1-115001"></a>
<!--l. 416--><p class="noindent">The <span
        class="cmssi-10">Greedy Tree </span>algorithm takes a Splits block as input and produces a Trees block as
     output. It produces a phylogenetic
     tree based on greedily selecting a compatible set of splits.
     <!--l. 418-->
<p class="noindent">Reference: [<a
        href="#XHusonetal2012">Huson et&#x00A0;al.</a>,&#x00A0;<a
        href="#XHusonetal2012">2012</a>]
     <!--l. 420-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-116000C.3"></a>Dimension Filter</h5>
<a
        id="x1-116001"></a>
<!--l. 422--><p class="noindent">The <span
        class="cmssi-10">Dimension Filter </span>algorithm takes a Splits block as input and produces a Splits block as
     output. It
     heuristically remove splits that lead to high-dimensional boxes in a split network. The algorithm has the following
     options:
     <!--l. 426-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MaxDimension</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- heuristically remove splits that create configurations of a higher dimension than this threshold</span>
     <!--l. 429-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-117000C.3"></a>Show Splits</h5>
<a
        id="x1-117001"></a>
<!--l. 431--><p class="noindent">The <span
        class="cmssi-10">Show Splits </span>algorithm takes a Splits block as input and produces a View block as output.
     It provides interactive
     visualizations of split networks. The algorithm has the following options:
     <!--l. 435-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">View</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{SplitNetwork}</span></span></span> <span
        class="cmss-8">- the type of splits viewer to use</span>
     <!--l. 438-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-118000C.3"></a>Splits Filter</h5>
<a
        id="x1-118001"></a>
<!--l. 440--><p class="noindent">The <span
        class="cmssi-10">Splits Filter </span>algorithm takes a Splits block as input and produces a Splits block as
     output. It filter splits. The
     algorithm has the following options:
     <!--l. 444-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">WeightThreshold</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Float&#x003E;</span></span></span> <span
        class="cmss-8">- set minimum split weight threshold</span>
     <!--l. 446-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ConfidenceThreshold</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Float&#x003E;</span></span></span> <span
        class="cmss-8">- set the minimum split confidence threshold</span>
     <!--l. 448-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MaximumDimension</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- set maximum dimension threshold (necessary to avoid computational overload)</span>
     <!--l. 450-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">FilterAlgorithm</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{None</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;GreedyCompatible</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;GreedyCircular</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;GreedyWeaklyCompatible</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;BlobTree}</span></span></span> <span
        class="cmss-8">- set the filter</span>
     <span
             class="cmss-8">algorithm</span>
     <!--l. 452-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">RecomputeCycle</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- recompute circular ordering</span>


     <!--l. 455-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-119000C.3"></a>Weights Slider</h5>
<a
        id="x1-119001"></a>
<!--l. 457--><p class="noindent">The <span
        class="cmssi-10">Weights Slider </span>algorithm takes a Splits block as input and produces a Splits block as
     output. It allows one to
     interactively filter splits by their weight. The algorithm has the following options:
     <!--l. 461-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">WeightThreshold</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- set minimum split weight threshold</span>
     <!--l. 464-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-120000C.3"></a>Incompatibility Score</h5>
<a
        id="x1-120001"></a>
<!--l. 466--><p class="noindent">The <span
        class="cmssi-10">Incompatibility Score </span>algorithm takes a Splits block as input and produces a Report
     block as output. It calculates an
     incompatibility score on splits.
     <!--l. 468-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-121000C.3"></a>Phylogenetic Diversity</h5>
<a
        id="x1-121001"></a>
<!--l. 470--><p class="noindent">The <span
        class="cmssi-10">Phylogenetic Diversity </span>algorithm takes a Splits block as input and produces a Report
     block as output. It calculates
     the phylogenetic diversity for selected taxa.
     <!--l. 472-->
<p class="noindent">Reference: [<a
        href="#XVolkmannetal2014">Volkmann et&#x00A0;al.</a>,&#x00A0;<a
        href="#XVolkmannetal2014">2014</a>]
     <!--l. 474-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-122000C.3"></a>Shapley Values</h5>
<a
        id="x1-122001"></a>
<!--l. 476--><p class="noindent">The <span
        class="cmssi-10">Shapley Values </span>algorithm takes a Splits block as input and produces a Report block as
     output. It calculates Shapley
     values on splits.
     <!--l. 478-->
<p class="noindent">Reference: [<a
        href="#XVolkmannetal2014">Volkmann et&#x00A0;al.</a>,&#x00A0;<a
        href="#XVolkmannetal2014">2014</a>]
     <!--l. 480-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">C.4   </span> <a
        id="x1-123000C.4"></a>Algorithms on a Trees Block</h3>


<!--l. 482--><p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-124000C.4"></a>Autumn Algorithm</h5>
<a
        id="x1-124001"></a>
<!--l. 484--><p class="noindent">The <span
        class="cmssi-10">Autumn Algorithm </span>algorithm takes a Trees block as input and produces a Trees block as
     output. It
     computes all minimum hybridization networks using the Autumn algorithm The algorithm has the following
     options:
     <!--l. 488-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">FirstTree</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- index of the first tree</span>
     <!--l. 490-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SecondTree</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- index of the second tree</span>
     <!--l. 492-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">OnlyOneNetwork</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- report only one network</span>
     <!--l. 494-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">RerootToMinimize</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- reroot input trees to minimize hybridization number</span>
     <!--l. 497-->
<p class="noindent">Reference: [<a
        href="#XHusonLinz2018">Huson and Linz</a>,&#x00A0;<a
        href="#XHusonLinz2018">2018</a>]
     <!--l. 499-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-125000C.4"></a>Phylo Fusion</h5>
<a
        id="x1-125001"></a>
<!--l. 501--><p class="noindent">The <span
        class="cmssi-10">Phylo Fusion </span>algorithm takes a Trees block as input and produces a Trees block as
     output. It combines multiple
     rooted phylogenetic trees into a rooted network. The algorithm has the following options:
     <!--l. 505-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">OnlyOneNetwork</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- report only one network</span>
     <!--l. 507-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MutualRefinement</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- mutually refine input trees</span>
     <!--l. 509-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">NormalizeEdgeWeights</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- normalize input edge weights</span>
     <!--l. 511-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SearchHeuristic</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Thorough</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Medium</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Fast}</span></span></span> <span
        class="cmss-8">- fast, Medium, or Thorough search</span>
     <!--l. 513-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">GroupNonSeparated</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- improve performance by grouping taxa that are not separated by a non-trivial edge</span>
     <!--l. 515-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">CladeReduction</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- improve performance using clade reduction</span>
     <!--l. 518-->
<p class="noindent">References: [<a
        href="#XZhangetal2023">Zhang et&#x00A0;al.</a>,&#x00A0;<a
        href="#XZhangetal2023">2023</a>,&#x00A0;<a
        href="#XZhangetal2024">2024</a>]
     <!--l. 520-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-126000C.4"></a>Average Consensus</h5>
<a
        id="x1-126001"></a>
<!--l. 522--><p class="noindent">The <span
        class="cmssi-10">Average Consensus </span>algorithm takes a Trees block as input and produces a Splits block as
     output. It calculates
     average consensus tree.
     <!--l. 524-->
<p class="noindent">Reference: [<a
        href="#XLapointeCucumel1997">Lapointe and Cucumel</a>,&#x00A0;<a
        href="#XLapointeCucumel1997">1997</a>]
     <!--l. 526-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-127000C.4"></a>Blob Tree</h5>
<a
        id="x1-127001"></a>


<!--l. 528--><p class="noindent">The <span
        class="cmssi-10">Blob Tree </span>algorithm takes a Trees block as input and produces a Trees block as output.
     It extract the blob tree from
     a rooted network The algorithm has the following options:
     <!--l. 532-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SeparateBlobs</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- for any blob that shares its top node with some other blob, insert an edge above it to keep blobs</span>
     <span
             class="cmss-8">separate</span>
     <!--l. 535-->
<p class="noindent">Reference: [<a
        href="#XHusonetal2012">Huson et&#x00A0;al.</a>,&#x00A0;<a
        href="#XHusonetal2012">2012</a>]
     <!--l. 537-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-128000C.4"></a>Bootstrap Tree Splits</h5>
<a
        id="x1-128001"></a>
<!--l. 539--><p class="noindent">The <span
        class="cmssi-10">Bootstrap Tree Splits </span>algorithm takes a Trees block as input and produces a Splits block
     as output. It performs
     bootstrapping on trees. The algorithm has the following options:
     <!--l. 543-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Replicates</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- number of bootstrap replicates</span>
     <!--l. 545-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MinPercent</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- minimum percentage support for a split to be included</span>
     <!--l. 547-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ShowAllSplits</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- show all bootstrap splits, not just the original splits</span>
     <!--l. 549-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">RandomSeed</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- if non-zero, is used as seed for random number generator</span>
     <!--l. 551-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">HighDimensionFilter</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- heuristically remove splits causing high-dimensional network</span>
     <!--l. 554-->
<p class="noindent">Reference: [<a
        href="#XFelsenstein1985">Felsenstein</a>,&#x00A0;<a
        href="#XFelsenstein1985">1985</a>]
     <!--l. 556-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-129000C.4"></a>Bootstrap Tree</h5>
<a
        id="x1-129001"></a>
<!--l. 558--><p class="noindent">The <span
        class="cmssi-10">Bootstrap Tree </span>algorithm takes a Trees block as input and produces a Trees block as
     output. It performs
     bootstrapping on trees. The algorithm has the following options:
     <!--l. 562-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Replicates</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- number of bootstrap replicates</span>
     <!--l. 564-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">TransferBootstrap</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- use transform bootstrapping (TBE), less susceptible to rouge taxa</span>
     <!--l. 566-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MinPercent</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- minimum percentage support for a branch to be included</span>
     <!--l. 568-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">RandomSeed</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- if non-zero, is used as seed for random number generator</span>
     <!--l. 571-->
<p class="noindent">Reference: [<a
        href="#XFelsenstein1985">Felsenstein</a>,&#x00A0;<a
        href="#XFelsenstein1985">1985</a>]
     <!--l. 573-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-130000C.4"></a>Cluster Network</h5>
<a
        id="x1-130001"></a>
<!--l. 575--><p class="noindent">The <span
        class="cmssi-10">Cluster Network </span>algorithm takes a Trees block as input and produces a Trees block as
     output. It computes the
     cluster network that contains all input trees (in the hardwired sense). The algorithm has the following
     options:
     <!--l. 579-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">EdgeWeights</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Mean</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Count</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Sum</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Uniform}</span></span></span> <span
        class="cmss-8">- compute edge weights</span>


     <!--l. 581-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ThresholdPercent</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- minimum percentage of trees that a cluster must appear in</span>
     <!--l. 584-->
<p class="noindent">Reference: [<a
        href="#XHusonRupp2008">Huson and Rupp</a>,&#x00A0;<a
        href="#XHusonRupp2008">2008</a>]
     <!--l. 586-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-131000C.4"></a>Consensus Tree</h5>
<a
        id="x1-131001"></a>
<!--l. 588--><p class="noindent">The <span
        class="cmssi-10">Consensus Tree </span>algorithm takes a Trees block as input and produces a Trees block as
     output. It provides several
     methods for computing a consensus tree. The algorithm has the following options:
     <!--l. 592-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Consensus</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Majority</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Greedy</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Strict}</span></span></span> <span
        class="cmss-8">- consensus method to use</span>
     <!--l. 595-->
<p class="noindent">Reference: [<a
        href="#XBryant2001">Bryant</a>,&#x00A0;<a
        href="#XBryant2001">2001</a>]
     <!--l. 597-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-132000C.4"></a>Consensus Network</h5>
<a
        id="x1-132001"></a>
<!--l. 599--><p class="noindent">The <span
        class="cmssi-10">Consensus Network </span>algorithm takes a Trees block as input and produces a Splits block as
     output. It computes the
     consensus network. The algorithm has the following options:
     <!--l. 603-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">EdgeWeights</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Mean</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;TreeSizeWeightedMean</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Median</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Count</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Sum</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Uniform</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;TreeNormalizedSum}</span></span></span> <span
        class="cmss-8">- how to calculate edge</span>
     <span
             class="cmss-8">weights in resulting network</span>
     <!--l. 605-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ThresholdPercent</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- threshold for percent of input trees that split has to occur in for it to appear in the output</span>
     <!--l. 607-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">HighDimensionFilter</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- heuristically remove splits causing high-dimensional consensus network</span>
     <!--l. 610-->
<p class="noindent">Reference: [<a
        href="#XHollandetal2004">Holland et&#x00A0;al.</a>,&#x00A0;<a
        href="#XHollandetal2004">2004</a>]
     <!--l. 612-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-133000C.4"></a>Consensus Outline</h5>
<a
        id="x1-133001"></a>
<!--l. 614--><p class="noindent">The <span
        class="cmssi-10">Consensus Outline </span>algorithm takes a Trees block as input and produces a Splits block as
     output. It computes the
     consensus outline. The algorithm has the following options:
     <!--l. 618-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">EdgeWeights</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Mean</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;TreeSizeWeightedMean</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Median</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Count</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Sum</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Uniform</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;TreeNormalizedSum}</span></span></span> <span
        class="cmss-8">- how to calculate edge</span>
     <span
             class="cmss-8">weights in resulting network</span>
     <!--l. 620-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ThresholdPercent</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- threshold for percent of input trees that split has to occur in for it to appear in the output</span>
     <!--l. 623-->
<p class="noindent">Reference: [<a
        href="#XHusonCetinkaya2023">Huson and Cetinkaya</a>,&#x00A0;<a
        href="#XHusonCetinkaya2023">2023</a>]


     <!--l. 625-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-134000C.4"></a>Consensus Splits</h5>
<a
        id="x1-134001"></a>
<!--l. 627--><p class="noindent">The <span
        class="cmssi-10">Consensus Splits </span>algorithm takes a Trees block as input and produces a Splits block as
     output. It provides several
     consensus methods. The algorithm has the following options:
     <!--l. 631-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Consensus</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Strict</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Majority</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;GreedyCompatible</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;ConsensusOutline</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;GreedyWeaklyCompatible</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;ConsensusNetwork}</span></span></span> <span
        class="cmss-8">-</span>
     <span
             class="cmss-8">consensus method</span>
     <!--l. 633-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">EdgeWeights</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Mean</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;TreeSizeWeightedMean</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Median</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Count</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Sum</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Uniform</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;TreeNormalizedSum}</span></span></span> <span
        class="cmss-8">- how to calculate edge</span>
     <span
             class="cmss-8">weights in resulting network</span>
     <!--l. 635-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ThresholdPercent</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- threshold for percent of input trees that split has to occur in for it to appear in the output</span>
     <!--l. 637-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">HighDimensionFilter</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- heuristically remove splits causing high-dimensional consensus network</span>
     <!--l. 640-->
<p class="noindent">Reference: [<a
        href="#XHusonetal2012">Huson et&#x00A0;al.</a>,&#x00A0;<a
        href="#XHusonetal2012">2012</a>]
     <!--l. 642-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-135000C.4"></a>Filtered Super Network</h5>
<a
        id="x1-135001"></a>
<!--l. 644--><p class="noindent">The <span
        class="cmssi-10">Filtered Super Network </span>algorithm takes a Trees block as input and produces a Splits
     block as output. It computes a
     super network using the Z-closure method. The algorithm has the following options:
     <!--l. 648-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MinNumberTrees</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- set the min number trees</span>
     <!--l. 650-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MaxDistortionScore</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- set the max distortion score</span>
     <!--l. 652-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">UseTotalScore</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- set the use total score</span>
     <!--l. 655-->
<p class="noindent">Reference: [<a
        href="#XWhitfieldetal2008">Whitfield et&#x00A0;al.</a>,&#x00A0;<a
        href="#XWhitfieldetal2008">2008</a>]
     <!--l. 657-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-136000C.4"></a>LSA Tree</h5>
<a
        id="x1-136001"></a>
<!--l. 659--><p class="noindent">The <span
        class="cmssi-10">LSA Tree </span>algorithm takes a Trees block as input and produces a Trees block as output. It
     extract the LSA tree from
     a rooted network.
     <!--l. 661-->
<p class="noindent">Reference: [<a
        href="#XHusonetal2012">Huson et&#x00A0;al.</a>,&#x00A0;<a
        href="#XHusonetal2012">2012</a>]
     <!--l. 663-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-137000C.4"></a>Normalize Rooted Networks</h5>
<a
        id="x1-137001"></a>
<!--l. 665--><p class="noindent">The <span
        class="cmssi-10">Normalize Rooted Networks </span>algorithm takes a Trees block as input and produces a Trees
     block as output. It the
     Normalize Rooted Networks algorithm
     <!--l. 667-->
<p class="noindent">Reference: [<a
        href="#XFrancisetal2021">Francis et&#x00A0;al.</a>,&#x00A0;<a
        href="#XFrancisetal2021">2021</a>]


     <!--l. 669-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-138000C.4"></a>Rooted Consensus Tree</h5>
<a
        id="x1-138001"></a>
<!--l. 671--><p class="noindent">The <span
        class="cmssi-10">Rooted Consensus Tree </span>algorithm takes a Trees block as input and produces a Trees block
     as output. It provides
     several methods for computing a rooted consensus tree. The algorithm has the following options:
     <!--l. 675-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Consensus</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Majority</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Strict</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Greedy}</span></span></span> <span
        class="cmss-8">- consensus method to use</span>
     <!--l. 678-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-139000C.4"></a>Reroot Or Reorder Trees</h5>
<a
        id="x1-139001"></a>
<!--l. 680--><p class="noindent">The <span
        class="cmssi-10">Reroot Or Reorder Trees </span>algorithm takes a Trees block as input and produces a Trees
     block as output. It reroot or
     reorder all trees. The algorithm has the following options:
     <!--l. 684-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">RootBy</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Off</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;MidPoint</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;OutGroup}</span></span></span> <span
        class="cmss-8">- determine how to reroot</span>
     <!--l. 686-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">RearrangeBy</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Off</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;RotateChildren</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;RotateSubTrees</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;ReverseChildren</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;ReverseSubTrees}</span></span></span> <span
        class="cmss-8">- determine how to</span>
     <span
             class="cmss-8">rearrange</span>
     <!--l. 688-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Reorder</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Off</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;ByTaxa</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Lexicographically</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;ReverseOrder</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;LadderizedUp</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;LadderizedDown</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;LadderizedRandom</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Stabilize}</span></span></span> <span
        class="cmss-8">-</span>
     <span
             class="cmss-8">determine how to reorder</span>
     <!--l. 690-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Rescale</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- rescale each tree to total length of 100</span>
     <!--l. 693-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-140000C.4"></a>Show Trees</h5>
<a
        id="x1-140001"></a>
<!--l. 695--><p class="noindent">The <span
        class="cmssi-10">Show Trees </span>algorithm takes a Trees block as input and produces a View block as output.
     It provides several types of
     interactive visualizations of trees. The algorithm has the following options:
     <!--l. 699-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">View</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{TreeView</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;TreePages</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Tanglegram</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;DensiTree}</span></span></span> <span
        class="cmss-8">- the type of viewer to use</span>
     <!--l. 702-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-141000C.4"></a>Super Network</h5>
<a
        id="x1-141001"></a>
<!--l. 704--><p class="noindent">The <span
        class="cmssi-10">Super Network </span>algorithm takes a Trees block as input and produces a Splits block as
     output. It computes a super
     network using the Z-closure method. The algorithm has the following options:
     <!--l. 708-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">EdgeWeights</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{AverageRelative</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Mean</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;TreeSizeWeightedMean</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Sum</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Min</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;None}</span></span></span> <span
        class="cmss-8">- determine how to calculate edge weights in</span>
     <span
             class="cmss-8">resulting network</span>
     <!--l. 710-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SuperTree</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- enforce the strong induction property, which results in a super tree</span>
     <!--l. 712-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">NumberOfRuns</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- number of runs using random permutations of the input splits</span>
     <!--l. 714-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ApplyRefineHeuristic</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- apply a simple refinement heuristic</span>
     <!--l. 716-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Seed</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- set seed used for random permutations</span>
     <!--l. 718-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">HighDimensionFilter</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- heuristically remove splits causing high-dimensional network</span>


     <!--l. 721-->
<p class="noindent">Reference: [<a
        href="#XHusonetal2004">Huson et&#x00A0;al.</a>,&#x00A0;<a
        href="#XHusonetal2004">2004</a>]
     <!--l. 723-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-142000C.4"></a>Trees Filter</h5>
<a
        id="x1-142001"></a>
<!--l. 725--><p class="noindent">The <span
        class="cmssi-10">Trees Filter </span>algorithm takes a Trees block as input and produces a Trees block as
     output. It filter
     trees.
     <!--l. 727-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-143000C.4"></a>Trees Filter More</h5>
<a
        id="x1-143001"></a>
<!--l. 729--><p class="noindent">The <span
        class="cmssi-10">Trees Filter More </span>algorithm takes a Trees block as input and produces a Trees block as
     output. It filter trees. The
     algorithm has the following options:
     <!--l. 733-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">RequireAllTaxa</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- keep only trees that have the full set of taxa</span>
     <!--l. 735-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MinNumberOfTaxa</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- keep only trees that have at least this number of taxa</span>
     <!--l. 737-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MinTotalTreeLength</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- keep only trees that have at least this total length</span>
     <!--l. 740-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-144000C.4"></a>Trees Edges Filter</h5>
<a
        id="x1-144001"></a>
<!--l. 742--><p class="noindent">The <span
        class="cmssi-10">Trees Edges Filter </span>algorithm takes a Trees block as input and produces a Trees block as
     output. It filter edges in
     trees. The algorithm has the following options:
     <!--l. 746-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MinConfidence</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- keep only edges that have this minimum confidence value</span>
     <!--l. 748-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">MinEdgeLength</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- keep only edges that have this minimum length</span>
     <!--l. 750-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">UniformEdgeLengths</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- change all edge weights to 1</span>
     <!--l. 752-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Rescale</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- rescale each tree to total length of 100</span>
     <!--l. 755-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-145000C.4"></a>Tree Selector</h5>
<a
        id="x1-145001"></a>
<!--l. 757--><p class="noindent">The <span
        class="cmssi-10">Tree Selector </span>algorithm takes a Trees block as input and produces a Trees block as
     output. It select a tree to use.
     The algorithm has the following options:
     <!--l. 761-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Which</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- which tree to use</span>


     <!--l. 764-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-146000C.4"></a>Tree Selector Splits</h5>
<a
        id="x1-146001"></a>
<!--l. 766--><p class="noindent">The <span
        class="cmssi-10">Tree Selector Splits </span>algorithm takes a Trees block as input and produces a Splits block
     as output. It selects a single
     tree and extracts its splits. The algorithm has the following options:
     <!--l. 770-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;Which</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span>
     <!--l. 773-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-147000C.4"></a>Unique Topologies</h5>
<a
        id="x1-147001"></a>
<!--l. 775--><p class="noindent">The <span
        class="cmssi-10">Unique Topologies </span>algorithm takes a Trees block as input and produces a Trees block as
     output. It filters trees or
     rooted networks returning all unique topologies (using hardwired clusters). The algorithm has the following
     options:
     <!--l. 779-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Unrooted</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- ignore location of root</span>
     <!--l. 782-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-148000C.4"></a>Confidence Network</h5>
<a
        id="x1-148001"></a>
<!--l. 784--><p class="noindent">The <span
        class="cmssi-10">Confidence Network </span>algorithm takes a Trees block as input and produces a Splits block as
     output. It computes a
     credibility network using Beran&#8217;s algorithm. The algorithm has the following options:
     <!--l. 788-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Level</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Double&#x003E;</span></span></span> <span
        class="cmss-8">- set the level (between 0 and 1)</span>
     <!--l. 790-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">HighDimensionFilter</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- heuristically remove splits causing high-dimensional consensus network</span>
     <!--l. 793-->
<p class="noindent">Reference: [<a
        href="#XHusonBryant2006">Huson and Bryant</a>,&#x00A0;<a
        href="#XHusonBryant2006">2006</a>]
     <!--l. 795-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-149000C.4"></a>Phylogenetic Diversity</h5>
<a
        id="x1-149001"></a>
<!--l. 797--><p class="noindent">The <span
        class="cmssi-10">Phylogenetic Diversity </span>algorithm takes a Trees block as input and produces a Report
     block as output. It calculates
     the phylogenetic diversity for selected taxa. The algorithm has the following options:
     <!--l. 801-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Rooted</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- interpret trees as rooted?</span>
     <!--l. 803-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ApplyTo</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{OneTree</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;AllTrees}</span></span></span> <span
        class="cmss-8">- determine whether to apply to one or all trees</span>
     <!--l. 805-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">WhichTree</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- the index of the tree that the method will be applied to</span>
     <!--l. 808-->
<p class="noindent">Reference: [<a
        href="#XFaith1992">Faith</a>,&#x00A0;<a
        href="#XFaith1992">1992</a>]


     <!--l. 810-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-150000C.4"></a>Robinson Foulds Distances</h5>
<a
        id="x1-150001"></a>
<!--l. 812--><p class="noindent">The <span
        class="cmssi-10">Robinson Foulds Distances </span>algorithm takes a Trees block as input and produces a Report
     block as
     output. It calculates the Robinson-Foulds distance between each pair of trees The algorithm has the following
     options:
     <!--l. 816-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Normalize</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- normalized distances</span>
     <!--l. 819-->
<p class="noindent">Reference: [<span
        class="cmssbx-10">?</span>]
     <!--l. 821-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-151000C.4"></a>Tree Diversity Index</h5>
<a
        id="x1-151001"></a>
<!--l. 823--><p class="noindent">The <span
        class="cmssi-10">Tree Diversity Index </span>algorithm takes a Trees block as input and produces a Report block
     as output. It calculates the
     fair-proportion and equal-splits values on trees. The algorithm has the following options:
     <!--l. 827-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">Method</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{FairProportions</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;EqualSplits}</span></span></span> <span
        class="cmss-8">- choose the type of index calculation</span>
     <!--l. 829-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ApplyTo</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{OneTree</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;AllTrees}</span></span></span> <span
        class="cmss-8">- determine whether to apply to one or all trees</span>
     <!--l. 831-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">WhichTree</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- the index of the tree that the method will be applied to</span>
     <!--l. 834-->
<p class="noindent">References: [<a
        href="#XRedding2003">Redding</a>,&#x00A0;<a
        href="#XRedding2003">2003</a>,&#x00A0;<a
        href="#XReddingMooers2006">Redding and Mooers</a>,&#x00A0;<a
        href="#XReddingMooers2006">2006</a>]
     <!--l. 836-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-152000C.4"></a>Unrooted Shapley Values</h5>
<a
        id="x1-152001"></a>
<!--l. 838--><p class="noindent">The <span
        class="cmssi-10">Unrooted Shapley Values </span>algorithm takes a Trees block as input and produces a Report
     block as output. It calculates
     unrooted Shapley values on trees. The algorithm has the following options:
     <!--l. 842-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ApplyTo</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{OneTree</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;AllTrees}</span></span></span> <span
        class="cmss-8">- determine whether to apply to one or all trees</span>
     <!--l. 844-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">WhichTree</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- the index of the tree that the method will be applied to</span>
     <!--l. 847-->
<p class="noindent">Reference: [<a
        href="#XHaakeetal2008">Haake et&#x00A0;al.</a>,&#x00A0;<a
        href="#XHaakeetal2008">2008</a>]
     <!--l. 849-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-153000C.4"></a>Average Distances</h5>
<a
        id="x1-153001"></a>
<!--l. 851--><p class="noindent">The <span
        class="cmssi-10">Average Distances </span>algorithm takes a Trees block as input and produces a Distances block
     as output. It calculates the
     average distances between taxa over a set of trees.
     <!--l. 853-->
<p class="noindent">Reference: [<a
        href="#XLapointeCucumel1997">Lapointe and Cucumel</a>,&#x00A0;<a
        href="#XLapointeCucumel1997">1997</a>]


     <!--l. 855-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-154000C.4"></a>Enumerate Contained Trees</h5>
<a
        id="x1-154001"></a>
<!--l. 857--><p class="noindent">The <span
        class="cmssi-10">Enumerate Contained Trees </span>algorithm takes a Trees block as input and produces a Trees
     block as output. It
     enumerates all contained trees. The algorithm has the following options:
     <!--l. 861-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">RemoveDuplicates</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- suppress duplicate trees in output</span>
     <!--l. 864-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-155000C.4"></a>Loose And Lacy</h5>
<a
        id="x1-155001"></a>
<!--l. 866--><p class="noindent">The <span
        class="cmssi-10">Loose And Lacy </span>algorithm takes a Trees block as input and produces a Trees block as
     output. It computes the &#8216;loose&#8217;
     and &#8216;lacy&#8217; species for a given tree and taxon trait. The algorithm has the following options:
     <!--l. 870-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">SpeciesDefinition</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Loose</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Lacy</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Both}</span></span></span> <span
        class="cmss-8">- species definition to use</span>
     <!--l. 872-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">TraitNumber</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- number of specific trait to use</span>
     <!--l. 874-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">UseAllTraits</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Boolean&#x003E;</span></span></span> <span
        class="cmss-8">- use all traits</span>
     <!--l. 877-->
<p class="noindent">Reference: [<a
        href="#XHoppeetal2019">Hoppe et&#x00A0;al.</a>,&#x00A0;<a
        href="#XHoppeetal2019">2019</a>]
     <!--l. 879-->
<p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-156000C.4"></a>List One RSPR Trees</h5>
<a
        id="x1-156001"></a>
<!--l. 881--><p class="noindent">The <span
        class="cmssi-10">List One RSPR Trees </span>algorithm takes a Trees block as input and produces a Report block
     as output. It determines
     which trees are exactly on rSPR away from each other. The algorithm has the following options:
     <!--l. 885-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">ApplyTo</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{OneTree</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;AllTrees}</span></span></span> <span
        class="cmss-8">- determine whether to apply to one or all trees</span>
     <!--l. 887-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">WhichTree</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;&#x003C;Integer&#x003E;</span></span></span> <span
        class="cmss-8">- the index of the tree that the method will be applied to</span>
     <!--l. 890-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">C.5   </span> <a
        id="x1-157000C.5"></a>Algorithms on a Network Block</h3>
<!--l. 892--><p class="noindent">
<h5 class="subsubsectionHead"><a
        id="x1-158000C.5"></a>Show Network</h5>
<a
        id="x1-158001"></a>
<!--l. 894--><p class="noindent">The <span
        class="cmssi-10">Show Network </span>algorithm takes a Network block as input and produces a View block as
     output. It provides interactive
     visualizations of networks. The algorithm has the following options:
     <!--l. 898-->
<p class="noindent"><span class="obeylines-h"><span class="verb"><span
        class="cmtt-8">View</span><span
        class="cmtt-8">&#x00A0;=</span><span
        class="cmtt-8">&#x00A0;</span><span
        class="cmtt-8">&#x00A0;{Network</span><span
        class="cmtt-8">&#x00A0;|</span><span
        class="cmtt-8">&#x00A0;Text}</span></span></span> <span
        class="cmss-8">- the type of network viewer to use</span>


     <a
             id="x1-158002r63"></a>
<h2 class="appendixHead"><span class="titlemark">Appendix&#x00A0;D</span><br/><a
        id="x1-159000D"></a>Supported import and export formats</h2>
<!--l. 6--><p class="noindent">The program support several widely-used import and export formats.
<h3 class="sectionHead"><span class="titlemark">D.1   </span> <a
        id="x1-160000D.1"></a>Supported import formats</h3>
<!--l. 10--><p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.1.1   </span> <a
        id="x1-161000D.1.1"></a>Importers for a characters block</h4>
<a
        id="x1-161001"></a>
<!--l. 12--><p class="noindent">Can import characters data in the following formats: FastA, MSF, Stockholm, Nexus,
     Phylip. <a
             id="x1-161002"></a><a
             id="x1-161003"></a> <a
             id="x1-161004"></a><a
             id="x1-161005"></a> <a
             id="x1-161006"></a>
     <!--l. 15-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.1.2   </span> <a
        id="x1-162000D.1.2"></a>Importers for a distances block</h4>
<a
        id="x1-162001"></a>
<!--l. 17--><p class="noindent">Can import distances data in the following formats: Nexus, Phylip, CSV. <a
        id="x1-162002"></a><a
        id="x1-162003"></a> <a
        id="x1-162004"></a>
     <!--l. 20-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.1.3   </span> <a
        id="x1-163000D.1.3"></a>Importers for a trees block</h4>
<a
        id="x1-163001"></a>
<!--l. 22--><p class="noindent">Can import trees data in the following formats: Newick, Nexml, Nexus. <a
        id="x1-163002"></a><a
        id="x1-163003"></a> <a
        id="x1-163004"></a>
     <!--l. 25-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.1.4   </span> <a
        id="x1-164000D.1.4"></a>Importers for a splits block</h4>
<a
        id="x1-164001"></a>
<!--l. 27--><p class="noindent">Can import splits data in the following formats: Newick, Nexus. <a
        id="x1-164002"></a><a
        id="x1-164003"></a>
     <!--l. 30-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.1.5   </span> <a
        id="x1-165000D.1.5"></a>Importers for a network block</h4>
<a
        id="x1-165001"></a>
<!--l. 32--><p class="noindent">Can import network data in the following formats: Nexus. <a
        id="x1-165002"></a>


     <!--l. 35-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.1.6   </span> <a
        id="x1-166000D.1.6"></a>Importers for a genomes block</h4>
<a
        id="x1-166001"></a>
<!--l. 37--><p class="noindent">Can import genomes data in the following formats: Nexus. <a
        id="x1-166002"></a>
     <!--l. 41-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">D.2   </span> <a
        id="x1-167000D.2"></a>Supported output formats</h3>
<!--l. 43--><p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.2.1   </span> <a
        id="x1-168000D.2.1"></a>Exporters for a taxa block</h4>
<a
        id="x1-168001"></a>
<!--l. 44--><p class="noindent">Can export taxa data in the following formats: Nexus. <a
        id="x1-168002"></a>
     <!--l. 47-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.2.2   </span> <a
        id="x1-169000D.2.2"></a>Exporters for a characters block</h4>
<a
        id="x1-169001"></a>
<!--l. 48--><p class="noindent">Can export characters data in the following formats: Clustal, FastA, Nexus, Phylip. <a
        id="x1-169002"></a><a
        id="x1-169003"></a> <a
        id="x1-169004"></a><a
        id="x1-169005"></a>
     <!--l. 51-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.2.3   </span> <a
        id="x1-170000D.2.3"></a>Exporters for a distances block</h4>
<a
        id="x1-170001"></a>
<!--l. 52--><p class="noindent">Can export distances data in the following formats: Nexus, Phylip. <a
        id="x1-170002"></a><a
        id="x1-170003"></a>
     <!--l. 55-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.2.4   </span> <a
        id="x1-171000D.2.4"></a>Exporters for a trees block</h4>
<a
        id="x1-171001"></a>
<!--l. 56--><p class="noindent">Can export trees data in the following formats: NeXML, Newick, Nexus. <a
        id="x1-171002"></a><a
        id="x1-171003"></a> <a
        id="x1-171004"></a>


     <!--l. 59-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.2.5   </span> <a
        id="x1-172000D.2.5"></a>Exporters for a splits block</h4>
<a
        id="x1-172001"></a>
<!--l. 60--><p class="noindent">Can export splits data in the following formats: FastA, Newick, Nexus. <a
        id="x1-172002"></a><a
        id="x1-172003"></a> <a
        id="x1-172004"></a>
     <!--l. 63-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.2.6   </span> <a
        id="x1-173000D.2.6"></a>Exporters for a network block</h4>
<a
        id="x1-173001"></a>
<!--l. 64--><p class="noindent">Can export network data in the following formats: Nexus. <a
        id="x1-173002"></a>
     <!--l. 67-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.2.7   </span> <a
        id="x1-174000D.2.7"></a>Exporters for a genomes block</h4>
<a
        id="x1-174001"></a>
<!--l. 68--><p class="noindent">Can export genomes data in the following formats: Nexus. <a
        id="x1-174002"></a>
     <!--l. 71-->
<p class="noindent">
<h4 class="subsectionHead"><span class="titlemark">D.2.8   </span> <a
        id="x1-175000D.2.8"></a>Exporters for a view block</h4>
<a
        id="x1-175001"></a>
<!--l. 72--><p class="noindent">Can export view data in the following formats: GML, Nexus. <a
        id="x1-175002"></a><a
        id="x1-175003"></a>


<h3 class="sectionHead"><span class="titlemark">D.3   </span> <a
        id="x1-176000D.3"></a>Taxon display labels import</h3>
<a
        id="x1-176001"></a>
<!--l. 4--><p class="noindent">Taxon display labels can be imported from a text file. Each line of the file must contain
     two tab-separated entries. The
     first entry is the taxon name, as used in the input data, and the second entry is the corresponding display label,
     which
     may contain HTML formatting.
     <!--l. 8-->
<p class="noindent">Here is an example. The first label has font size 24 and text color blue. The second label is shown
     in bold. The remaining
     four labels have a yellow background.
     <!--l. 11-->
<div class="lstlisting" id="listing-1"><span class="label"><a
        id="x1-176002r1"></a></span><span
        class="cmtt-10">Co90</span><span
        class="cmtt-10">-125</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x003C;</span><span
        class="cmtt-10">size</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">"24"&#x003E;&#x003C;</span><span
        class="cmtt-10">c</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">blue</span><span
        class="cmtt-10">&#x003E;</span><span
        class="cmtt-10">Co90</span><span
        class="cmtt-10">-125</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-176003r2"></a></span><span
        class="cmtt-10">s428</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x003C;</span><span
        class="cmtt-10">b</span><span
        class="cmtt-10">&#x003E;</span><span
        class="cmtt-10">s428</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-176004r3"></a></span><span
        class="cmtt-10">s421</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x003C;</span><span
        class="cmtt-10">bg</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">yellow</span><span
        class="cmtt-10">&#x003E;</span><span
        class="cmtt-10">s421</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-176005r4"></a></span><span
        class="cmtt-10">s433</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x003C;</span><span
        class="cmtt-10">bg</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">yellow</span><span
        class="cmtt-10">&#x003E;</span><span
        class="cmtt-10">s433</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-176006r5"></a></span><span
        class="cmtt-10">s434</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x003C;</span><span
        class="cmtt-10">bg</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">yellow</span><span
        class="cmtt-10">&#x003E;</span><span
        class="cmtt-10">s434</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-176007r6"></a></span><span
        class="cmtt-10">s498</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x003C;</span><span
        class="cmtt-10">bg</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">yellow</span><span
        class="cmtt-10">&#x003E;</span><span
        class="cmtt-10">s498</span>
</div>
<!--l. 21--><p class="noindent">
<h3 class="sectionHead"><span class="titlemark">D.4   </span> <a
        id="x1-177000D.4"></a>Traits import</h3>
<a
        id="x1-177001"></a>
<!--l. 24--><p class="noindent">Traits can be imported from a text file. The first line of the file must define the
     names of the traits. The line must start
     with the keyword <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">Traits</span></span></span> and then must contain a list of the names of the different
     traits, separated by
     tabs
     <!--l. 30-->
<p class="noindent">Then there must be one line for each taxon. The first entry must be the taxon name and this must be
     followed by one
     value for each of the listed traits, separated by tabs.
     <!--l. 33-->
<p class="noindent">Here is an example defining five traits, <span class="obeylines-h"><span class="verb"><span
        class="cmtt-10">Europe</span></span></span> to <span class="obeylines-h"><span class="verb"><span
        class="cmtt-10">America</span></span></span>, for seven taxa <span class="obeylines-h"><span class="verb"><span
        class="cmtt-10">seq_1</span></span></span> to <span class="obeylines-h"><span class="verb"><span
        class="cmtt-10">seq_7</span></span></span>. In this case, the trait values
     are counts.
     <!--l. 37-->
<p class="noindent">The second line starting with the key word <span class="obeylines-h"><span class="verb"><span
        class="cmtt-10">Coordinates</span></span></span> is optional. When present, it provides the latitude and
     longitude
     associated with each trait.
     <!--l. 40-->
<div class="lstlisting" id="listing-2"><span class="label"><a
        id="x1-177002r1"></a></span><span
        class="cmtt-10">Traits</span><span
        class="cmtt-10">Europe</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">Asia</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">Africa</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">Australia</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">America</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177003r2"></a></span><span
        class="cmtt-10">Coordinates</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">53,16.75</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">43.68,87.33</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">5.4,26.5</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">-25.61,134.35</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0,-76</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177004r3"></a></span><span
        class="cmtt-10">seq_1</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">3</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">3</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177005r4"></a></span><span
        class="cmtt-10">seq_2</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">10</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">5</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">6</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177006r5"></a></span><span
        class="cmtt-10">seq_3</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">3</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">5</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177007r6"></a></span><span
        class="cmtt-10">seq_4</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">4</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">2</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177008r7"></a></span><span
        class="cmtt-10">seq_5</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">4</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">10</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177009r8"></a></span><span
        class="cmtt-10">seq_6</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">7</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">3</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177010r9"></a></span><span
        class="cmtt-10">seq_7</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">5</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0</span>
</div>
<!--l. 52--><p class="noindent">The program also supports a second way of specifying taxon-trait associations. After
     specifying the first (one or two)
     lines, the taxon-trait counts can also be specified by listing a taxon, a trait and then the desired count, like
     this:
     <!--l. 57-->
<div class="lstlisting" id="listing-3"><span class="label"><a
        id="x1-177011r1"></a></span><span
        class="cmtt-10">Traits</span><span
        class="cmtt-10">Europe</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">Asia</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">Africa</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">Australia</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">America</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177012r2"></a></span><span
        class="cmtt-10">Coordinates</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">53,16.75</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">43.68,87.33</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">5.4,26.5</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">-25.61,134.35</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">0,-76</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177013r3"></a></span><span
        class="cmtt-10">seq_1</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">Australia</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">3</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177014r4"></a></span><span
        class="cmtt-10">seq_1</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">America</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">3</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177015r5"></a></span><span
        class="cmtt-10">seq_2</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">Europe</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">10</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177016r6"></a></span><span
        class="cmtt-10">seq_2</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">Asia</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">5</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177017r7"></a></span><span
        class="cmtt-10">seq_2</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">Australia</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">&#x00A0;</span><span
        class="cmtt-10">6</span><span
        class="cmtt-10">&#x00A0;</span><br/><span class="label"><a
        id="x1-177018r8"></a></span><span
        class="cmtt-10">...</span>
</div>


<!--l. 1--><p class="noindent">


     <a
             id="x1-177019r63"></a>
<h2 class="appendixHead"><span class="titlemark">Appendix&#x00A0;E</span><br/><a
        id="x1-178000E"></a>Workflow</h2>
<!--l. 4--><p class="noindent">SplitsTree is designed around the concept of a workflow. This is a provenance graph in
     which nodes explicitly represent
     data blocks and algorithms.
<h3 class="sectionHead"><span class="titlemark">E.1   </span> <a
        id="x1-179000E.1"></a>Input and working nodes</h3>
<!--l. 11--><p class="noindent">The graph is displayed the workflow tab (see Section&#x00A0;<a
        href="#x1-160001.13">1.13<!--tex4ht:ref: sec:workflow-tab --></a>) and is also presented in the sidebar (see
     Section&#x00A0;<a
             href="#x1-210001.18">1.18<!--tex4ht:ref: sec:sidebar --></a>). While
     the casual user will use menu items to set up and change the graph (without being aware of the graph), a user more
     familiar with the program will use controls in the workflow tab and sidebar to explicitly add, delete, duplicate or
     modify
     nodes and edges in the workflow graph.
     <!--l. 19-->
<p class="noindent">Each main window contains one workflow and the workflow represents one input dataset, all applied
     algorithms and
     derived data.
     <!--l. 22-->
<p class="noindent">In more detail, the graph is a tree with two root nodes (see Fig.&#x00A0;<a
        href="#x1-1790011">E.1<!--tex4ht:ref: fig:top-nodes --></a>).
     <!--l. 24-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-1790011"></a>


     <!--l. 25--><p class="noindent"><img
        src="figs/top-nodes.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;E.1: </span><span
             class="content">The workflow is a directed tree with two root nodes, the first representing the set of input taxa and
the second representing the input data. The taxa filter node can be used to remove input taxa and thus produces
the working taxa node. Filtering taxa also has the effect of removing the corresponding data from the input data
node and gives rise to the working data node. Subsequent algorithms are run on the data represented by the
working taxa and working data nodes. </span></div><!--tex4ht:label?: x1-1790011 -->


     <!--l. 32--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 34--><p class="noindent">The first root node represents the set of input taxa<a
        id="x1-179002"></a>. The number of input taxa is fixed and each taxon has a unique name
     (a label that does not contain special characters such as a single or double quote and must not be a number.) In
     addition,
     each taxon can have a display label, which may contain certain HTML commands, that is used to draw the label
     associated with the taxon.
     <!--l. 42-->
<p class="noindent">The second root node represents the input data<a
        id="x1-179003"></a>. This may be a set of characters (or sequence alignment), a distance
     matrix, a collection of trees (or rooted networks), or a set of splits.
     <!--l. 47-->
<p class="noindent">The input-taxa node has one child, the taxa filter node<a
        id="x1-179004"></a>. This can be used to remove some of the input taxa. This node
     has one child that contains the set of working taxa<a
             id="x1-179005"></a>. This node represents the set of taxa that are actually used in all
     computations.
     <!--l. 53-->
<p class="noindent">The input data node has one child that contains the working data<a
        id="x1-179006"></a>. The data associated with this node
     is copied from the input data node, removing any taxa that have been deactivated using the taxa filter
     node.
     <!--l. 57-->
<p class="noindent">All calculations undertaken in the program are based on the set of working taxa and the working
     data.
     <!--l. 59-->
<p class="noindent">If the input data is character data (or a multiple sequence alignment), then the input data is
     displayed in the alignment
     tab (see Section&#x00A0;<a
             href="#x1-80001.5">1.5<!--tex4ht:ref: sec:alignment-tab --></a>) and controls associated with the tab allow
     the user to add or remove taxa in the same manner as
     when using the taxa filter.
     <!--l. 61-->
<p class="noindent">The taxa filter also allows the user to edit the display labels associated with taxa.
<h3 class="sectionHead"><span class="titlemark">E.2   </span> <a
        id="x1-180000E.2"></a>Data and algorithm nodes</h3>
<!--l. 66--><p class="noindent">The workflow contains two main types of nodes, algorithm nodes<a
        id="x1-180001"></a> and data nodes<a
        id="x1-180002"></a>. The workflow is a bipartite graph: data
     nodes only have algorithm nodes as children and vice versa (see Fig.&#x00A0;<a
             href="#x1-1800032">E.2<!--tex4ht:ref: fig:chain-nodes --></a>).
     <!--l. 73-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-1800032"></a>


     <!--l. 74--><p class="noindent"><img
        src="figs/chain-nodes.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;E.2: </span><span
             class="content">Here we show the workflow that shows a split network computed from characters data. The workflow
contains chains of alternating algorithm and data nodes. The characters data (working data node) is followed
by the <span
                  class="cmmi-10">p</span>-distance calculation, giving rise to a data node containing distances. The distances are provided to the
neighbor-net algorithm, giving rise to a set of splits. The splits are passed to the show splits node, which computes
the visualization, which is represented by the splits view node. </span></div><!--tex4ht:label?: x1-1800032 -->


     <!--l. 85--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 87--><p class="noindent">In the Figure (see Fig.&#x00A0;<a
        href="#x1-1790011">E.1<!--tex4ht:ref: fig:top-nodes --></a>) it looks like the input data node and working data
     node are directly connected to each other.
     However, actually there is a special algorithm node between the two that facilitates the filtering of data
     when the taxa filter is used. As this node is only used internally, it is not displayed in the workflow tab or
     sidebar.
     <!--l. 92-->
<p class="noindent">Each algorithm node computes data that is based on to a data node, while each data node is provided
     with the set of
     working taxa and one parental data node to work with. (Some algorithms, such as those that perform bootstrapping,
     additionally access other ancestral nodes to perform their calculations.)
     <!--l. 95-->
<p class="noindent">The final nodes of the workflow (leaves) are always data nodes and each is of one of two special
     kinds. First, there
     are view nodes<a
             id="x1-180004"></a> that represent the graphical visualization of trees or networks. Second, there are
     reporter
     nodes<a
             id="x1-180005"></a> that are used to present the textual output of a calculation, such as the computation
     of Tajima&#8217;s
     D.
     <!--l. 100-->
<p class="noindent">An algorithm node can represent one of several algorithms, depending on the type of input data and
     output data. For
     example, there are three algorithms that take as input a distance matrix and produce, as output, a set of splits,
     namely
     neighbor-net, split decomposition and the Buneman tree algorithm. In the Figure (see Fig.&#x00A0;<a
             href="#x1-1800063">E.3<!--tex4ht:ref: fig:algorithm-node --></a>) we show such an
     algorithm node and the controls associated with it.
     <!--l. 107-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-1800063"></a>


     <!--l. 108--><p class="noindent"><img
        src="figs/algorithm-node.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;E.3: </span><span
             class="content">The highlighted algorithm node has two buttons, an edit button that opens the corresponding
algorithm tab, and a run button, that runs the algorithm and all downstream calculations. In addition, there is
status icon that indicates whether the algorithm has been successfully run (green tick), is waiting to be run (yellow
clock) or has failed or has been canceled (red cross). </span></div><!--tex4ht:label?: x1-1800063 -->


     <!--l. 118--><p class="noindent"></div>
<hr class="endfigure">
<!--l. 120--><p class="noindent">A data node usually represents a block of data, which can be either characters (aligned
     sequences), a distance matrix, a
     set of splits, a collection of trees (and/or rooted networks) or a haplotype network (see Fig.&#x00A0;<a
             href="#x1-1800074">E.4<!--tex4ht:ref: fig:data-node --></a>). A view node is a data
     node that corresponds to a viewer for trees or networks.
     <!--l. 123-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-1800074"></a>


     <!--l. 124--><p class="noindent"><img
        src="figs/data-node.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;E.4: </span><span
             class="content">A data node has a drop-down menu that allows one to attach a new algorithm to the node, thus
starting a new chain of nodes. The show button opens a tab that provides a textual representation of the data.
The status icon whether the data is up-to-date (green tick), is waiting to be updated (yellow clock) or is missing
due to a failed or canceled algorithm (red cross). </span></div><!--tex4ht:label?: x1-1800074 -->


     <!--l. 128--><p class="noindent"></div>
<hr class="endfigure">
<h3 class="sectionHead"><span class="titlemark">E.3   </span> <a
        id="x1-181000E.3"></a>Exporting the workflow</h3>
<!--l. 134--><p class="noindent">When saving a document, the workflow, including all input data and computed data, are
     saved to a file to be opened
     later. Such a full file has file suffix <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">.stree6</span></span></span>.
     <!--l. 139-->
<p class="noindent">In addition, using the <span class="obeylines-h"><span class="verb"><span
        class="cmtt-10">File-&#x003E;Export-&#x003E;Workflow...</span></span></span> menu item, the user can save just
     the workflow graph, without the
     current data, to a file. Such a workflow file has file suffix <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">.wflow6</span></span></span>.
     <!--l. 142-->
<p class="noindent">A saved workflow can be reopened in SplitsTree and data can then be loaded using the <span
        class="obeylines-h"><span class="verb"><span
        class="cmtt-10">File-&#x003E;Replace</span><span
        class="cmtt-10">&#x00A0;Data...</span></span></span> menu
     item.
     <!--l. 146-->
<p class="noindent">
<h3 class="sectionHead"><span class="titlemark">E.4   </span> <a
        id="x1-182000E.4"></a>Running a workflow on multiple datasets</h3>
<!--l. 150--><p class="noindent">To run an analysis on multiple datasets, the SplitsTree packages comes with a
     commandline tool called <span
             class="cmssi-10">run-workflow</span><a
             id="x1-182001"></a>,
     located in the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">tools</span></span></span> directory. The basic idea is as follows. First, use SplitsTree
     to interactively set up the desired
     analysis. Then save export the workflow to a file(see Section&#x00A0;<a
             href="#x1-181000E.3">E.3<!--tex4ht:ref: sec:export-workflow --></a>). Then use the <span
             class="obeylines-h"><span class="verb"><span
             class="cmtt-10">run-workflow</span></span></span> program to apply the
     workflow to multiple datasets.
     <!--l. 160-->
<p class="noindent">The program is configured using a number of commandline options, to set the workflow file, to set
     the input data files, to
     set the output file or files, and to determine what should be written to the output.
     <!--l. 163-->
<p class="noindent">Here is a synopsis of the program:


<div class="verbatim" id="verbatim-15">
     SYNOPSIS
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;RunWorkflow&#x00A0;[options]
     &#x00A0;<br/>DESCRIPTION
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;Runs&#x00A0;a&#x00A0;SplitsTree6&#x00A0;workflow&#x00A0;on&#x00A0;input&#x00A0;data
     &#x00A0;<br/>OPTIONS
     &#x00A0;<br/>&#x00A0;Input&#x00A0;Output:
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-w,&#x00A0;--workflow&#x00A0;[string]&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;File&#x00A0;containing&#x00A0;SplitsTree6&#x00A0;workflow.&#x00A0;Mandatory&#x00A0;option.
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-i,&#x00A0;--input&#x00A0;[string(s)]&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;File(s)&#x00A0;containing&#x00A0;input&#x00A0;data&#x00A0;(or&#x00A0;directory).&#x00A0;Mandatory&#x00A0;option.
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-f,&#x00A0;--format&#x00A0;[string]&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;Input&#x00A0;format.&#x00A0;Default:&#x00A0;Unknown.&#x00A0;Legal&#x00A0;values:&#x00A0;&#8217;Unknown&#8217;&#x00A0;&#8217;FastA&#8217;&#x00A0;&#8217;MSF&#8217;&#x00A0;&#8217;Newick&#8217;
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#8217;Nexml&#8217;&#x00A0;&#8217;Nexus&#8217;&#x00A0;&#8217;Phylip&#8217;&#x00A0;&#8217;Stockholm&#8217;
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-o,&#x00A0;--output&#x00A0;[string(s)]&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;Output&#x00A0;file(s)&#x00A0;(or&#x00A0;directory&#x00A0;or&#x00A0;stdout).&#x00A0;Default&#x00A0;value(s):&#x00A0;&#8217;stdout&#8217;
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-n,&#x00A0;--node&#x00A0;[string]&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;Title&#x00A0;of&#x00A0;node&#x00A0;to&#x00A0;be&#x00A0;exported&#x00A0;(if&#x00A0;none&#x00A0;given,&#x00A0;will&#x00A0;save&#x00A0;whole&#x00A0;file).
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-e,&#x00A0;--exporter&#x00A0;[string]&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;Name&#x00A0;of&#x00A0;exporter&#x00A0;to&#x00A0;use.&#x00A0;&#x00A0;Legal&#x00A0;values:&#x00A0;&#8217;Clustal&#8217;&#x00A0;&#8217;FastA&#8217;&#x00A0;&#8217;GML&#8217;&#x00A0;&#8217;NeXML&#8217;&#x00A0;&#8217;Newick&#8217;
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#8217;Nexus&#8217;&#x00A0;&#8217;Phylip&#8217;&#x00A0;&#8217;PlainText&#8217;&#x00A0;&#8217;NexusWithTaxa&#8217;
     &#x00A0;<br/>&#x00A0;Other:
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-x,&#x00A0;--inputExt&#x00A0;[string]&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;File&#x00A0;extension&#x00A0;for&#x00A0;input&#x00A0;files&#x00A0;(when&#x00A0;providing&#x00A0;directory&#x00A0;for&#x00A0;input).
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-r,&#x00A0;--recursive&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;Recursively&#x00A0;visit&#x00A0;all&#x00A0;sub-directories.&#x00A0;Default:&#x00A0;false.
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-t,&#x00A0;--time&#x00A0;[string]&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;Maximum&#x00A0;wall-clock&#x00A0;time&#x00A0;to&#x00A0;run&#x00A0;(e.g.&#x00A0;100s,&#x00A0;2m,&#x00A0;3h&#x00A0;or&#x00A0;4d).&#x00A0;Default:&#x00A0;unlimited.
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-p,&#x00A0;--propertiesFile&#x00A0;[string]&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;Properties&#x00A0;file.
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-s,&#x00A0;--silent&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;Silent&#x00A0;mode&#x00A0;(hide&#x00A0;all&#x00A0;stderr&#x00A0;output).&#x00A0;Default:&#x00A0;false.
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-v,&#x00A0;--verbose&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;Echo&#x00A0;commandline&#x00A0;options&#x00A0;and&#x00A0;be&#x00A0;verbose.&#x00A0;Default:&#x00A0;false.
     &#x00A0;<br/>&#x00A0;&#x00A0;&#x00A0;&#x00A0;-h,&#x00A0;--help&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;Show&#x00A0;program&#x00A0;usage&#x00A0;and&#x00A0;quit.
</div>
<!--l. 189--><p class="nopar">
     <!--l. 192-->
<p class="noindent">Here is a summary of the most important options:
<ul class="itemize1">
     <li class="itemize">Use the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">-w</span></span></span> option to specify the workflow file.
     </li>
     <li class="itemize">Use the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">-i</span></span></span> option to list the input files or a directory that that contains
          the input files.
     </li>
     <li class="itemize">Use the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">-f</span></span></span> option to specify the file format of the input files.
     </li>
     <li class="itemize">Optionally, use the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">-x</span></span></span> option to specify the file suffixes of the input files, if a
          specified directories contains
          some files that should not be used.
     </li>
     <li class="itemize">Optionally, use the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">-r</span></span></span> option to recursively visit all directories contained in the
          specified input directory.
     </li>
     <li class="itemize">Use the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">-o</span></span></span> option to specify the output file or files. If you specify one
          directory, then one output file per
          input file will be created and written to that directory. If you specify one file, then all output is written
          to
          that file. You can also specify one output file per input file. Use the keyword <span
                  class="obeylines-h"><span class="verb"><span
                  class="cmtt-10">stdout</span></span></span> to have all output
          written to the console. If output files end on <span class="obeylines-h"><span class="verb"><span
                  class="cmtt-10">.gz</span></span></span> then they will be written in gzip format.
     </li>
     <li class="itemize">Use the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">-n</span></span></span> option to specify a specific data node to be saved to output. For
          example, if your analysis
          generates a trees block called <span class="obeylines-h"><span class="verb"><span
                  class="cmtt-10">Trees</span></span></span>, then you can specify <span class="obeylines-h"><span
                  class="verb"><span
                  class="cmtt-10">-n</span><span
                  class="cmtt-10">&#x00A0;Trees</span></span></span> to output the trees. If this option is
          not specified, then the whole workflow is output containing all input and computed data.
     </li>
     <li class="itemize">Use the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">-e</span></span></span> option to specify which exporter (and thus format) to use when
          generating output. This option
          is ignored if the <span class="obeylines-h"><span class="verb"><span
                  class="cmtt-10">-n</span></span></span> is not specified.
     </li>
</ul>


<a
        id="x1-182002r63"></a>
<h2 class="appendixHead"><span class="titlemark">Appendix&#x00A0;F</span><br/><a
        id="x1-183000F"></a>Styling labels</h2>
<!--l. 4--><p class="noindent">In SplitsTree, each taxon is represented by a unique label. These labels are specified in
     the input taxa block. In addition to
     these labels, the taxon block may also contain a set of &#8220;display labels&#8221;, one for each taxon. If
     provided, the display labels
     are used in drawings of trees and networks. Display labels can be styled using a set HTML tags. There are several
     ways to
     do this:
<ul class="itemize1">
     <li class="itemize">Use the side panel (on the right) to choose a font, font size, text color, background color or
          mark. Each of
          these interactive choices produce corresponding HTML tags in the display labels.
     </li>
     <li class="itemize">Right-clicking on a taxon label displayed in a tree or network will open an input dialog for
          the display label.
          The dialog allows the user to edit the display label any changes to the label are immediately shown in a
          preview pane.
     </li>
     <li class="itemize">In the <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">TaxaFilter</span></span></span> tab, there is an entry for each taxon, which can be edited.
     </li>
     <li class="itemize">Display labels can be imported from a file (and exported to a file for reuse). Each line of the
          file must
          contain a key and a value separated by a tab. The key is the unique taxon name and the value is the desired
          display label.
     </li>
</ul>
<!--l. 25--><p class="noindent">Here is a list of all supported HTML tags, most are standard HTML, a few are specific to
     SplitsTree:
<ul class="itemize1">
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;i&#x003E;text&#x003C;/i&#x003E;</span></span></span> - display enclosed text in
          italics,
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;b&#x003E;text&#x003C;/b&#x003E;</span></span></span> - display enclosed text in
          bold,
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;sup&#x003E;text&#x003C;/sup&#x003E;</span></span></span> - display enclosed text as
          super-script,
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;sub&#x003E;text&#x003C;/sub&#x003E;</span></span></span> - display enclosed text as
          super-script,
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;u&#x003E;text&#x003C;/u&#x003E;</span></span></span> - display enclosed text as
          underlined,
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;a&#x003E;text&#x003C;/a&#x003E;</span></span></span> - display enclosed text using
          strike-through,
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;br&#x003E;</span></span></span> - add a new-line,
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;font</span><span
             class="cmtt-10">&#x00A0;"name"&#x003E;text&#x003C;/font&#x003E;</span></span></span> - display enclosed
          text using the named font,
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;size</span><span
             class="cmtt-10">&#x00A0;"value"&#x003E;text&#x003C;/size&#x003E;</span></span></span> - display enclosed
          text using the given font size value,
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;c</span><span
             class="cmtt-10">&#x00A0;"value"&#x003E;text&#x003C;/c&#x003E;</span></span></span> - display enclosed text
          using the given color value,
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;bg</span><span
             class="cmtt-10">&#x00A0;"value"&#x003E;background-color&#x003C;/bg&#x003E;</span></span></span> - display
          enclosed text using the given background color value,


     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;mark</span><span
             class="cmtt-10">&#x00A0;shape="value"</span><span
             class="cmtt-10">&#x00A0;width="value"</span><span
             class="cmtt-10">&#x00A0;height="value"</span><span
             class="cmtt-10">&#x00A0;fill="color"</span><span
             class="cmtt-10">&#x00A0;stroke="color"&#x003E;</span></span></span> - adds a
          mark to text, using the provided shape, width, height, fill and stroke colors, and
     </li>
     <li class="itemize"><span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&#x003C;img</span><span
             class="cmtt-10">&#x00A0;src="url"</span><span
             class="cmtt-10">&#x00A0;alt="text"</span><span
             class="cmtt-10">&#x00A0;width="value"</span><span
             class="cmtt-10">&#x00A0;height="value"&#x003E;</span></span></span> - adds an image using the provided
          URL and width and/or height.
     </li>
</ul>
<!--l. 48--><p class="noindent">HTML numerical codes can be used to specify characters.
     <!--l. 51-->
<p class="noindent">In the Figure (see Fig.&#x00A0;<a
        href="#x1-1830011">F.1<!--tex4ht:ref: fig:bees-with-images --></a>) a display label has been set for each taxon
     (except for <span
             class="cmssi-10">A.</span><span
             class="cmssi-10">&#x00A0;Andrenof</span>). Using HTML, the
     display labels include a Wikipedia image of the represented species of bees. In addition, the display label editor
     is shown
     open for <span
             class="cmssi-10">A.</span><span
             class="cmssi-10">&#x00A0;Koschev</span>. The display label contains HTML tags for italics and a tag to
     start a superscript. The HTML entity
     <span class="obeylines-h"><span class="verb"><span
             class="cmtt-10">&amp;dagger;</span></span></span> is used to display a dagger.
     <!--l. 59-->
<p class="noindent">
<hr class="figure">
<div class="figure"
>


     <a
             id="x1-1830011"></a>


     <!--l. 60--><p class="noindent"><img
        src="figs/bees-with-images.png" alt="PIC"
>
     <br/>
     <div class="caption"
     ><span class="id">Figure&#x00A0;F.1: </span><span
             class="content">Here, for all taxa (except for <span
             class="cmssi-10">A.</span><span
             class="cmssi-10">&#x00A0;Andrenof</span>), the display label includes a Wikipedia image of the
represented species of bees. By right-clicking on the <span
                  class="cmssi-10">A.</span><span
                  class="cmssi-10">&#x00A0;Koschev </span>label, the display-label editor was opened for this
label. </span></div><!--tex4ht:label?: x1-1830011 -->


     <!--l. 67--><p class="noindent"></div>
<hr class="endfigure">


<!--l. 111--><p class="noindent">


<h3 class="likesectionHead"><a
        id="x1-184000F"></a>Acknowledgments</h3>
<!--l. 115--><p class="noindent">We thank Daria Evseeva for working on the code with us. This program uses the JAMA
     library in the neighbor-net
     algorithm.


     <!--l. 1-->
<p class="noindent">


<h2 class="likechapterHead"><a
        id="x1-185000F"></a>Bibliography</h2>
<div class="thebibliography">
     <p class="bibitem"><span class="biblabel">
<a
        id="XBagcietal2021"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>C.&#x00A0;Bagci, D.&#x00A0;Bryant,
          B.&#x00A0;Cetinkaya, and DH&#x00A0;Huson. Microbial phylogenetic context using phylogenetic
          outlines. <span
                  class="cmssi-10">Genome Biology and Evolution</span>, 13(9), 2021.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XBandelt1995"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>H-J Bandelt. Combination of data
          in phylogenetic analysis. In <span
                  class="cmssi-10">Systematics and Evolution of the Ranunculiflorae</span>,
          pages 355&#8211;361. Springer, 1995.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XBandeltDress1992"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>HJ&#x00A0;Bandelt and AWM
          Dress. A canonical decomposition theory for metrics on a finite set. <span
                  class="cmssi-10">Advances in</span>
          <span
                  class="cmssi-10">Mathematics</span>, 92:47&#8211;105, 1992.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XBandeltetal1999"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>HJ&#x00A0;Bandelt, P.&#x00A0;Forster,
          and A.&#x00A0;Rhl. Median-joining networks for inferring intraspecific phylogenies. <span
                  class="cmssi-10">Molecular</span>
          <span
                  class="cmssi-10">Biology and Evolution</span>, 16:37&#8211;48, 1999.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XBoothLueker1976"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>K.&#x00A0;S. Booth and G.&#x00A0;S.
          Lueker. Testing for the consecutive ones property, interval graphs, and graph planarity
          using pq-tree algorithms. <span
                  class="cmssi-10">Journal of Computer and System Sciences</span>, 13(3):335&#8211;379, 1976.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XBouckaert2010"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>RR&#x00A0;Bouckaert.
          Densitree: making sense of sets of phylogenetic trees. <span
                  class="cmssi-10">Bioinformatics</span>, 26(1):1372&#8211;1373, 2010.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XBruenetal2006"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>TC&#x00A0;Bruen, H.&#x00A0;Philippe,
          and D.&#x00A0;Bryant. A simple and robust statistical test for detecting the presence of
          recombination. <span
                  class="cmssi-10">Genetics</span>, 173:2665&#8211;2681, 2006.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XBryant2001"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>D.&#x00A0;Bryant. A
          classification of consensus methods for phylogenetics. In <span
                  class="cmssi-10">Bioconsensus</span>, 2001.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XBryantHuson2023"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>D.&#x00A0;Bryant and D.&#x00A0;H.
          Huson. NeighborNet&#8211;improved algorithms and implementation. <span
                  class="cmssi-10">Frontiers  in</span>
          <span
                  class="cmssi-10">Bioinformatics</span>, 2023.
     </p>


     <p class="bibitem"><span class="biblabel">
<a
        id="XBryantMoulton2004"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>D.&#x00A0;Bryant and V.&#x00A0;Moulton.
          Neighbor-net: An agglomerative method for the construction of phylogenetic
          networks. <span
                  class="cmssi-10">Molecular Biology and Evolution</span>, 21(2):255&#8211;265, 2004.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHusonScornavacca2012"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>C.&#x00A0;Scornavacca
          D.&#x00A0;H.&#x00A0;Huson. Dendroscope 3: An interactive tool for rooted phylogenetic trees and networks.
          <span
                  class="cmssi-10">Systematic Biology</span>, 61(6):1061&#8211;1067, 2012.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XDice1945"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>LR&#x00A0;Dice. Measures of the
          amount of ecologic association between species. <span
                  class="cmssi-10">Ecology</span>, 26(3):297&#8211;302, 1945.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XDressHuson2004"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>Andreas W&#x00A0;M Dress and
          Daniel&#x00A0;H Huson. Constructing splits graphs. <span
                  class="cmssi-10">IEEE/ACM  Transactions  on</span>
          <span
                  class="cmssi-10">Computational Biology and Bioinformatics</span>, 1(3):109&#8211;115, 2004.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XExcoffierSmouse1994"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>L.&#x00A0;Excoffier and
          PE&#x00A0;Smouse. Using allele frequencies and geographic subdivision to reconstruct gene trees
          within a species: molecular variance parsimony. <span
                  class="cmssi-10">Genetics</span>, 136(1):343&#8211;359, 1994.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XFaith1992"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>DP&#x00A0;Faith. Conservation
          evaluation and phylogenetic diversity. <span
                  class="cmssi-10">Biological Conservation</span>, 61:1&#8211;10, 1992.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XFelsenstein1981"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>J.&#x00A0;Felsenstein.
          Evolutionary trees from DNA sequences: a maximum likelihood approach. <span
                  class="cmssi-10">Journal of Molecular</span>
          <span
                  class="cmssi-10">Evolution</span>, 17(6):368&#8211;376, 1981.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XFelsenstein1985"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>J.&#x00A0;Felsenstein.
          Confidence limits on phylogenies: an approach using the bootstrap. <span
                  class="cmssi-10">Evolution</span>, 39(4):783&#8211;791,
          1985.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XFelsensteinChurchill1996"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>J.&#x00A0;Felsenstein
          and GA&#x00A0;Churchill. A hidden markov model approach to variation among sites in rate of evolution,
          and the branching order in hominoidea. <span
                  class="cmssi-10">Molecular Biology and Evolution</span>, 13(1):93&#8211;104, 1996.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XFrancisetal2021"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>A.&#x00A0;Francis, D.H.
          Huson, and M.A. Steel. Normalising phylogenetic networks. <span
                  class="cmssi-10">Molecular Phylogenetics and</span>
          <span
                  class="cmssi-10">Evolution</span>, 163, 2021.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XGascuel1997"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>O.&#x00A0;Gascuel. BIONJ: an
          improved version of the nj algorithm based on a simple model of sequence data.
          <span
                  class="cmssi-10">Molecular Biology and Evolution</span>, 14:685&#8211;695, 1997.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XGower1966"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>JC&#x00A0;Gower. Some distance
          properties of latent root and vector methods used in multivariate analysis.
          <span
                  class="cmssi-10">Biometrika</span>, 53(3-4):325&#8211;338, 1966.
     </p>


     <p class="bibitem"><span class="biblabel">
<a
        id="XGruenstaeudl2019"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>M.&#x00A0;Gruenstaeudl. Why
          the monophyly of nymphaeaceae currently remains indeterminate: an assessment based
          on gene-wise plastid phylogenomics. <span
                  class="cmssi-10">Plant Systematics and Evolution</span>, 305:827&#8211;836, 2019.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHaakeetal2008"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>CJ&#x00A0;Haake, A.&#x00A0;Kashiwada,
          and FE&#x00A0;Su. The Shapley value of phylogenetic trees. <span
                  class="cmssi-10">J Math Biol</span>, 56:479&#8211;497,
          2008.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHamming1950"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>RW&#x00A0;Hamming. Error
          detecting and error correcting codes. <span
                  class="cmssi-10">Bell System Technical Journal</span>, 29(2):147&#8211;160,
          1950.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHasegawaetal1985"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>M.&#x00A0;Hasegawa, H.&#x00A0;Kishino,
          and T.&#x00A0;Yano. Dating of human-ape splitting by a molecular clock of mitochondrial
          DNA. <span
                  class="cmssi-10">Journal of Molecular Evolution</span>, 22(2):160&#8211;174, 1985.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHollandetal2002"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>BR&#x00A0;Holland, KT&#x00A0;Huber,
          AWM Dress, and V.&#x00A0;Moulton. Delta plots: A tool for analyzing phylogenetic distance
          data. <span
                  class="cmssi-10">Molecular Biology and Evolution</span>, 19(12):2051&#8211;2059, 2002.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHollandetal2004"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>BR&#x00A0;Holland, KT&#x00A0;Huber,
          and Vincent PJ&#x00A0;Lockhart V.&#x00A0;Moulton. Using Consensus Networks to Visualize
          Contradictory Evidence for Species Phylogeny. <span
                  class="cmssi-10">Molecular Biology and Evolution</span>, 21(7):1459&#8211;1461, 2004.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHoppeetal2019"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>A.&#x00A0;Hoppe, S.&#x00A0;Tuerpitz,
          and MA&#x00A0;Steel. Species notions that combine phylogenetic trees and phenotypic
          partitions. <span
                  class="cmssi-10">Journal of Mathematical Biology</span>, 78:117&#8211;134, 2019.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHuson2025Tanglegrams"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>Daniel&#x00A0;H. Huson.
          Displacement optimized tanglegrams for rooted phylogenetic trees and networks. In
          preparation, 2025.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHusonBryant2024"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>Daniel&#x00A0;H. Huson and
          David Bryant. The splitstree app: interactive analysis and visualization using phylogenetic
          trees and networks. <span
                  class="cmssi-10">Nature Methods</span>, 2024. URL <a
                  href="https://doi.org/10.1038/s41592-024-02406-3" class="url"><span
                  class="cmtt-10">https://doi.org/10.1038/s41592-024-02406-3</span></a>.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHusonBryant2006"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>DH&#x00A0;Huson and D&#x00A0;Bryant.
          Application of phylogenetic networks in evolutionary studies. <span
                  class="cmssi-10">Molecular Biology</span>
          <span
                  class="cmssi-10">and Evolution</span>, 23(2):254&#8211;267, 2006.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHusonCetinkaya2023"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>DH&#x00A0;Huson and B&#x00A0;Cetinkaya.
          Visualizing incompatibilities in phylogenetic trees using consensus outlines. <span
                  class="cmssi-10">Front.</span>
          <span
                  class="cmssi-10">Bioinform.</span>, 2023.
     </p>


     <p class="bibitem"><span class="biblabel">
<a
        id="XHusonLinz2018"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>DH&#x00A0;Huson and S.&#x00A0;Linz.
          Autumn algorithmcomputation of hybridization networks for realistic phylogenetic
          trees. <span
                  class="cmssi-10">IEEE/ACM Transactions on Computational Biology and Bioinformatics</span>, 15:398&#8211;420,
          2018.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHusonRupp2008"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>DH&#x00A0;Huson and R.&#x00A0;Rupp.
          Summarizing multiple gene trees using cluster networks. In <span
                  class="cmssi-10">Algorithms  in</span>
          <span
                  class="cmssi-10">Bioinformatics. WABI 2008</span>, volume 5251 of <span
                  class="cmssi-10">Lecture Notes in Computer Science</span>, 2008.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHusonSteel2004"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>DH&#x00A0;Huson and MA&#x00A0;Steel.
          Phylogenetic trees based on gene content. <span
                  class="cmssi-10">Bioinformatics</span>, 20(13):2044&#8211;2049,
          2004.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHusonetal2004"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>DH&#x00A0;Huson, T.&#x00A0;Dezulian,
          T.&#x00A0;Kloepper, and MA&#x00A0;Steel. Phylogenetic super-networks from partial trees.
          <span
                  class="cmssi-10">IEEE/ACM Transactions in Computational Biology and Bioinformatics</span>, 1(4):151&#8211;158,
          2004.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XHusonetal2012"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>DH&#x00A0;Huson, R.&#x00A0;Rupp,
          and C.&#x00A0;Scornavacca. <span
                  class="cmssi-10">Phylogenetic Networks</span>. Cambridge, 2012.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XJaccard1901"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>P.&#x00A0;Jaccard. tude
          comparative de la distribution florale dans une portion des alpes et des jura. <span
                  class="cmssi-10">Bulletin de la</span>
          <span
                  class="cmssi-10">Socit Vaudoise des Sciences Naturelles</span>, 37:547&#8211;579, 1901.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XJukesCantor1969"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>TH&#x00A0;Jukes and CR&#x00A0;Cantor.
          <span
                  class="cmssi-10">Evolution of Protein Molecules</span>, pages 21&#8211;132. Academic Press, New York,
          1969.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XKimura1980"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>M.&#x00A0;Kimura. A simple method
          for estimating evolutionary rates of base substitutions through comparative
          studies of nucleotide sequences. <span
                  class="cmssi-10">Journal of Molecular Evolution</span>, 16(2):111&#8211;120, 1980.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XLapointeCucumel1997"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>FJ&#x00A0;Lapointe and
          G.&#x00A0;Cucumel. The average consensus procedure: Combination of weighted trees containing
          identical or overlapping sets of taxa. <span
                  class="cmssi-10">Systematic Biology</span>, 46(2):306&#8211;312, 1997.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XLeighBryant2015"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>J.&#x00A0;W. Leigh and D.&#x00A0;Bryant.
          PopART: Full-feature software for haplotype network construction. <span
                  class="cmssi-10">Methods in</span>
          <span
                  class="cmssi-10">Ecology and Evolution</span>, 6(9):1110&#8211;1116, 2015.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XMaddisonetal1997"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>DR&#x00A0;Maddison, DL&#x00A0;Swofford,
          and WP&#x00A0;Maddison. NEXUS: An extensible file format for systematic information.
          <span
                  class="cmssi-10">Systematic Biology</span>, 46(4):590&#8211;621, 1997.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XNeiLi1979"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>M.&#x00A0;Nei and WH&#x00A0;Li.
          Mathematical model for studying genetic variation in terms of restriction endonucleases.
          <span
                  class="cmssi-10">Proceedings of the National Academy of Sciences</span>, 79(1):5269&#8211;5273, 1979.
     </p>


     <p class="bibitem"><span class="biblabel">
<a
        id="XOndovetal2016"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>BD&#x00A0;Ondov, TJ&#x00A0;Treangen,
          P.&#x00A0;Melsted, AB&#x00A0;Mallonee, NH&#x00A0;Bergman, S.&#x00A0;Koren, and AM&#x00A0;Phillippy. Mash: fast
          genome and metagenome distance estimation using minhash. <span
                  class="cmssi-10">Genome Biol</span>, 17(132), 2016.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XParksetal2018"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>Donovan&#x00A0;H. Parks, Maria
          Chuvochina, D.&#x00A0;W. Waite, Christian Rinke, A.&#x00A0;Skarshewski, P.&#x00A0;A. Chaumeil, and
          Phil Hugenholtz. A standardized bacterial taxonomy based on genome phylogeny substantially revises the tree
          of life. <span
                  class="cmssi-10">Nature Biotechnology</span>, 36(10):996&#8211;1004, 2018.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XRedding2003"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>D.&#x00A0;Redding. Incorporating
          genetic distinctness and reserve occupancy into a conservation priorisation approach.
          Master&#8217;s thesis, University of East Anglia, 2003.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XReddingMooers2006"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>DW&#x00A0;Redding and AO&#x00A0;Mooers.
          Incorporating evolutionary measures into conservation prioritization.
          <span
                  class="cmssi-10">Conservation Biology</span>, 20:1670&#8211;1678, 2006.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XSaitouNei1987"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>N.&#x00A0;Saitou and M.&#x00A0;Nei.
          The neighbor-joining method: a new method for reconstructing phylogenetic trees.
          <span
                  class="cmssi-10">Molecular Biology and Evolution</span>, 4:406&#8211;425, 1987.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XScornavaccaetal2011"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>C.&#x00A0;Scornavacca,
          F.&#x00A0;Zickmann, and DH&#x00A0;Huson. Tanglegrams for rooted phylogenetic trees and networks.
          <span
                  class="cmssi-10">Bioinformatics</span>, 27(13):i248&#8211;i256, 2011.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XSneletal1997"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>B.&#x00A0;Snel, P.&#x00A0;Bork,
          and M.&#x00A0;A. Huynen. Genome phylogeny based on gene content. <span
                  class="cmssi-10">Nature Genetics</span>, 21:108&#8211;110,
          1997.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XSokalMichener1958"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>RR&#x00A0;Sokal and CD&#x00A0;Michener.
          A statistical method for evaluating systematic relationships. <span
                  class="cmssi-10">University of Kansas</span>
          <span
                  class="cmssi-10">Scientific Bulletin</span>, 28:1409&#8211;1438, 1958.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XSteel1994"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>MA&#x00A0;Steel. Recovering a tree
          from the leaf colorations it generates under a markov model. <span
                  class="cmssi-10">Appl. Math. Lett.</span>,
          7(2):19&#8211;24, 1994.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XSteeletal2000"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>MA&#x00A0;Steel, DH&#x00A0;Huson,
          and PJ&#x00A0;Lockhart. Invariable site models and their use in phylogeny reconstruction.
          <span
                  class="cmssi-10">Sys. Biol.</span>, 49(2):225&#8211;232, 2000.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XSwoffordetal1996"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>DL&#x00A0;Swofford, GJ&#x00A0;Olsen,
          PJ&#x00A0;Waddell, and DM&#x00A0;Hillis. <span
                  class="cmssi-10">Phylogenetic  inference</span>, pages 407&#8211;514. Sinauer
          Associates, Inc., 2nd edition, 1996.
     </p>


     <p class="bibitem"><span class="biblabel">
<a
        id="XTajima1989"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>F.&#x00A0;Tajima. Statistical
          method for testing the neutral mutation hypothesis by DNA polymorphism. <span
                  class="cmssi-10">Genetics</span>,
          123(3):585&#8211;595, 1989.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XTavar1986"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>S.&#x00A0;Tavar. Some
          probabilistic and statistical problems in the analysis of dna sequences. <span
                  class="cmssi-10">Lectures on Mathematics</span>
          <span
                  class="cmssi-10">in the Life Sciences</span>, 17:57&#8211;86, 1986.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XUpholt1977"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>WB&#x00A0;Upholt. Estimation of
          dna sequence divergence from comparison of restriction endonuclease digests.
          <span
                  class="cmssi-10">Nucleic Acids Res.</span>, 4(5):1257&#8211;1265, 1977.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XVolkmannetal2014"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>L&#x00A0;Volkmann, I&#x00A0;Martyn,
          V&#x00A0;Moulton, A&#x00A0;Spillner, and AO&#x00A0;Mooers. Prioritizing populations for conservation
          using phylogenetic networks. <span
                  class="cmssi-10">PLoS ONE</span>, 9(2):e88945, 2014.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XWhitfieldetal2008"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>JB&#x00A0;Whitfield, SA&#x00A0;Cameron,
          DH&#x00A0;Huson, and MA&#x00A0;Steel. Filtered z-closure supernetworks for extracting and
          visualizing recurrent signal from incongruent gene trees. <span
                  class="cmssi-10">Systematic Biology</span>, 57(6):939&#8211;947, 2008.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XZhangetal2023"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>L.&#x00A0;Zhang, N.&#x00A0;Abhari,
          C.&#x00A0;Colijn, and Y.&#x00A0;Wu. A fast and scalable method for inferring phylogenetic networks
          from trees by aligning lineage taxon strings. <span
                  class="cmssi-10">Genome Res</span>, 2023.
     </p>
     <p class="bibitem"><span class="biblabel">
<a
        id="XZhangetal2024"></a><span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span>L.&#x00A0;Zhang, B.&#x00A0;Cetinkaya,
          and DH&#x00A0;Huson. Phylofusion- fast and easy fusion of rooted phylogenetic trees into
          a network. in preparation, 2024.
     </p>
</div>


</body>
</html>

                                                                                         


